Subroutine documentation generated on Fri 12 Dec 2025 17:37:54 AEDT
 


## Subroutine Index 

[Aa](#anchor0)                             /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Aa_chirality](#anchor1)                   /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Aa_one_letter](#anchor2)                  /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Aa_three_letter](#anchor3)                /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Acceptable_neutral_valences](#anchor4)    /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Add_bondorders_to_atom_tree](#anchor5)    /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Add_layer](#anchor6)                      /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Add_monomer](#anchor7)                    /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Add_option](#anchor8)                     /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Add_part_of_layer](#anchor9)              /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Align_bonds](#anchor10)                    /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Amino_acid_and_nucleotide_base_codes](#anchor11) /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Angle_between_vectors](#anchor12)          /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Asl_substring_to_specifiers](#anchor13)    /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Atom_add_1h](#anchor14)                    /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Atom_add_2h](#anchor15)                    /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Atom_add_3h](#anchor16)                    /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Atom_add_h](#anchor17)                     /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Atom_calc_bin](#anchor18)                  /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Atom_calc_bin_](#anchor19)                 /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Atom_check_valence](#anchor20)             /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Atom_colour](#anchor21)                    /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Atom_cross_prod](#anchor22)                /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Atom_data_to_sdf](#anchor23)               /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Atom_data_to_sdf](#anchor24)               /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Atom_formal_charge](#anchor25)             /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Atom_guess_element](#anchor26)             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Atom_guess_element_mol2](#anchor27)        /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Atom_keys](#anchor28)                      /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Atom_lone_pairs](#anchor29)                /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Atom_norm_vector](#anchor30)               /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Atom_number_sort_function](#anchor31)      /vcp1/people/david/bin/Silico1.15/bin/write_itp <br>

[Atom_number_sort_function](#anchor32)      /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Atom_oxidation_state](#anchor33)           /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Atom_path_traverse](#anchor34)             /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Atom_path_traverse_branched](#anchor35)    /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Atom_printout](#anchor36)                  /vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm <br>

[Atom_score_hybridisation](#anchor37)       /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Atom_smiles_label](#anchor38)              /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Atom_specifier_setup](#anchor39)           /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Atom_vector](#anchor40)                    /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Atomic_elements](#anchor41)                /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Atominfo](#anchor42)                       /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Base16](#anchor43)                         /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Base36](#anchor44)                         /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Bin_values](#anchor45)                     /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Bl](#anchor46)                             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bndangle](#anchor47)                       /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Bond_create](#anchor48)                    /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bond_create_atom](#anchor49)               /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bond_delete](#anchor50)                    /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bond_delete_all](#anchor51)                /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bond_delete_atom](#anchor52)               /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bond_modify_order](#anchor53)              /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bond_modify_order_check](#anchor54)        /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bond_vector](#anchor55)                    /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Bonded](#anchor56)                         /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bondlengths](#anchor57)                    /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Bondlengths_2](#anchor58)                  /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Branchsort](#anchor59)                     /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Calc_area_sphere_with_holes](#anchor60)    /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Calc_binsize](#anchor61)                   /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Calc_box_size](#anchor62)                  /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Calc_chiral_sdf_stereo](#anchor63)         /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Calc_chiral_sdf_stereo](#anchor64)         /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Calc_chiral_volume](#anchor65)             /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Calc_chiral_volume](#anchor66)             /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Calc_conservation_group_membership](#anchor67) /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Calc_conservation_group_percentage](#anchor68) /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Calc_distance_function](#anchor69)         /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Calc_human_readable_time](#anchor70)       /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Calc_physiological_protonation](#anchor71) /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Calc_rotation_vector_onto_vector](#anchor72) /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Calc_valence_change](#anchor73)            /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Calculate_bin](#anchor74)                  /vcp1/people/david/bin/Silico1.15/bin/rdf <br>

[Calculate_one_rdf](#anchor75)              /vcp1/people/david/bin/Silico1.15/bin/rdf <br>

[Calculate_rdfs](#anchor76)                 /vcp1/people/david/bin/Silico1.15/bin/rdf <br>

[Carbanion_filter_neutraliser](#anchor77)   /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Centroid](#anchor78)                       /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Centroid_periodic](#anchor79)              /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Centroid_periodic_bv](#anchor80)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Ch](#anchor81)                             /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Change_bondorder](#anchor82)               /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Check_arguments](#anchor83)                /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Check_data_type](#anchor84)                /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Check_grace_output_format](#anchor85)      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Check_itp_is_bonded](#anchor86)            /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Check_log](#anchor87)                      /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Check_log](#anchor88)                      /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Check_number_components](#anchor89)        /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Check_printerror](#anchor90)               /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Checkvalence](#anchor91)                   /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Chiral_volume](#anchor92)                  /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Chonspx_filter](#anchor93)                 /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Clash_periodic_binned](#anchor94)          /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Clean_pdb_line](#anchor95)                 /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Clean_up_split_submols](#anchor96)         /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Cleanline](#anchor97)                      /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Cleansdf](#anchor98)                       /vcp1/people/david/bin/Silico1.15/bin/sdf_merge <br>

[Clear_smiles_sets](#anchor99)              /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Close_contact](#anchor100)                  /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Close_file](#anchor101)                     /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Close_molfile](#anchor102)                  /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Combine](#anchor103)                        /vcp1/people/david/bin/Silico1.15/bin/mol_combine <br>

[Compare_atom_order](#anchor104)             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Condense_smiles_ranks](#anchor105)          /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Connect_atoms_by_distance](#anchor106)      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Connect_atoms_by_distance_old](#anchor107)  /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Connect_atoms_by_distance_simple](#anchor108) /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Connect_atoms_by_distance_simple_bv_periodic](#anchor109) /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Connected](#anchor110)                      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Construct_atom_relationships](#anchor111)   /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Construct_family_tree](#anchor112)          /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Construct_residue_relationships](#anchor113) /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Convert_aromatic_bonds](#anchor114)         /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Convert_aromatic_bonds_kekule](#anchor115)  /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Copy_atom](#anchor116)                      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Copy_minimal_molecule](#anchor117)          /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Correlation_coeff](#anchor118)              /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Corresponding_prime_old](#anchor119)        /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Count_formal_charges](#anchor120)           /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Count_polar_nonpolar](#anchor121)           /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Create_angle](#anchor122)                   /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Create_molecule](#anchor123)                /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Create_planar_improper](#anchor124)         /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Create_water_molecule](#anchor125)          /vcp1/people/david/bin/Silico1.15/lib/silico_solvate.pm <br>

[Crossprod](#anchor126)                      /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Csv_fix](#anchor127)                        /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Deep_copy](#anchor128)                      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Deep_copy_orig](#anchor129)                 /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Define_pdb_chain_segid](#anchor130)         /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Deg_to_rad](#anchor131)                     /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Delete_atomset](#anchor132)                 /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Delete_flag](#anchor133)                    /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Delete_rings](#anchor134)                   /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Delimited_read_error](#anchor135)           /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Distance](#anchor136)                       /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_atoms_min](#anchor137)             /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_points_xyz](#anchor138)   /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_points_xyz_mat](#anchor139) /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_signed_xyz](#anchor140)   /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_sq](#anchor141)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_sq_bv](#anchor142)        /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_xyz](#anchor143)          /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_xyz_bv](#anchor144)       /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_periodic_xyz_old](#anchor145)      /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_point_atoms_min](#anchor146)       /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_points](#anchor147)                /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Distance_sq](#anchor148)                    /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Divide](#anchor149)                         /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Dotprod](#anchor150)                        /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Double_and_tie_break](#anchor151)           /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Dump_itp_data](#anchor152)                  /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Edit_mdpfile](#anchor153)                   /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Edit_mdpfile](#anchor154)                   /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Electronegativities](#anchor155)            /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Element_number2symbol](#anchor156)          /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Element_symbol2number](#anchor157)          /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[En](#anchor158)                             /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Ens](#anchor159)                            /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ens_smiles_string](#anchor160)              /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Ens_smiles_string_chemaxon](#anchor161)     /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Ensemble_calc_fragments](#anchor162)        /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Ensemble_check](#anchor163)                 /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Ensemble_cluster](#anchor164)               /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ensemble_connect_atoms](#anchor165)         /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ensemble_consolidate_nocopy](#anchor166)    /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Ensemble_consolidate_old](#anchor167)       /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ensemble_consolidate_small_molecules](#anchor168) /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Ensemble_consolidate{](#anchor169)          /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ensemble_copy](#anchor170)                  /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ensemble_insight_rename](#anchor171)        /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ensemble_print_connect](#anchor172)         /vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm <br>

[Ensemble_printout](#anchor173)              /vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm <br>

[Ensemble_renumber](#anchor174)              /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Ensemble_split](#anchor175)                 /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Ensemble_split_keeping_largest_mols](#anchor176) /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Et](#anchor177)                             /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Euclidian](#anchor178)                      /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Extract_mol2_sdf_data](#anchor179)          /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Fgroup](#anchor180)                         /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[File_modified_since](#anchor181)            /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[File_read_error](#anchor182)                /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[File_write_error](#anchor183)               /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Find_aggregates](#anchor184)                /vcp1/people/david/bin/Silico1.15/bin/find_aggregate <br>

[Find_atom_paths](#anchor185)                /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Find_atom_rings](#anchor186)                /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Find_atom_rings_not_recursive](#anchor187)  /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Find_available_localhost_cpu](#anchor188)   /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Find_available_localhost_gpu](#anchor189)   /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Find_bondorder](#anchor190)                 /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Find_bondorder2](#anchor191)                /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Find_bondorders_ring](#anchor192)           /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Find_close_atoms_binned](#anchor193)        /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Find_closest_atoms_periodic](#anchor194)    /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Find_closest_atoms_periodic_binned](#anchor195) /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Find_decoys_actives](#anchor196)            /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Find_elements_ring](#anchor197)             /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Find_hbonds](#anchor198)                    /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Find_host_queue](#anchor199)                /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Find_hostname](#anchor200)                  /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Find_intermolecular_hbonds_h](#anchor201)   /vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm <br>

[Find_intermolecular_hbonds_h_simple](#anchor202) /vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm <br>

[Find_intermolecular_hbonds_noh](#anchor203) /vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm <br>

[Find_intramolecular_hbonds_h](#anchor204)   /vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm <br>

[Find_nearby_charges_recursive](#anchor205)  /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Find_nearest_atoms](#anchor206)             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Find_ngpu](#anchor207)                      /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Find_nproc](#anchor208)                     /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Find_polar_atoms](#anchor209)               /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Find_schrodinger_exe](#anchor210)           /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Find_substructure](#anchor211)              /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Findclash](#anchor212)                      /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Fix_cell_values](#anchor213)                /vcp1/people/david/bin/Silico1.15/bin/find_aggregate <br>

[Fix_dative_bonds](#anchor214)               /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Fix_double_bonds_carbonyl](#anchor215)      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Fix_double_bonds_carboxyl](#anchor216)      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Fix_double_bonds_general](#anchor217)       /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Fix_hydrogens](#anchor218)                  /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Fix_pdb_chains](#anchor219)                 /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Flag_is_boolean](#anchor220)                /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Fnfp](#anchor221)                           /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Format_doc_html](#anchor222)                /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Format_doc_html_header_footer](#anchor223)  /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Format_doc_markdown](#anchor224)            /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Format_doc_txt](#anchor225)                 /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Format_oname](#anchor226)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Format_string_html](#anchor227)             /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Formatdata](#anchor228)                     /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Formatpara_txt](#anchor229)                 /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Formula_string](#anchor230)                 /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Fr_printout](#anchor231)                    /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Frag_aray_to_hashes](#anchor232)            /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Fragment_printout](#anchor233)              /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Fragmentlist_printout](#anchor234)          /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Ftfp](#anchor235)                           /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Fundamental_constants](#anchor236)          /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[General_distance_sq](#anchor237)            /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[General_distance_xyz](#anchor238)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Get_aa_frequency](#anchor239)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Get_allflags](#anchor240)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_arguments2](#anchor241)                 /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_atomarray_using_specifier](#anchor242)  /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Get_atomnums_spechash](#anchor243)          /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Get_atomnums_using_specifier](#anchor244)   /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Get_atoms_using_specifier](#anchor245)      /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Get_default_flags_standard](#anchor246)     /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_file_compression](#anchor247)           /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_file_dir](#anchor248)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_filebase](#anchor249)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_filebase_short](#anchor250)             /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_fileext](#anchor251)                    /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_fitness_strings](#anchor252)            /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Get_flag](#anchor253)                       /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_gmx_cmd](#anchor254)                    /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Get_hosttype](#anchor255)                   /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Get_ion_data](#anchor256)                   /vcp1/people/david/bin/Silico1.15/bin/water_to_ion <br>

[Get_join_list](#anchor257)                  /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Get_key](#anchor258)                        /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Get_lflag](#anchor259)                      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_membrane_cylinder_atoms](#anchor260)    /vcp1/people/david/bin/Silico1.15/bin/make_index <br>

[Get_metal_oxidation_state](#anchor261)      /vcp1/people/david/bin/Silico1.15/bin/water_to_ion <br>

[Get_ofilebase](#anchor262)                  /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_ofilename](#anchor263)                  /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_ofilename2](#anchor264)                 /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_oformat](#anchor265)                    /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_polar_nonpolar_centres](#anchor266)     /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Get_sflag](#anchor267)                      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_std_output_options](#anchor268)         /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Get_tempname](#anchor269)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Getcol](#anchor270)                         /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Getcols](#anchor271)                        /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Getcols_all](#anchor272)                    /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Getcols_not](#anchor273)                    /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Getheaders_all](#anchor274)                 /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Getlines](#anchor275)                       /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Getrows_all](#anchor276)                    /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Gmx_make_heavy_h](#anchor277)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Gmx_merge_coords_into_top](#anchor278)      /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Gmx_replace_peptide_dihedrals](#anchor279)  /vcp1/people/david/bin/Silico1.15/bin/write_itp <br>

[Gmx_replace_peptide_dihedrals](#anchor280)  /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Gnuplot_plot_bins](#anchor281)              /vcp1/people/david/bin/Silico1.15/lib/silico_gnuplot.pm <br>

[Grom_putline](#anchor282)                   /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Grom_readline](#anchor283)                  /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Gromacs_check_itp](#anchor284)              /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Gromacs_regenerate_pairs](#anchor285)       /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Heading](#anchor286)                        /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Host_is_slurm_cluster](#anchor287)          /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Identify_heterocycle](#anchor288)           /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Insight_rename](#anchor289)                 /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Is_aa_backbone](#anchor290)                 /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Is_gap](#anchor291)                         /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Is_hbond_mcdonald_thornton](#anchor292)     /vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm <br>

[Is_integer](#anchor293)                     /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Is_ring_aromatic_by_bondorder](#anchor294)  /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Is_ring_aromatic_by_connectivity](#anchor295) /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Is_ring_planar](#anchor296)                 /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Is_ring_planar_by_dihedral](#anchor297)     /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Itp_delete_angle_parameters_atom](#anchor298) /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_delete_parameters_atom](#anchor299)     /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_make_angle](#anchor300)                 /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_make_bond](#anchor301)                  /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_make_dihedral](#anchor302)              /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_make_dihedral_restraint](#anchor303)    /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_make_pair](#anchor304)                  /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_make_settle](#anchor305)                /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Itp_restrain_dihedrals](#anchor306)         /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Label_aa_backbone](#anchor307)              /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_amides](#anchor308)                   /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_amino_acids](#anchor309)              /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Label_dendrimer_residues](#anchor310)       /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Label_fg_alcohol](#anchor311)               /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_aromatic](#anchor312)              /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_benzhydrylamine](#anchor313)       /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_carbanion](#anchor314)             /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_carbon](#anchor315)                /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_carbonyl](#anchor316)              /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_halogen](#anchor317)               /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_nitrogen](#anchor318)              /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_oxygen](#anchor319)                /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_phosphorous](#anchor320)           /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_planar_carbon](#anchor321)         /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_sulphur](#anchor322)               /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_fg_tfm](#anchor323)                   /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_solvent](#anchor324)                  /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Label_waters](#anchor325)                   /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Lcss](#anchor326)                           /vcp1/people/david/bin/Silico1.15/bin/sdf_merge <br>

[Lipophilicity_dipole](#anchor327)           /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Mae_get_tok](#anchor328)                    /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mae_ignore_block](#anchor329)               /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mae_parse_atoms](#anchor330)                /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mae_parse_block_header](#anchor331)         /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mae_parse_bonds](#anchor332)                /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mae_parse_ct_data](#anchor333)              /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mae_split_data](#anchor334)                 /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Magnitude](#anchor335)                      /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Make_alternate_aa_forms](#anchor336)        /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Make_amides_trans](#anchor337)              /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Make_aromatic_bonds](#anchor338)            /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Make_atom_names_unique](#anchor339)         /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Make_atomset](#anchor340)                   /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Make_atomset_colour](#anchor341)            /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Make_atomset_fg](#anchor342)                /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Make_atomset_formal_charge](#anchor343)     /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Make_bins](#anchor344)                      /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Make_dendrimer](#anchor345)                 /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Make_flag](#anchor346)                      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Make_gromacs_cell_from_pdb_cell_and_angles](#anchor347) /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Make_gromacs_itp](#anchor348)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Make_hist](#anchor349)                      /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Make_molecule_copies](#anchor350)           /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Make_smiles_string_from_list](#anchor351)   /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Make_subset](#anchor352)                    /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Make_types_summary](#anchor353)             /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Make_vectors_connected](#anchor354)         /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Makeboxmolecule](#anchor355)                /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Makedir](#anchor356)                        /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Match_amino_acid](#anchor357)               /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Match_amino_acid2](#anchor358)              /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Maximum](#anchor359)                        /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Maximum_minus_mean](#anchor360)             /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Maximum_valences](#anchor361)               /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Mdrun_log_is_complete](#anchor362)          /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Mdrun_log_is_complete](#anchor363)          /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Mean](#anchor364)                           /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Mean_minus_minimum](#anchor365)             /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Median](#anchor366)                         /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Median_absolute_deviation](#anchor367)      /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Memuse](#anchor368)                         /vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm <br>

[Merck_calculate_formal_charge](#anchor369)  /vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm <br>

[Merck_charge_code_to_formal](#anchor370)    /vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm <br>

[Merck_formal_to_charge_code](#anchor371)    /vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm <br>

[Merge_filename](#anchor372)                 /vcp1/people/david/bin/Silico1.15/bin/sdf_merge <br>

[Merge_itp_coord](#anchor373)                /vcp1/people/david/bin/Silico1.15/bin/write_itp <br>

[Merge_itp_coord](#anchor374)                /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Minimise_dendrimer](#anchor375)             /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Minimum](#anchor376)                        /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Mmod_anneal](#anchor377)                    /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_calc_energy](#anchor378)               /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_csearch](#anchor379)                   /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_csearch_write_com](#anchor380)         /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_logfile_normal_termination](#anchor381) /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_logfile_num_conf](#anchor382)          /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_make_complex](#anchor383)              /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_min](#anchor384)                       /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_min_h](#anchor385)                     /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_mult_min](#anchor386)                  /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_qikprop](#anchor387)                   /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mmod_run](#anchor388)                       /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mol2_atom_record](#anchor389)               /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_bond_record](#anchor390)               /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_centroid_record](#anchor391)           /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_crystal_record](#anchor392)            /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_fastsplit](#anchor393)                 /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Mol2_molecule_record](#anchor394)           /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_process_comments](#anchor395)          /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_set_formal_charges](#anchor396)        /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_set_record](#anchor397)                /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol2_substructure_record](#anchor398)       /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol_add_atom](#anchor399)                   /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_add_hydrogens](#anchor400)              /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Mol_add_hydrogens_nocoords](#anchor401)     /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Mol_add_lone_pairs](#anchor402)             /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Mol_aromatic_smiles](#anchor403)            /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Mol_bonded_interactions](#anchor404)        /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_bv_mat](#anchor405)                     /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_calc_charge](#anchor406)                /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_calc_hcount](#anchor407)                /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Mol_check](#anchor408)                      /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_all](#anchor409)                  /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_amides](#anchor410)               /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_aromatic_bonds](#anchor411)       /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_atom_overlap](#anchor412)         /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_bondlengths](#anchor413)          /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_bondorders](#anchor414)           /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_cell](#anchor415)                 /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_check_distorted_rings](#anchor416)      /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_elements](#anchor417)             /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_check_unit_cell](#anchor418)            /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_check_valences](#anchor419)             /vcp1/people/david/bin/Silico1.15/lib/silico_check.pm <br>

[Mol_copy_rec](#anchor420)                   /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Mol_copy_unit_cell](#anchor421)             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_count_heavy_atoms](#anchor422)          /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_count_residue_contacts](#anchor423)     /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Mol_count_rot_bonds](#anchor424)            /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_count_valence_electrons](#anchor425)    /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Mol_delete_duplicate_atoms](#anchor426)     /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_find_all_angles](#anchor427)            /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_find_all_bonds](#anchor428)             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_find_all_dihedrals](#anchor429)         /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_find_chiral_impropers](#anchor430)      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_find_donors_acceptors](#anchor431)      /vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm <br>

[Mol_find_neighbouring_atoms](#anchor432)    /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_find_planar_impropers](#anchor433)      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_fix_atom_names](#anchor434)             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_fix_multiple_bonds_general](#anchor435) /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_fix_sodiums](#anchor436)                /vcp1/people/david/bin/Silico1.15/bin/make_index <br>

[Mol_fragment](#anchor437)                   /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Mol_fragment_and_path](#anchor438)          /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Mol_get_functional_group](#anchor439)       /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_get_sequence](#anchor440)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Mol_has_h](#anchor441)                      /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Mol_label_alpha_fg](#anchor442)             /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_label_beta_fg](#anchor443)              /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_label_functional_group](#anchor444)     /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_label_heterocycles](#anchor445)         /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_label_smiles_invariants](#anchor446)    /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Mol_label_united_atom](#anchor447)          /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Mol_map_sequence](#anchor448)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Mol_move_atoms_into_cell](#anchor449)       /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_move_into_cell](#anchor450)             /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_move_into_cell_using_cell_vectors](#anchor451) /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Mol_orient_polarity](#anchor452)            /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_partition_on_bond](#anchor453)          /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_partition_on_bond2](#anchor454)         /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_path_to_fragments](#anchor455)          /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Mol_print_warnings](#anchor456)             /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Mol_rama](#anchor457)                       /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Mol_rama_angles](#anchor458)                /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Mol_read_error](#anchor459)                 /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Mol_rename_atoms](#anchor460)               /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_renumber_residues](#anchor461)          /vcp1/people/david/bin/Silico1.15/bin/make_index <br>

[Mol_renumber_substructures](#anchor462)     /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_rescale_bonds](#anchor463)              /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_rotate_euler{](#anchor464)              /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Mol_score_hybridisation](#anchor465)        /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_sdf_stereo](#anchor466)                 /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Mol_sdf_stereo](#anchor467)                 /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Mol_set_source_data](#anchor468)            /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Mol_smiles_string](#anchor469)              /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Mol_sort_smiles](#anchor470)                /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Mol_tanimoto](#anchor471)                   /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Mol_transfer_coordinates](#anchor472)       /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mol_type_mmod](#anchor473)                  /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Mol_type_sybyl](#anchor474)                 /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Mol_warn](#anchor475)                       /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Molecule_bin_atoms](#anchor476)             /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_bin_atoms_old](#anchor477)         /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_centre](#anchor478)                /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_centre_of_mass](#anchor479)        /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_check_and_fix_connectivity](#anchor480) /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_check_join_points](#anchor481)     /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Molecule_connect_pdb_atoms_by_name](#anchor482) /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Molecule_copy_coordinates](#anchor483)      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_copy_header](#anchor484)           /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_copy_noatoms](#anchor485)          /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_count_numbonds](#anchor486)        /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_delete_alternate_atoms](#anchor487) /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Molecule_delete_atom](#anchor488)           /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_delete_atoms](#anchor489)          /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_delete_dummy](#anchor490)          /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_delete_h](#anchor491)              /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Molecule_delete_h_resname](#anchor492)      /vcp1/people/david/bin/Silico1.15/bin/mol_del_h <br>

[Molecule_delete_nonpolar_h](#anchor493)     /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Molecule_delete_polar_h](#anchor494)        /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Molecule_dimensions](#anchor495)            /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_find_planar_atoms](#anchor496)     /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Molecule_find_planar_atoms_bondorders](#anchor497) /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Molecule_find_planar_atoms_connections](#anchor498) /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Molecule_find_planar_atoms_geom](#anchor499) /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Molecule_find_rings](#anchor500)            /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Molecule_fix_and_get_residues](#anchor501)  /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Molecule_fix_sybyl_substructures](#anchor502) /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Molecule_formal_charge](#anchor503)         /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Molecule_formula](#anchor504)               /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Molecule_formula_string](#anchor505)        /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Molecule_generate_bondorders](#anchor506)   /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_generate_bonds](#anchor507)        /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_get_chains](#anchor508)            /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_get_residues](#anchor509)          /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Molecule_get_residues_hash](#anchor510)     /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Molecule_label_submols](#anchor511)         /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Molecule_make_residues](#anchor512)         /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Molecule_mw](#anchor513)                    /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Molecule_pack](#anchor514)                  /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_path](#anchor515)                  /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Molecule_pdb_rename_backbone](#anchor516)   /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Molecule_pdb_rename_h](#anchor517)          /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Molecule_pdb_rename_h_dataread](#anchor518) /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Molecule_printout](#anchor519)              /vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm <br>

[Molecule_random_move_cartesian](#anchor520) /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_renumber](#anchor521)              /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_renumber2](#anchor522)             /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_renumber2_new](#anchor523)         /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_reorder](#anchor524)               /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Molecule_rot_angle_x](#anchor525)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_rot_angle_y](#anchor526)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_rot_angle_z](#anchor527)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_rotate](#anchor528)                /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_rotate_axis](#anchor529)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_solvate](#anchor530)               /vcp1/people/david/bin/Silico1.15/lib/silico_solvate.pm <br>

[Molecule_split](#anchor531)                 /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Molecule_split_keeping_largest_mol](#anchor532) /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Molecule_trans_to_centre](#anchor533)       /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molecule_translate](#anchor534)             /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Molfile_divide](#anchor535)                 /vcp1/people/david/bin/Silico1.15/bin/mol_divide <br>

[Mols](#anchor536)                           /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Mopac_finalise](#anchor537)                 /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Name_pdb_molecule](#anchor538)              /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[New_h](#anchor539)                          /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[New_table_delimited](#anchor540)            /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Number_range](#anchor541)                   /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Open_file](#anchor542)                      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Open_molfile](#anchor543)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Optimise_geometry](#anchor544)              /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Origins](#anchor545)                        /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Orthogonal_least_squares_plane_atoms](#anchor546) /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Overwrite_check](#anchor547)                /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Pack_gromacs_angles](#anchor548)            /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Pack_molecules](#anchor549)                 /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Parse_amino_acids_smiles](#anchor550)       /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Parse_comma_separated_numbers](#anchor551)  /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Parse_pdb_space_group](#anchor552)          /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Parse_residue_dictionary](#anchor553)       /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Parse_smiles](#anchor554)                   /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Pdb_cell_to_gromacs_bv](#anchor555)         /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Pdb_fastsplit](#anchor556)                  /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Pdb_format_atom_name](#anchor557)           /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_format_atom_number](#anchor558)         /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_format_residue_number](#anchor559)      /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_format_subname](#anchor560)             /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_get_ligands](#anchor561)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_get_prot](#anchor562)                   /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_increment_chain](#anchor563)            /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_molecule_sort_atoms](#anchor564)        /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Pdb_rewind](#anchor565)                     /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Plot_amides](#anchor566)                    /vcp1/people/david/bin/Silico1.15/bin/mol_amides <br>

[Pmmod](#anchor567)                          /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Primes](#anchor568)                         /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Print_finished_message](#anchor569)         /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Print_flags](#anchor570)                    /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Print_keys](#anchor571)                     /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Print_timing_message](#anchor572)           /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Print_title](#anchor573)                    /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Printbox](#anchor574)                       /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Printdoc](#anchor575)                       /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Printit](#anchor576)                        /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Process_name](#anchor577)                   /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Prompt](#anchor578)                         /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Prune_family_tree](#anchor579)              /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Quaternion](#anchor580)                     /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Quick_finalise](#anchor581)                 /vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm <br>

[Quiet](#anchor582)                          /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Rad_to_deg](#anchor583)                     /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Rama_find_backbone_dihedrals](#anchor584)   /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Ramaplot](#anchor585)                       /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Ramaplot_by_mol_open](#anchor586)           /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Ramaplot_open](#anchor587)                  /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Randomchar](#anchor588)                     /vcp1/people/david/bin/Silico1.15/bin/mol_characterise <br>

[Randomise_and_test_molecule_position](#anchor589) /vcp1/people/david/bin/Silico1.15/bin/bilayer_builder <br>

[Range](#anchor590)                          /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Rank_by_neighbouring_primes](#anchor591)    /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Rank_molecules](#anchor592)                 /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Rank_smiles_set](#anchor593)                /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Read_atom_specifier_file](#anchor594)       /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Read_clustal](#anchor595)                   /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Read_coord_mol](#anchor596)                 /vcp1/people/david/bin/Silico1.15/bin/write_itp <br>

[Read_coord_mol](#anchor597)                 /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_divcon_cart](#anchor598)               /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Read_doc](#anchor599)                       /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Read_doc_subroutine](#anchor600)            /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Read_fasta](#anchor601)                     /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Read_function_line](#anchor602)             /vcp1/people/david/bin/Silico1.15/bin/rdf <br>

[Read_gamess_cart](#anchor603)               /vcp1/people/david/bin/Silico1.15/lib/silico_gamess.pm <br>

[Read_gaussian_cart](#anchor604)             /vcp1/people/david/bin/Silico1.15/lib/silico_gaussian.pm <br>

[Read_gromacs_gro](#anchor605)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_gro_molecule](#anchor606)      /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_gro_single](#anchor607)        /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_indexfile](#anchor608)         /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_itp](#anchor609)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_mdp](#anchor610)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_rtp](#anchor611)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_top](#anchor612)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_gromacs_trr](#anchor613)               /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_maestro](#anchor614)                   /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Read_maestro_single](#anchor615)            /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Read_merck](#anchor616)                     /vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm <br>

[Read_mmod](#anchor617)                      /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Read_mmod_mmo](#anchor618)                  /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Read_mol2](#anchor619)                      /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Read_mol2_single](#anchor620)               /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Read_mol_any](#anchor621)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Read_mol_single](#anchor622)                /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Read_mopac_cart](#anchor623)                /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Read_mopac_dat_single](#anchor624)          /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Read_packings](#anchor625)                  /vcp1/people/david/bin/Silico1.15/bin/micelle <br>

[Read_pdb](#anchor626)                       /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_atom](#anchor627)                  /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_atom_fast](#anchor628)             /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_compnd](#anchor629)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_conect](#anchor630)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_connected](#anchor631)             /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_cryst](#anchor632)                 /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_expdta](#anchor633)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_fast](#anchor634)                  /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_header_line](#anchor635)           /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_hetnam](#anchor636)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_jrnl](#anchor637)                  /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_molecule](#anchor638)              /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_remark](#anchor639)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_single](#anchor640)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_title](#anchor641)                 /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_pdb_user](#anchor642)                  /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Read_quick_in_single](#anchor643)           /vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm <br>

[Read_quick_out](#anchor644)                 /vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm <br>

[Read_rdf](#anchor645)                       /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Read_rdf](#anchor646)                       /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Read_rdf_input_file](#anchor647)            /vcp1/people/david/bin/Silico1.15/bin/rdf <br>

[Read_rtp_single](#anchor648)                /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Read_sdf](#anchor649)                       /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Read_sdf](#anchor650)                       /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Read_sdf_atoms_bonds](#anchor651)           /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Read_sdf_atoms_bonds](#anchor652)           /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Read_sdf_single](#anchor653)                /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Read_sdf_single](#anchor654)                /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Read_seq_any](#anchor655)                   /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Read_sleuth_abstract](#anchor656)           /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Read_smiles](#anchor657)                    /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Read_smiles_single](#anchor658)             /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Read_table_delimited](#anchor659)           /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Read_tcoffee_score](#anchor660)             /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Read_textfile](#anchor661)                  /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Read_tinker_xyz](#anchor662)                /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Read_tinker_xyz_single](#anchor663)         /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Read_xyz](#anchor664)                       /vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm <br>

[Read_xyz_single](#anchor665)                /vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm <br>

[Record_best_coords](#anchor666)             /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Remove_aromatic_bondorders](#anchor667)     /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Remove_duplicates](#anchor668)              /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Remove_option](#anchor669)                  /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Reos_filter](#anchor670)                    /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Rescue_unbonded_hydrogens](#anchor671)      /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Residue_printout](#anchor672)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Residues](#anchor673)                       /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Revert_to_best_coords](#anchor674)          /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Rgmx_setup](#anchor675)                     /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Rms](#anchor676)                            /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Rms_atomlist](#anchor677)                   /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Rms_fp](#anchor678)                         /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Rotate_torsion_angle](#anchor679)           /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Rotate_torsion_angle2](#anchor680)          /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Run_gmx_local](#anchor681)                  /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Run_gmx_local](#anchor682)                  /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Run_gmx_slurm_multi](#anchor683)            /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Run_gmx_slurm_multi](#anchor684)            /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Run_gmx_slurm_single](#anchor685)           /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Run_gmx_slurm_single](#anchor686)           /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Run_gnuplot](#anchor687)                    /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Run_gnuplot1](#anchor688)                   /vcp1/people/david/bin/Silico1.15/lib/silico_gnuplot.pm <br>

[Run_gnuplot2](#anchor689)                   /vcp1/people/david/bin/Silico1.15/lib/silico_gnuplot.pm <br>

[Run_gromacs](#anchor690)                    /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Run_gromacs](#anchor691)                    /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Run_grompp_gmx_local](#anchor692)           /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Run_grompp_gmx_local](#anchor693)           /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Run_grompp_gmx_slurm_single](#anchor694)    /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Run_grompp_gmx_slurm_single](#anchor695)    /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Run_grompp_gromacs](#anchor696)             /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Run_grompp_gromacs](#anchor697)             /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Run_mopac](#anchor698)                      /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Run_quick_batch](#anchor699)                /vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm <br>

[Run_ssh](#anchor700)                        /vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm <br>

[Same_substructure](#anchor701)              /vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm <br>

[Sanitise_hydrogens](#anchor702)             /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Sanitise_segid_chain](#anchor703)           /vcp1/people/david/bin/Silico1.15/bin/find_amino_acids <br>

[Scale_vector](#anchor704)                   /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Schrodinger_wait_for_license_availability](#anchor705) /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Schrodinger_wait_for_license_use](#anchor706) /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Score_ring_het](#anchor707)                 /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Sdf_calc_atom_parity_error](#anchor708)     /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Sdf_calc_atom_parity_error](#anchor709)     /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Sdf_check_chiral_parity](#anchor710)        /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Sdf_check_chiral_parity](#anchor711)        /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Sdf_data_to_atom](#anchor712)               /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Sdf_data_to_atom](#anchor713)               /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Sdf_data_to_table_delimited](#anchor714)    /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Sdf_error_mol](#anchor715)                  /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Sdf_error_mol](#anchor716)                  /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Sdf_fastsplit](#anchor717)                  /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Sdf_fastsplit_readmol](#anchor718)          /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Sdf_fastsplit_writemol](#anchor719)         /vcp1/people/david/bin/Silico1.15/lib/silico_split.pm <br>

[Select_pareto_front](#anchor720)            /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Select_residues](#anchor721)                /vcp1/people/david/bin/Silico1.15/bin/find_aggregate <br>

[Select_rotate_bond](#anchor722)             /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Seq_fix_residue_numbers](#anchor723)        /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_label_by_field](#anchor724)             /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_label_by_homology](#anchor725)          /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_label_indels](#anchor726)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_mark_start_end](#anchor727)             /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_merge](#anchor728)                      /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_printout](#anchor729)                   /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_printout_header](#anchor730)            /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_printout_one](#anchor731)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_printout_three](#anchor732)             /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_read_error](#anchor733)                 /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Seq_residue_fields](#anchor734)             /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Seq_string](#anchor735)                     /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_STD](#anchor736)                  /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_STD_nonSTD](#anchor737)           /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_clustal_homology](#anchor738)     /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_consensus](#anchor739)            /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_get_column](#anchor740)           /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_length](#anchor741)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_nonSTD](#anchor742)               /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_printout](#anchor743)             /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_read_mols](#anchor744)            /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Sequences_write_mols](#anchor745)           /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Set_acidbase](#anchor746)                   /vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm <br>

[Set_debg17](#anchor747)                     /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Set_default_oappend](#anchor748)            /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Set_flag](#anchor749)                       /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Set_flag_from_arguments](#anchor750)        /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Set_lflag](#anchor751)                      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Set_sflag](#anchor752)                      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Set_torsion_angle](#anchor753)              /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Setup_flags](#anchor754)                    /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Setup_fp_flags](#anchor755)                 /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Setup_mopac_flags](#anchor756)              /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Setup_quick_flags](#anchor757)              /vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm <br>

[Shared_rings](#anchor758)                   /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Silico_msg](#anchor759)                     /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Silico_putline](#anchor760)                 /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Silico_readline](#anchor761)                /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Silico_setup](#anchor762)                   /vcp1/people/david/bin/Silico1.15/bin/bilayer_builder <br>

[Silico_setup](#anchor763)                   /vcp1/people/david/bin/Silico1.15/bin/checkbox <br>

[Silico_setup](#anchor764)                   /vcp1/people/david/bin/Silico1.15/bin/chop_cell <br>

[Silico_setup](#anchor765)                   /vcp1/people/david/bin/Silico1.15/bin/cluster <br>

[Silico_setup](#anchor766)                   /vcp1/people/david/bin/Silico1.15/bin/dist_filter <br>

[Silico_setup](#anchor767)                   /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Silico_setup](#anchor768)                   /vcp1/people/david/bin/Silico1.15/bin/ens_randomise_order <br>

[Silico_setup](#anchor769)                   /vcp1/people/david/bin/Silico1.15/bin/ens_sort <br>

[Silico_setup](#anchor770)                   /vcp1/people/david/bin/Silico1.15/bin/extract_lig_prot <br>

[Silico_setup](#anchor771)                   /vcp1/people/david/bin/Silico1.15/bin/file_rename <br>

[Silico_setup](#anchor772)                   /vcp1/people/david/bin/Silico1.15/bin/find_aggregate <br>

[Silico_setup](#anchor773)                   /vcp1/people/david/bin/Silico1.15/bin/find_clash <br>

[Silico_setup](#anchor774)                   /vcp1/people/david/bin/Silico1.15/bin/find_close_atoms <br>

[Silico_setup](#anchor775)                   /vcp1/people/david/bin/Silico1.15/bin/find_close_water <br>

[Silico_setup](#anchor776)                   /vcp1/people/david/bin/Silico1.15/bin/find_groups <br>

[Silico_setup](#anchor777)                   /vcp1/people/david/bin/Silico1.15/bin/find_ligand_contacts <br>

[Silico_setup](#anchor778)                   /vcp1/people/david/bin/Silico1.15/bin/find_rings <br>

[Silico_setup](#anchor779)                   /vcp1/people/david/bin/Silico1.15/bin/find_similar <br>

[Silico_setup](#anchor780)                   /vcp1/people/david/bin/Silico1.15/bin/fix_cis_amides <br>

[Silico_setup](#anchor781)                   /vcp1/people/david/bin/Silico1.15/bin/formatdoc <br>

[Silico_setup](#anchor782)                   /vcp1/people/david/bin/Silico1.15/bin/make_box <br>

[Silico_setup](#anchor783)                   /vcp1/people/david/bin/Silico1.15/bin/make_index <br>

[Silico_setup](#anchor784)                   /vcp1/people/david/bin/Silico1.15/bin/measure_dist <br>

[Silico_setup](#anchor785)                   /vcp1/people/david/bin/Silico1.15/bin/micelle <br>

[Silico_setup](#anchor786)                   /vcp1/people/david/bin/Silico1.15/bin/mmod_csearch <br>

[Silico_setup](#anchor787)                   /vcp1/people/david/bin/Silico1.15/bin/mmod_min <br>

[Silico_setup](#anchor788)                   /vcp1/people/david/bin/Silico1.15/bin/mmod_qikprop <br>

[Silico_setup](#anchor789)                   /vcp1/people/david/bin/Silico1.15/bin/mol2seq <br>

[Silico_setup](#anchor790)                   /vcp1/people/david/bin/Silico1.15/bin/mol2split <br>

[Silico_setup](#anchor791)                   /vcp1/people/david/bin/Silico1.15/bin/mol_add_h <br>

[Silico_setup](#anchor792)                   /vcp1/people/david/bin/Silico1.15/bin/mol_add_segid <br>

[Silico_setup](#anchor793)                   /vcp1/people/david/bin/Silico1.15/bin/mol_amides <br>

[Silico_setup](#anchor794)                   /vcp1/people/david/bin/Silico1.15/bin/mol_backbone <br>

[Silico_setup](#anchor795)                   /vcp1/people/david/bin/Silico1.15/bin/mol_boltz_pop <br>

[Silico_setup](#anchor796)                   /vcp1/people/david/bin/Silico1.15/bin/mol_bond <br>

[Silico_setup](#anchor797)                   /vcp1/people/david/bin/Silico1.15/bin/mol_centre <br>

[Silico_setup](#anchor798)                   /vcp1/people/david/bin/Silico1.15/bin/mol_centroid <br>

[Silico_setup](#anchor799)                   /vcp1/people/david/bin/Silico1.15/bin/mol_characterise <br>

[Silico_setup](#anchor800)                   /vcp1/people/david/bin/Silico1.15/bin/mol_charge <br>

[Silico_setup](#anchor801)                   /vcp1/people/david/bin/Silico1.15/bin/mol_check <br>

[Silico_setup](#anchor802)                   /vcp1/people/david/bin/Silico1.15/bin/mol_combine <br>

[Silico_setup](#anchor803)                   /vcp1/people/david/bin/Silico1.15/bin/mol_del_atoms <br>

[Silico_setup](#anchor804)                   /vcp1/people/david/bin/Silico1.15/bin/mol_del_dummy <br>

[Silico_setup](#anchor805)                   /vcp1/people/david/bin/Silico1.15/bin/mol_del_h <br>

[Silico_setup](#anchor806)                   /vcp1/people/david/bin/Silico1.15/bin/mol_del_solvent <br>

[Silico_setup](#anchor807)                   /vcp1/people/david/bin/Silico1.15/bin/mol_divide <br>

[Silico_setup](#anchor808)                   /vcp1/people/david/bin/Silico1.15/bin/mol_ensemble_average <br>

[Silico_setup](#anchor809)                   /vcp1/people/david/bin/Silico1.15/bin/mol_extents <br>

[Silico_setup](#anchor810)                   /vcp1/people/david/bin/Silico1.15/bin/mol_filter <br>

[Silico_setup](#anchor811)                   /vcp1/people/david/bin/Silico1.15/bin/mol_flatten <br>

[Silico_setup](#anchor812)                   /vcp1/people/david/bin/Silico1.15/bin/mol_hydrogen_bonds <br>

[Silico_setup](#anchor813)                   /vcp1/people/david/bin/Silico1.15/bin/mol_label_fg <br>

[Silico_setup](#anchor814)                   /vcp1/people/david/bin/Silico1.15/bin/mol_mw <br>

[Silico_setup](#anchor815)                   /vcp1/people/david/bin/Silico1.15/bin/mol_orient <br>

[Silico_setup](#anchor816)                   /vcp1/people/david/bin/Silico1.15/bin/mol_rama <br>

[Silico_setup](#anchor817)                   /vcp1/people/david/bin/Silico1.15/bin/mol_rename <br>

[Silico_setup](#anchor818)                   /vcp1/people/david/bin/Silico1.15/bin/mol_renumber <br>

[Silico_setup](#anchor819)                   /vcp1/people/david/bin/Silico1.15/bin/mol_rescale_bonds <br>

[Silico_setup](#anchor820)                   /vcp1/people/david/bin/Silico1.15/bin/mol_rot <br>

[Silico_setup](#anchor821)                   /vcp1/people/david/bin/Silico1.15/bin/mol_rot_bond <br>

[Silico_setup](#anchor822)                   /vcp1/people/david/bin/Silico1.15/bin/mol_rotrans <br>

[Silico_setup](#anchor823)                   /vcp1/people/david/bin/Silico1.15/bin/mol_scale <br>

[Silico_setup](#anchor824)                   /vcp1/people/david/bin/Silico1.15/bin/mol_segment <br>

[Silico_setup](#anchor825)                   /vcp1/people/david/bin/Silico1.15/bin/mol_smiles <br>

[Silico_setup](#anchor826)                   /vcp1/people/david/bin/Silico1.15/bin/mol_solvate <br>

[Silico_setup](#anchor827)                   /vcp1/people/david/bin/Silico1.15/bin/mol_sort <br>

[Silico_setup](#anchor828)                   /vcp1/people/david/bin/Silico1.15/bin/mol_split <br>

[Silico_setup](#anchor829)                   /vcp1/people/david/bin/Silico1.15/bin/mol_split_segid <br>

[Silico_setup](#anchor830)                   /vcp1/people/david/bin/Silico1.15/bin/mol_transfer_coords <br>

[Silico_setup](#anchor831)                   /vcp1/people/david/bin/Silico1.15/bin/mol_unsplit <br>

[Silico_setup](#anchor832)                   /vcp1/people/david/bin/Silico1.15/bin/mol_wrap_cell <br>

[Silico_setup](#anchor833)                   /vcp1/people/david/bin/Silico1.15/bin/n_dist <br>

[Silico_setup](#anchor834)                   /vcp1/people/david/bin/Silico1.15/bin/pdb_chain_split <br>

[Silico_setup](#anchor835)                   /vcp1/people/david/bin/Silico1.15/bin/pdb_rename_hydrogens <br>

[Silico_setup](#anchor836)                   /vcp1/people/david/bin/Silico1.15/bin/pdbsplit <br>

[Silico_setup](#anchor837)                   /vcp1/people/david/bin/Silico1.15/bin/plot_dihedrals <br>

[Silico_setup](#anchor838)                   /vcp1/people/david/bin/Silico1.15/bin/radius_of_gyration <br>

[Silico_setup](#anchor839)                   /vcp1/people/david/bin/Silico1.15/bin/random_box <br>

[Silico_setup](#anchor840)                   /vcp1/people/david/bin/Silico1.15/bin/randomise <br>

[Silico_setup](#anchor841)                   /vcp1/people/david/bin/Silico1.15/bin/rdf <br>

[Silico_setup](#anchor842)                   /vcp1/people/david/bin/Silico1.15/bin/renumber_residues <br>

[Silico_setup](#anchor843)                   /vcp1/people/david/bin/Silico1.15/bin/residue_distance <br>

[Silico_setup](#anchor844)                   /vcp1/people/david/bin/Silico1.15/bin/residue_rename <br>

[Silico_setup](#anchor845)                   /vcp1/people/david/bin/Silico1.15/bin/rms <br>

[Silico_setup](#anchor846)                   /vcp1/people/david/bin/Silico1.15/bin/rms_pairs <br>

[Silico_setup](#anchor847)                   /vcp1/people/david/bin/Silico1.15/bin/run_mopac <br>

[Silico_setup](#anchor848)                   /vcp1/people/david/bin/Silico1.15/bin/run_quick <br>

[Silico_setup](#anchor849)                   /vcp1/people/david/bin/Silico1.15/bin/sdf_merge <br>

[Silico_setup](#anchor850)                   /vcp1/people/david/bin/Silico1.15/bin/sdfsplit <br>

[Silico_setup](#anchor851)                   /vcp1/people/david/bin/Silico1.15/bin/smooth <br>

[Silico_setup](#anchor852)                   /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Silico_setup](#anchor853)                   /vcp1/people/david/bin/Silico1.15/bin/tabulate <br>

[Silico_setup](#anchor854)                   /vcp1/people/david/bin/Silico1.15/bin/vesicle_builder <br>

[Silico_setup](#anchor855)                   /vcp1/people/david/bin/Silico1.15/bin/water_to_ion <br>

[Silico_setup](#anchor856)                   /vcp1/people/david/bin/Silico1.15/bin/write_g03zmat <br>

[Silico_setup](#anchor857)                   /vcp1/people/david/bin/Silico1.15/bin/write_gro <br>

[Silico_setup](#anchor858)                   /vcp1/people/david/bin/Silico1.15/bin/write_itp <br>

[Silico_setup](#anchor859)                   /vcp1/people/david/bin/Silico1.15/bin/write_mae <br>

[Silico_setup](#anchor860)                   /vcp1/people/david/bin/Silico1.15/bin/write_merck <br>

[Silico_setup](#anchor861)                   /vcp1/people/david/bin/Silico1.15/bin/write_mmod <br>

[Silico_setup](#anchor862)                   /vcp1/people/david/bin/Silico1.15/bin/write_mol2 <br>

[Silico_setup](#anchor863)                   /vcp1/people/david/bin/Silico1.15/bin/write_mopac <br>

[Silico_setup](#anchor864)                   /vcp1/people/david/bin/Silico1.15/bin/write_pdb <br>

[Silico_setup](#anchor865)                   /vcp1/people/david/bin/Silico1.15/bin/write_sdf <br>

[Silico_setup](#anchor866)                   /vcp1/people/david/bin/Silico1.15/bin/write_seq <br>

[Silico_setup](#anchor867)                   /vcp1/people/david/bin/Silico1.15/bin/write_tab <br>

[Silico_setup](#anchor868)                   /vcp1/people/david/bin/Silico1.15/bin/write_tinker <br>

[Silico_setup](#anchor869)                   /vcp1/people/david/bin/Silico1.15/bin/write_xyz <br>

[Silico_vdw_radii](#anchor870)               /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Simple_atomtype](#anchor871)                /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Sleep_or_stop](#anchor872)                  /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Slurp_file](#anchor873)                     /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Smiles_invariant](#anchor874)               /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Smiles_list_recursive](#anchor875)          /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Smooth_table_delimited](#anchor876)         /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Sort_atoms_label](#anchor877)               /vcp1/people/david/bin/Silico1.15/bin/water_to_ion <br>

[Sort_itp_by_atom_number](#anchor878)        /vcp1/people/david/bin/Silico1.15/bin/write_itp <br>

[Sort_itp_by_atom_number](#anchor879)        /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Sort_itp_by_smiles](#anchor880)             /vcp1/people/david/bin/Silico1.15/bin/write_itp <br>

[Sort_itp_by_smiles](#anchor881)             /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Space_group_pdb_to_silico](#anchor882)      /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Space_group_silico_to_pdb](#anchor883)      /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Splitline](#anchor884)                      /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Standard_bond_lengths](#anchor885)          /vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm <br>

[Standard_deviation_bias_corrected](#anchor886) /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Standard_deviation_bias_uncorrected](#anchor887) /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Standardise_boolean](#anchor888)            /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Standardise_ring](#anchor889)               /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Standardise_ring_by_heteroatoms](#anchor890) /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Stop_file_exists](#anchor891)               /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm <br>

[Stop_file_exists](#anchor892)               /vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm <br>

[Submit_quick_job](#anchor893)               /vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm <br>

[Submol_active_atoms](#anchor894)            /vcp1/people/david/bin/Silico1.15/bin/find_aggregate <br>

[Submol_set_cluster](#anchor895)             /vcp1/people/david/bin/Silico1.15/bin/find_aggregate <br>

[Subname](#anchor896)                        /vcp1/people/david/bin/Silico1.15/lib/silico_control.pm <br>

[Sybyl_sort](#anchor897)                     /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Sybyl_spectrum](#anchor898)                 /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Table_delimited_addcolumn](#anchor899)      /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Table_delimited_getrow](#anchor900)         /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Table_delimited_getval](#anchor901)         /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Table_delimited_inc_row](#anchor902)        /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Tag_markup](#anchor903)                     /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Tan_substructure_frac](#anchor904)          /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Tanimoto](#anchor905)                       /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Tanimoto_freq](#anchor906)                  /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Tanimoto_matrix](#anchor907)                /vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm <br>

[Tautomer_probability_field](#anchor908)     /vcp1/people/david/bin/Silico1.15/bin/dock_sort <br>

[Test_all_shapes](#anchor909)                /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Test_cylinder_arb](#anchor910)              /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Test_energy_molcount](#anchor911)           /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Test_plane](#anchor912)                     /vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm <br>

[Tinker_space_group](#anchor913)             /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Traverse](#anchor914)                       /vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm <br>

[Two_bond_angle](#anchor915)                 /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Two_bond_normal](#anchor916)                /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Unit_normal_vector](#anchor917)             /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Unit_vector](#anchor918)                    /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Valence](#anchor919)                        /vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm <br>

[Variance_bias_corrected](#anchor920)        /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Variance_bias_uncorrected](#anchor921)      /vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm <br>

[Vector](#anchor922)                         /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Vector_add](#anchor923)                     /vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm <br>

[Verify_input_data](#anchor924)              /vcp1/people/david/bin/Silico1.15/lib/silico_data.pm <br>

[Veryquiet](#anchor925)                      /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Warn_atom_type_error](#anchor926)           /vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm <br>

[Wraphtml](#anchor927)                       /vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm <br>

[Write_asl_atoms](#anchor928)                /vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm <br>

[Write_binfile](#anchor929)                  /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Write_clustal](#anchor930)                  /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Write_divcon_cart](#anchor931)              /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Write_fasta](#anchor932)                    /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Write_gaussian_cart](#anchor933)            /vcp1/people/david/bin/Silico1.15/lib/silico_gaussian.pm <br>

[Write_gaussian_zmatrix](#anchor934)         /vcp1/people/david/bin/Silico1.15/lib/silico_gaussian.pm <br>

[Write_gromacs_gro](#anchor935)              /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Write_gromacs_gro_molecule](#anchor936)     /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Write_gromacs_index_file](#anchor937)       /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Write_gromacs_itp](#anchor938)              /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Write_gromacs_itp_mol](#anchor939)          /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Write_gromacs_top](#anchor940)              /vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm <br>

[Write_mae](#anchor941)                      /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_mae_molecule](#anchor942)             /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_maegz](#anchor943)                    /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_merck](#anchor944)                    /vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm <br>

[Write_merck_mol](#anchor945)                /vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm <br>

[Write_mmod](#anchor946)                     /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_mmod_com_mdyn_stage](#anchor947)      /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_mmod_com_mini_stage](#anchor948)      /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_mmod_com_setup](#anchor949)           /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_mmod_com_write_stage](#anchor950)     /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_mmod_sbc](#anchor951)                 /vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm <br>

[Write_mol2](#anchor952)                     /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_atoms](#anchor953)               /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_bonds](#anchor954)               /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_comments](#anchor955)            /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_crystal](#anchor956)             /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_molecule](#anchor957)            /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_sdfdata](#anchor958)             /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_set](#anchor959)                 /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol2_substructures](#anchor960)       /vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm <br>

[Write_mol_any](#anchor961)                  /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Write_mol_any_check](#anchor962)            /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Write_mol_single](#anchor963)               /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Write_mopac_cart](#anchor964)               /vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm <br>

[Write_pdb](#anchor965)                      /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Write_pdb_cryst](#anchor966)                /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Write_pdb_molecule](#anchor967)             /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Write_pdb_sdfdata](#anchor968)              /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Write_pdbx](#anchor969)                     /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Write_pdbx_cryst](#anchor970)               /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Write_pdbx_molecule](#anchor971)            /vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm <br>

[Write_pir](#anchor972)                      /vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm <br>

[Write_pymol_load_file](#anchor973)          /vcp1/people/david/bin/Silico1.15/bin/mol_divide <br>

[Write_quick_in](#anchor974)                 /vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm <br>

[Write_rama_structure](#anchor975)           /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Write_rama_structure_by_mol](#anchor976)    /vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm <br>

[Write_sdf](#anchor977)                      /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Write_sdf](#anchor978)                      /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Write_sdf_atoms_bonds](#anchor979)          /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm <br>

[Write_sdf_atoms_bonds](#anchor980)          /vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 <br>

[Write_seq_any](#anchor981)                  /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Write_seq_any_check](#anchor982)            /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Write_smiles](#anchor983)                   /vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm <br>

[Write_starmaker_comment](#anchor984)        /vcp1/people/david/bin/Silico1.15/bin/starmaker <br>

[Write_table_delimited](#anchor985)          /vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm <br>

[Write_textfile](#anchor986)                 /vcp1/people/david/bin/Silico1.15/lib/silico_io.pm <br>

[Write_tinker_prm](#anchor987)               /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Write_tinker_prm_mol_dihedrals_oplsaa](#anchor988) /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Write_tinker_prm_oplsaa](#anchor989)        /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Write_tinker_xyz](#anchor990)               /vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm <br>

[Write_xyz](#anchor991)                      /vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm <br>

[Write_xyz_molecule](#anchor992)             /vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm <br>


## Subroutine descriptions 


### <a name="anchor0"></a>Aa                             
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Return one letter code of the first amino acid in a sequence record 


Requires:&nbsp;&nbsp; Silico residue record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; one-letter code <br>

### <a name="anchor1"></a>Aa_chirality                   
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find chirality of amino acid 


Requires:&nbsp;&nbsp; atoms N, CA, C, CB <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; D, L or U if chiral volume is close to 0 <br>

### <a name="anchor2"></a>Aa_one_letter                  
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Return the one letter AA code from three letter 


Returns 'X' if unknown 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor3"></a>Aa_three_letter                
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Return the three letter AA code from one letter code 


Returns 'UNK' if unknown 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor4"></a>Acceptable_neutral_valences    
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Determines, for each element, a list of acceptable neutral valences. The ANV 
is like an oxidation state, but it doesn't take into account atom 
electronegativities. 


Requires:&nbsp;&nbsp; nothing (creation of package global variable) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing (creation of package global variable) <br>

### <a name="anchor5"></a>Add_bondorders_to_atom_tree    
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Add bondorders to the branches in an atom tree. 


Requires:&nbsp;&nbsp; molecule, atom tree <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; modified atom tree <br>

### <a name="anchor6"></a>Add_layer                      
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Add a layer to a dendrimer 


 <br>
 <br>

### <a name="anchor7"></a>Add_monomer                    
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Adds a single monomer to one of the available connection points on a 
dendrimer. 


Requires:&nbsp;&nbsp; existing dendrimeric framework (mol1), monomer to add (mol2), join point on dendrimer (j1), join point on monomer (j2), residue number <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; new dendrimer with monomer added <br>

### <a name="anchor8"></a>Add_option                     
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Add a text string to the list of command-line options 


Requires:&nbsp;&nbsp; Option to add (string), list of existing options (string) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Modifies string provided to function <br>

### <a name="anchor9"></a>Add_part_of_layer              
/vcp1/people/david/bin/Silico1.15/bin/starmaker 




 <br>
 <br>

### <a name="anchor10"></a>Align_bonds                    
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Align two bonds 


Overlaps p1 and j2, p2 and j1 (approx) 


Requires:&nbsp;&nbsp; mol1, mol2, a1, a2, a3, a4, bond length <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor11"></a>Amino_acid_and_nucleotide_base_codes 
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Prepare lists of three-letter and one-letter codes for amino acids and 
nucleotide bases. Also prepare hashes mapping the three-letter and one-letter 
codes to each other. 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing (sets package global arrays and hashes) <br>

### <a name="anchor12"></a>Angle_between_vectors          
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Angle between two vectors (radians) 


Requires:&nbsp;&nbsp; vector1, vector2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; angle <br>

### <a name="anchor13"></a>Asl_substring_to_specifiers    
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Parse an atom specifier string and return a hashes containing keys and key 
values 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of hashes <br>

### <a name="anchor14"></a>Atom_add_1h                    
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add one hydrogen to an atom 


This code needs a good clean up! 


This code can add hydrogens with a steric clash. e.g biguanides 


### <a name="anchor15"></a>Atom_add_2h                    
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add two hydrogens to an atom 


If atom has only one connected atom hydrogens are added in the plane generated 
from the connected atom 


### <a name="anchor16"></a>Atom_add_3h                    
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add three hydrogens to an atom 


### <a name="anchor17"></a>Atom_add_h                     
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add specified number of hydrogens to an atom 


Requires:&nbsp;&nbsp; atom, molecule, bondlength, number of hydrogens to add (in addition to any existing atoms) <br>

### <a name="anchor18"></a>Atom_calc_bin                  
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Determine which bin an atom belongs to 


Triclinic version 


Requires:&nbsp;&nbsp; atom, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; bin (array); <br>

### <a name="anchor19"></a>Atom_calc_bin_                 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Determine which bin an atom belongs to 


Requires:&nbsp;&nbsp; atom, binsize, number of bins in each dimension <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; bin (array); <br>

### <a name="anchor20"></a>Atom_check_valence             
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Simple check that atom valences are OK 


Requires:&nbsp;&nbsp; atom, molecule, (optional: flag to denote hydrogens not present) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; -1 if error, otherwise returns valence <br>

### <a name="anchor21"></a>Atom_colour                    
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Colour an atom. 


Compatible with Mol2 files 


Requres:&nbsp;&nbsp;&nbsp;&nbsp; atom, colour <br>

### <a name="anchor22"></a>Atom_cross_prod                
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Calculate the normalised cross product of vectors defined by $atom0 ->atom1 
and atom0 -> atom2; Warning: this contains a horrible fudge if two atoms 
overlap or all three atoms are in a straight line 


Requires:&nbsp;&nbsp; atom0, atom1, atom2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector x, y, z  A1 ---- A0 | | A2 <br>

### <a name="anchor23"></a>Atom_data_to_sdf               
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Copy data to SDF_DATA from atom records 


### <a name="anchor24"></a>Atom_data_to_sdf               
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Copy data to SDF_DATA from atom records 


### <a name="anchor25"></a>Atom_formal_charge             
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate the formal charge on an atom. 


Uses the acceptable oxidation states of that atom, and whether that element is 
more intrinsically likely to be a cation or an anion. Assumes that all 
valences are filled with hydrogens if 'noh' flag is set 


Requires:&nbsp;&nbsp; atom, molecule, optional flag to denote missing hydrogens, flag to print charged atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; formal charge <br>

### <a name="anchor26"></a>Atom_guess_element             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Determine the atom element using available information 


Requires:&nbsp;&nbsp; atom, 'quiet' flag, flag to force preference for 'CHONSP' elements (ie H will be chosen over Hg) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; element and element number. Also sets $atom-&gt;{ELEMENT} and $atom-&gt;{ELEMENT_NUM} <br>

### <a name="anchor27"></a>Atom_guess_element_mol2        
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Guess the element of an atom from a mol2 file. 


Uses the Sybyl atom type to generate the atom element. 


Requires:&nbsp;&nbsp; atom. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Atomic symbol, atomic number. <br>

### <a name="anchor28"></a>Atom_keys                      
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


### <a name="anchor29"></a>Atom_lone_pairs                
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate the number of lone pairs present on an atom. 


Requires:&nbsp;&nbsp; atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of lone pairs <br>

### <a name="anchor30"></a>Atom_norm_vector               
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Return a unit vector atom1 -> atom2 for two atoms 


### <a name="anchor31"></a>Atom_number_sort_function      
/vcp1/people/david/bin/Silico1.15/bin/write_itp 


Sort function for sort_itp_by_atom_number() 


### <a name="anchor32"></a>Atom_number_sort_function      
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Sort function for sort_itp_by_atom_number() 


### <a name="anchor33"></a>Atom_oxidation_state           
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate the oxidation state of an atom, not including a formal-charge 
component. 


Requires:&nbsp;&nbsp; atom, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number <br>

### <a name="anchor34"></a>Atom_path_traverse             
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Recursive subroutine to enumerate LINEAR atom paths starting from a particular 
atom 


Requires:&nbsp;&nbsp; molecule, pointer to array containing starting atom number, path list (pointer to empty array to start) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; path_list to an array of atom paths <br>

### <a name="anchor35"></a>Atom_path_traverse_branched    
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Recursive subroutine to enumerate BRANCHED atom paths starting from a 
particular atom 


Requires:&nbsp;&nbsp; molecule, pointer to array containing starting atom number, path list (pointer to empty array to start) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; path_list to an array of atom paths <br>

### <a name="anchor36"></a>Atom_printout                  
/vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm 


Print out a formated atom record for debugging purposes 


Reauires: atom, filehandle (optional) 


### <a name="anchor37"></a>Atom_score_hybridisation       
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Attempt to determine the hybridisation scheme of an atom. 


Assumption is that all hydrogens are present. 


Requires:&nbsp;&nbsp; atom, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor38"></a>Atom_smiles_label              
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Provide an atom with additional data in the SMILES string (square brackets, 
attached hydrogen atoms, formal charge, etc.) 


Requires:&nbsp;&nbsp; atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor39"></a>Atom_specifier_setup           
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Set up flags for simple atom specifiers 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of atom specifier names, list of specifiers; <br>

### <a name="anchor40"></a>Atom_vector                    
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates a vector from two atoms (a1 - a2) 


Requires:&nbsp;&nbsp; coord1, coord2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector as an array <br>

### <a name="anchor41"></a>Atomic_elements                
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Prepare an array atomic element symbols, with the offset corresponding to the 
atomic number (dummy atom is 0). Also prepare a hash with the element name as 
the key and the element number as the value 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing (sets package global arrays and hashes) <br>

### <a name="anchor42"></a>Atominfo                       
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Convert atom info into a string for printing & catch errors 


Requires:&nbsp;&nbsp; atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor43"></a>Base16                         
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Convert a decimal number to base 36 


Source: https://www.perlmonks.org/?node_id=1197777 


### <a name="anchor44"></a>Base36                         
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Convert a decimal number to base 36 


Source: https://www.perlmonks.org/?node_id=1197777 


### <a name="anchor45"></a>Bin_values                     
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Bin values into 'bins' data structure 


Bin minimum and maximum values should be set in $bins->{$key}{MINVAL} and 
$bins->{$key}{MINVAL} 


Counts of overflows and underflows are kept in $bins->{$key}{OVERFLOW} and 
$bins->{$key}{UNDERLFLOW} 


Non-numeric data is not counted 


Requires: data value, key, pointer to bins data structure 


Returns: nothing 


### <a name="anchor46"></a>Bl                             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Return an expected bond length based on an input string. Single bonds A-B, 
double A=B, triple A%B 


Requires:&nbsp;&nbsp; String <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number, or the undefined value if no bond length <br>

### <a name="anchor47"></a>Bndangle                       
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the angle defined by three atoms 


Requires:&nbsp;&nbsp; atom1, atom2, atom3. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; angle in degrees. <br>

### <a name="anchor48"></a>Bond_create                    
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create a bond between two atoms (takes atom offsets numbered from ZERO as 
arguments). 


Creates a bond in both directions. ie makes connections atom2->atom1 and 
atom1->atom2 


Silico internal bond types: 


1 single, 2 double, 3 triple, 4 aromatic, 


5 single or double, 6 single or aromatic, 7 double or aromatic, 


8 any (undefined), 0 zero order. 


Requires:&nbsp;&nbsp; molecule, atomnum1, atomnum2, bondorder, flag to not increment mol2-&gt;NUMBONDS. <br>

### <a name="anchor49"></a>Bond_create_atom               
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create a bond between two atoms (takes atoms as arguments). 


See bond_create 


### <a name="anchor50"></a>Bond_delete                    
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Delete a bond between two atoms. 


Requires:&nbsp;&nbsp; molecule, atomnum1, atomnum2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor51"></a>Bond_delete_all                
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Delete all bonds 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor52"></a>Bond_delete_atom               
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Delete a bond between two atoms. 


Same as bond_delete but using atoms instead of atom numbers. this version is 
to be preferred 


Requires:&nbsp;&nbsp; molecule, atom1, atom2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor53"></a>Bond_modify_order              
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Modify the order of an existing bond Note: this requires that atom->{NUM} is 
numbered correctly 


Modifies bond in both directions. 


Silico internal bond types: 1 single, 2 double, 3 triple, 4 aromatic, 5 single 
or double, 6 single or aromatic, 7 double or aromatic, 8 any (undefined), 0 
zero order. 


Requires:&nbsp;&nbsp; Molecule, atom 1, atom 2, bondorder (default 1) <br>

### <a name="anchor54"></a>Bond_modify_order_check        
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Modify the order of a bond but check that we are not breaking bonding rules 
before doing it 


Calls bond_modify_order 


Requires:&nbsp;&nbsp; Molecule, atom 1, atom 2, order <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if bond was modified 0 if not <br>

### <a name="anchor55"></a>Bond_vector                    
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates a vector from two atoms. 


Requires:&nbsp;&nbsp; atom1 (origin), atom2 (vector end) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector as an array <br>

### <a name="anchor56"></a>Bonded                         
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Count the number of bonds to an atom and the number of bonded C, H, O, N atoms 


Dummy atoms are not included 


Requires:&nbsp;&nbsp; atom, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; hash containing the element (key) and element count (value) of all connected atoms. The hash also contains the key value NOT_H. The values are zeroed for C, H, N, O, P and NOT_H. Other elements may return 'undef' <br>

### <a name="anchor57"></a>Bondlengths                    
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Routine that returns a hash of maximum reasonable bondlengths for common atom 
pairs 


This routine is used by the various 'connect_atom' routines. For single bonds, 
the bond lengths are maximum allowable lengths and are generally 10% longer 
than standard bond lengths. For double and triple bonds they are 5% larger. 


Note. Some bondlengths are longer than the maximum bondlength in the connect 
routine. 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Hash maximum bondlengths. Keys have the form El_El where 'El' is the atom element <br>

### <a name="anchor58"></a>Bondlengths_2                  
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Routine that returns a hash of squared bondlengths. See bondlengths. 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Hash maximum bondlengths squared. Keys have the form El_El where 'El' is the atom element <br>

### <a name="anchor59"></a>Branchsort                     
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Sort connected atoms by their ring priorities then by their SMILES ranks 


### <a name="anchor60"></a>Calc_area_sphere_with_holes    
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Calculate the area of a sphere with holes 


Does not consider hole overlap 


Requires:&nbsp;&nbsp; Radius, number of holes, hole radius <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Radius <br>

### <a name="anchor61"></a>Calc_binsize                   
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Determine size of bins in cell given an approximate binsize 


Requires:&nbsp;&nbsp; molecule, approx binsize(default 2 ang) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; binsize in each dimension (array), number of bins in each dimension (array) <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; mol-&gt;{BINSIZE} and mol-&gt;{NUMBINS}; <br>

### <a name="anchor62"></a>Calc_box_size                  
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Determine an appropriate periodic cell size for a molecule NOTE: Current 
assumption is that alpha = beta = gamma = 90 deg!!! 


Uses specified values if supplied 


Will then use mol->{CELL} if defined 


Otherwise prompts for values 


$mol->{CELL} is set_box 


Requires: molecule, boxx, boxy, boxz, flag to translate to centre, additional margin around structure, flag to ignore dimensions contained in molecule file <br>
Returns: boxx, boxy, boxz <br>

### <a name="anchor63"></a>Calc_chiral_sdf_stereo         
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Calculate the normaliased cross between vectors defined by $atom0 ->atom1 and 
atom0 -> atom2; Warning this contains a horrible fudge if two atoms overlap or 
all three atoms are in a straight line 


Requires:&nbsp;&nbsp; atom0, atom1, atom2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector x, y, z  A1 ---- A0 | | A2 <br>

### <a name="anchor64"></a>Calc_chiral_sdf_stereo         
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Calculate the normaliased cross between vectors defined by $atom0 ->atom1 and 
atom0 -> atom2; Warning this contains a horrible fudge if two atoms overlap or 
all three atoms are in a straight line 


Requires:&nbsp;&nbsp; atom0, atom1, atom2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector x, y, z  A1 ---- A0 | | A2 <br>

### <a name="anchor65"></a>Calc_chiral_volume             
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Calculated the chiral volume for four atoms 


Uses the equations (A . B x C) where A is the vector atom1->atom2 B is 
atom1->atom3 and C is atom1 -> atom4 


Requires:&nbsp;&nbsp; atom1, atom2, atom3, atom4 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; signed volume of parallelipiped defined by the vectors <br>

### <a name="anchor66"></a>Calc_chiral_volume             
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Calculated the chiral volume for four atoms 


Uses the equations (A . B x C) where A is the vector atom1->atom2 B is 
atom1->atom3 and C is atom1 -> atom4 


Requires:&nbsp;&nbsp; atom1, atom2, atom3, atom4 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; signed volume of parallelipiped defined by the vectors <br>

### <a name="anchor67"></a>Calc_conservation_group_membership 
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Calculate strong and weak conservation groups defined by Clustal 


"These are all the positively scoring groups that occur in the Gonnet Pam250 
matrix. The strong and weak groups are defined as strong score >0.5 and weak 
score =<0.5 respectively." Clustalw documentation 


Requires: pointer to array of amino acids <br>
Returns: *, :, . or '' (Corresponding to identity, strong conservation, weak conservation, no conservation) <br>

### <a name="anchor68"></a>Calc_conservation_group_percentage 
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 




### <a name="anchor69"></a>Calc_distance_function         
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates a repulsive potential energy term between two sets of atoms 


This is based on distance from others in the molecule. This routine makes use 
of the cutoff distance in the global variable $cut and a minimum approach 
distance (clash). A fixed penalty is added for each atom clash. 


Requires:&nbsp;&nbsp; Molecule, atom list 1, atom list2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Distance function, number of clashes <br>

### <a name="anchor70"></a>Calc_human_readable_time       
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Calc_human_readable_time 


Takes a length of time (in seconds) and converts it into ddd:hh:mm:ss.ss 


Requires:&nbsp;&nbsp; Time in seconds <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Time as a text string <br>

### <a name="anchor71"></a>Calc_physiological_protonation 
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Use a simple algorithm to determine which functional groups will be ionised at 
physiological pH. 


Atoms&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; are marked with a {FORMAL_CHARGE} label (These should be cleared before running this routine) <br>
Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor72"></a>Calc_rotation_vector_onto_vector 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate rotation matrix and axis to rotate one vector onto another. 


See: 
http://immersivemath.com/forum/question/rotation-matrix-from-one-vector-to-another/ 


Note: The implementation could be more efficient 


Requires:&nbsp;&nbsp; origin vector, target vector (3 coordinates) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; axis, rotation matrix <br>

### <a name="anchor73"></a>Calc_valence_change            
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Calculate the difference between expected and actual valence 


Takes $atom->{CONNECTED_NON_H} flag into consideration 


Note treatment of P valence is simplistic 


Requires:&nbsp;&nbsp; atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; difference between expected and actual <br>

### <a name="anchor74"></a>Calculate_bin                  
/vcp1/people/david/bin/Silico1.15/bin/rdf 


Calculate the bin to which a given distance belongs 


Requires:&nbsp;&nbsp; distance (scalar) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; bin number (scalar) <br>

### <a name="anchor75"></a>Calculate_one_rdf              
/vcp1/people/david/bin/Silico1.15/bin/rdf 


A subroutine to calculate the radial distribution of one group about a 
reference point or each atom in another group 


Requires:&nbsp;&nbsp; Reference group, Target group, various fully-qualified variables <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; A list of all distances in this RDF <br>

### <a name="anchor76"></a>Calculate_rdfs                 
/vcp1/people/david/bin/Silico1.15/bin/rdf 


A subroutine to calculate all radial distribution functions in an input hash 


Passed&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; variables: Input hashref ($input) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor77"></a>Carbanion_filter_neutraliser   
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Some incorrectly formatted SDF files can contain incorrect carbanions on 
aromatic rings 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if carbanion is detected (except for isocyanides -N+%C-) and all metal containing compounds. The carbanion is also corrected. <br>

### <a name="anchor78"></a>Centroid                       
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate the average atom position of a molecule or subset. 


Requires:&nbsp;&nbsp; Molecule, optional atom list <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array <br>

### <a name="anchor79"></a>Centroid_periodic              
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate the average atom position of a molecule or subset considering 
periodic boundary conditions 


First atom of set is used as the reference atom 


Requires:&nbsp;&nbsp; Molecule, optional atom list, molecule cell <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array <br>

### <a name="anchor80"></a>Centroid_periodic_bv           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate the average atom position of a molecule or subset considering 
periodic boundary conditions 


First atom of set is used as the reference atom 


Requires:&nbsp;&nbsp; Molecule, optional atom list, box vector <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array <br>

### <a name="anchor81"></a>Ch                             
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Check if a hashkey is defined; return 0 if not 


Requires:&nbsp;&nbsp; Hash, key <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Value of key, or 0 if undefined <br>

### <a name="anchor82"></a>Change_bondorder               
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Change bond order between 'atom' and any connected atom of the specified 
element 


If order = 10 then this routine alternates single,double (for carboxylate 
groups) 


Note: calls bond_modify_order and requires that atom->{NUM} is correct 


Requires:&nbsp;&nbsp; atom, desired element of connected atom, molecule, bondorder <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor83"></a>Check_arguments                
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Check the argument list for how many arguments there are. If an error has been 
made in the instruction for number of arguments, the script will die. If the 
tests are not passed, the script will exit after running PRINTDOC. Otherwise, 
nothing is done: the program carries on. 


Requires:&nbsp;&nbsp; string specifying number of arguments; list of arguments <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor84"></a>Check_data_type                
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Checks an input string to see if it is a specified type 


Currently looks for: ALPHA (A-Z characters only), WORD (A-Z plus _), INT or 
INTEGER (integers), DECIMAL (real number in standard or mantissa format), 
FILENAME, BOOLEAN 


Also has numeric checks >, >=, <, <= 


Requires:&nbsp;&nbsp; String of data, string of type information <br>
Optional:&nbsp;&nbsp; Description, Suppression of standard warnings (will still warn if serious errors are encountered) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; -1 if a serious error; 0 if a data type problem; 1 if OK <br>

### <a name="anchor85"></a>Check_grace_output_format      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Checks an output format for compatibility with Grace 


Requires:&nbsp;&nbsp; Output format (text) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Correctly formatted output format, extension <br>

### <a name="anchor86"></a>Check_itp_is_bonded            
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Check that atoms in a GMX_BOND, GMX_ANGLE, or GMX_DIHEDRAL ($angle) are 
actually bonded 


### <a name="anchor87"></a>Check_log                      
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Check that log file is complete 


Requires:&nbsp;&nbsp; output filebase <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef (job is running), 0 (start new job) or 1 (restart job) <br>

### <a name="anchor88"></a>Check_log                      
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Check that log file is complete 


Requires:&nbsp;&nbsp; output filebase <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef (job is running), 0 (start new job) or 1 (restart job) <br>

### <a name="anchor89"></a>Check_number_components        
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Check that number of components is set for each component. If it is not, then 
ask 


### <a name="anchor90"></a>Check_printerror               
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Print an error line for an atom with incorrect valences 


Requires:&nbsp;&nbsp; molecule, atom, valence <br>

### <a name="anchor91"></a>Checkvalence                   
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Check if a proposed valence is less than the maximum valence for an atom 
taking into account atomic charge (where present) or whether an atom is an 
aromatic ring and might be charged 


Requires:&nbsp;&nbsp; atom, proposed valence <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if OK 0 if not <br>

### <a name="anchor92"></a>Chiral_volume                  
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate the chiral volume 


For atoms AB(C)D - B is the central atom 


v1 = BA 


v2 = BC 


v3 = BD 


vol = v1 . (v2 x v3) 


Requires:&nbsp;&nbsp; four atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; scalar <br>

### <a name="anchor93"></a>Chonspx_filter                 
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Find if molecule contains other than CHONSP+Halogen elements 


H* is included to take implicit hydrogens into account 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Element name (true) if non CHONSPX elements are present or 0 otherwise <br>

### <a name="anchor94"></a>Clash_periodic_binned          
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find clashes between two molecules, taking into account periodic boundary 
conditions 


A much faster routine than clash_periodic 


Requires:&nbsp;&nbsp; molecule1, molecule2, maximum number of clashes, vdW scale factor for added atoms <br>

mol2-> CELL, BINSIZE, NUMBINS and BINS must all be defined 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of clashes (up to maximum if this is set), pairs of violating atoms and distances <br>

### <a name="anchor95"></a>Clean_pdb_line                 
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Remove spaces and line name from PDB record 


### <a name="anchor96"></a>Clean_up_split_submols         
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Clean up ensemble that results from splitting a molecule into submolecules 


Requires:Ensemble, original molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ensemble <br>

### <a name="anchor97"></a>Cleanline                      
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Take a line of text and remove leading and trailing spaces 


Requires:&nbsp;&nbsp; Text <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Text <br>

### <a name="anchor98"></a>Cleansdf                       
/vcp1/people/david/bin/Silico1.15/bin/sdf_merge 


Clean up SDF_DATA 


### <a name="anchor99"></a>Clear_smiles_sets              
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Clear all markers on atoms identifying them as belonging to a SMILES set. 


Requires:&nbsp;&nbsp; molecule, list of atoms (default all atoms) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor100"></a>Close_contact                  
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Find if ligand makes a close contact with one or more protein atoms 


Requires:&nbsp;&nbsp; molecule, array of contat atoms. <br>
Returns: true or false <br>
Flags: 'cd' cutoff distance, 'ci' invert logic <br>

### <a name="anchor101"></a>Close_file                     
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Close the current input file opened using open_file 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor102"></a>Close_molfile                  
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Close a molecule file that has been opened to read or write 


Requires:&nbsp;&nbsp; file record <br>

### <a name="anchor103"></a>Combine                        
/vcp1/people/david/bin/Silico1.15/bin/mol_combine 


### <a name="anchor104"></a>Compare_atom_order             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Compare two atoms and see if they are in the correct order for output to mol2, 
merck files etc. 


This routine should be the same as the one used in pdb_molecule_sort_atoms 


### <a name="anchor105"></a>Condense_smiles_ranks          
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Given an input set of SMILES ranks, remove the "air gaps" and non-integral 
values from the ranks so atoms are ranked 1, 2, ... 


Requires:&nbsp;&nbsp; list of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of used ranks <br>

### <a name="anchor106"></a>Connect_atoms_by_distance      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Bond atoms together that are closer than their combined vdW radii. A faster 
routine that uses atom bins to substantially speed things up. All existing 
bonds are deleted before new bonds are calculated. All new bonds are created a 
single bonds. 


Note1: This routine utilises an overall maximum bond length of 2.3 Angstroms. 
I-I bonds fall outside this range 


Note2: Does not bond dummy atoms 


Note3: If a molecule contains 'Alternate atoms' eg from a pdb file - then 
atoms with specifier 'A' are bonded to atoms without an alternate specifier. 
Alternate atoms with specifier 'B' or greater are bonded only to themselves. 


Requires:&nbsp;&nbsp; molecule, flag to skip making bonds in water molecules <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of bonds in molecule <br>

### <a name="anchor107"></a>Connect_atoms_by_distance_old  
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Bond atoms together that are closer than their combined vdW radii. A faster 
routine that uses atom bins to substantially speed things up. All existing 
bonds are deleted before new bonds are calculated. All new bonds are created a 
single bonds. 


Note1: This routine utilises an overall maximum bond length of 2.3 Angstroms. 
I-I bonds fall outside this range 


Note2: Does not bond dummy atoms 


Note3: If a molecule contains 'Alternate atoms' eg from a pdb file - then 
atoms with specifier 'A' are bonded to atoms without an alternate specifier. 
Alternate atoms with specifier 'B' or greater are bonded only to themselves. 


Requires:&nbsp;&nbsp; molecule, flag to skip making bonds in water molecules <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of bonds in molecule <br>

### <a name="anchor108"></a>Connect_atoms_by_distance_simple 
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Bond atoms together that are closer than their combined vdW radii. 


Very simple connect atoms routine for testing 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of bonds created <br>

### <a name="anchor109"></a>Connect_atoms_by_distance_simple_bv_periodic 
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Bond atoms together that are closer than their combined vdW radii. 


Very simple connect atoms routine for testing 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of bonds created <br>

### <a name="anchor110"></a>Connected                      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Return atoms connected to an atom as an array 


Requires:&nbsp;&nbsp; atom, molecule <br>
array&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; of atoms <br>

### <a name="anchor111"></a>Construct_atom_relationships   
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Construct the parent-child relationships between pairs of atoms, relative to 
an arbitrary common ancestor, based on the contents of $atom->{PATH}. 


Requires:&nbsp;&nbsp; molecule, optional flag to terminate at substructure boundaries <br>
 <br>

### <a name="anchor112"></a>Construct_family_tree          
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Construct a family tree from parent-child relationships with an arbitrary 
ultimate ancestor. Can take any type of correctly formatted input. So far, 
atoms and residues use this. 


Requires:&nbsp;&nbsp; array of relationships (each one a hash containing PARENT and CHILD entries) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; branches <br>

### <a name="anchor113"></a>Construct_residue_relationships 
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 




 <br>
 <br>

### <a name="anchor114"></a>Convert_aromatic_bonds         
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Convert non-ring aromatic bonds to double/single 


Note that this conversion is 'undone' by the write_mol2 subroutine 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor115"></a>Convert_aromatic_bonds_kekule  
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Convert rings with aromatic bonds to kekule representation 


Converts systems up to 18 atoms 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor116"></a>Copy_atom                      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Return a new copy of an atom. 


Warning. This copy is only one level deep and does not include hashes. 


Requires:&nbsp;&nbsp; atom. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; new atom. <br>

### <a name="anchor117"></a>Copy_minimal_molecule          
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Make a very minimal copy of a molecule 


Note to save space, records which are not modified by the program are 
duplicates and NOT copies 


### <a name="anchor118"></a>Correlation_coeff              
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Calculate&nbsp;&nbsp; correlation coefficient <br>
Requires:&nbsp;&nbsp; molecule, SDF_data fields <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; correlation coeff or undef if variance is zero <br>

### <a name="anchor119"></a>Corresponding_prime_old        
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Determine the prime number corresponding to an integer. That is, for some 
number N, determine the Nth prime number. 


Requires:&nbsp;&nbsp; integer <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; integer <br>

### <a name="anchor120"></a>Count_formal_charges           
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate the formal charge on a molecule. Also count total numbers of 
positive and negative atoms 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Total charge. Counts of positive and negative atoms (not counting nitros, azides etc with adjacent +ve and -ve charges) <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{FORMAL_CHARGE}, $mol-&gt;{CHARGED_ATOMS} <br>

### <a name="anchor121"></a>Count_polar_nonpolar           
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Count polar (N, O) and nonpolar (all other) heavy atoms 


Requires:&nbsp;&nbsp; molecule <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{SDF_DATA}{POLAR} and $mol-&gt;{SDF_DATA}{NONPOLAR} <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or 0 if non-CHONSP+Halogen atom is encountered. <br>

### <a name="anchor122"></a>Create_angle                   
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create an ANGLE record between three atoms (whether they are actually 
connected or not). This is used, for example, in reading in PSF files. 


Requires:&nbsp;&nbsp; molecule, three atom offsets <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if successful <br>

### <a name="anchor123"></a>Create_molecule                
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create a new molecule. 


Requires:&nbsp;&nbsp; Optional molecule name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor124"></a>Create_planar_improper         
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create a planar improper dihedral, e.g. when reading in a PSF file 


Requires:&nbsp;&nbsp; molecule, list of four atom offsets (NB. not atom numbers) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor125"></a>Create_water_molecule          
/vcp1/people/david/bin/Silico1.15/lib/silico_solvate.pm 


Create a template of a water molecule. 


Requires: Residue name (optional) <br>
Returns: molecule <br>

### <a name="anchor126"></a>Crossprod                      
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates the cross product of two vectors. 


Requires:&nbsp;&nbsp; two vectors <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; cross product (an array {x, y, z}) <br>

### <a name="anchor127"></a>Csv_fix                        
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Fix delimiters in value taken from CSV file 


See http://tools.ietf.org/html/rfc4180 


### <a name="anchor128"></a>Deep_copy                      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Alternative deep copy routine for data structures 


Requires:&nbsp;&nbsp; Data structure <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; copy of data structure <br>

### <a name="anchor129"></a>Deep_copy_orig                 
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Recursive deep copy routine for data structures 


Taken from a tutorial by Randall Schwartz 


Requires:&nbsp;&nbsp; Data structure <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; copy of data structure <br>

### <a name="anchor130"></a>Define_pdb_chain_segid         
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Make sure that pdb CHAIN and SEGID are defined and sane 


Needs a tidy up 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Maximum pdb chain <br>

### <a name="anchor131"></a>Deg_to_rad                     
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Convert degrees to radians 


Requires:&nbsp;&nbsp; scalar <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; scalar <br>

### <a name="anchor132"></a>Delete_atomset                 
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Delete an atomset on the basis of its label 


Requires:&nbsp;&nbsp; Molecule, atomset label <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor133"></a>Delete_flag                    
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Delete a flag from the internal flags list. 


Searches by short name (most cases) or long name (if 'l' is supplied as a 
second argument). 


Requires:&nbsp;&nbsp; name of flag to delete (string), search type (optional string) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor134"></a>Delete_rings                   
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Remove all rings from a molecule 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor135"></a>Delimited_read_error           
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Print an error on failing to read delimited data from file 


Requires:&nbsp;&nbsp; filename, flag to write error message and die <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor136"></a>Distance                       
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the distance between two atoms. 


Requires:&nbsp;&nbsp; atom1, atom2. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; distance. <br>

### <a name="anchor137"></a>Distance_atoms_min             
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the shortest distance between one atom and an array of atoms 


Requires:&nbsp;&nbsp; atom, array of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; shortest distance <br>

### <a name="anchor138"></a>Distance_periodic_points_xyz   
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the X, Y and Z components of distance between two points For a system 
that has cubic, on-axis, periodic boundary conditions 


Note: values are always positive. 


Requires:&nbsp;&nbsp; 6 coordinates, cell <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; periodic distances x, y, z <br>

### <a name="anchor139"></a>Distance_periodic_points_xyz_mat 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the x y and components of the vector between two points For a system 
using Gromacs box vectors 


Requires:&nbsp;&nbsp; point1, point2, ptr to mat matrix, ptr to inverse matrix <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; x, y, z <br>

### <a name="anchor140"></a>Distance_periodic_signed_xyz   
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the X, Y and Z components of distance between two atoms. For a system 
that has cubic, on-axis, periodic boundary conditions 


Note: values are signed. 


Requires:&nbsp;&nbsp; atom1, atom2, $cell-&gt;[cellx, celly, cellz] <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; periodic distances x, y, z <br>

### <a name="anchor141"></a>Distance_periodic_sq           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the square of the distance between two atoms. For a system that has 
cubic, on-axis, periodic boundary conditions 


Requires:&nbsp;&nbsp; atom1, atom2, $cell-&gt;[cellx, celly, cellz] <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; distance_squared. <br>

### <a name="anchor142"></a>Distance_periodic_sq_bv        
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the square of the distance between two atoms. For cuboid systems with 
gromacs box vectors 


Requires:&nbsp;&nbsp; atom1, atom2, $cell-&gt;[cellx, celly, cellz] <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; distance_squared. <br>

### <a name="anchor143"></a>Distance_periodic_xyz          
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the X, Y and Z components of distance between two atoms. For a system 
that has cubic, on-axis, periodic boundary conditions 


Note: values are always positive. 


Requires:&nbsp;&nbsp; atom1, atom2, $cell-&gt;[cellx, celly, cellz] <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; periodic distances x, y, z <br>

### <a name="anchor144"></a>Distance_periodic_xyz_bv       
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the square of the distance between two atoms. For a system using Gromacs 
box vectors 


Requires:&nbsp;&nbsp; atom1, atom2, gromacs_box_vectors <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; x, y, z <br>

### <a name="anchor145"></a>Distance_periodic_xyz_old      
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the X, Y and Z components of distance between two atoms. For a system 
that has cubic, on-axis, periodic boundary conditions 


Requires:&nbsp;&nbsp; atom1, atom2, $cell-&gt;[cellx, celly, cellz] <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; periodic distances x, y, z <br>

### <a name="anchor146"></a>Distance_point_atoms_min       
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the shortest distance between a point and an array of atoms 


Requires:&nbsp;&nbsp; x, y, z, array of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; shortest distance <br>

### <a name="anchor147"></a>Distance_points                
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the distance between two sets of Cartesian points. 


Requires:&nbsp;&nbsp; x1, y1, z1, x2, y2, z2 <br>
 <br>

### <a name="anchor148"></a>Distance_sq                    
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the square of the distance between two atoms. 


Requires:&nbsp;&nbsp; atom1, atom2. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; square of distance. <br>

### <a name="anchor149"></a>Divide                         
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Divide structures into a directory. 


Should be able to replace with a library routine 


### <a name="anchor150"></a>Dotprod                        
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates the dot product of two vectors. 


Requires:&nbsp;&nbsp; two vectors <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; dot product (a scalar) <br>

### <a name="anchor151"></a>Double_and_tie_break           
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Attempt to resolve ties in SMILES ranks using the "double and tie break" 
method. This multiplies all ranks by 2, and then reduces the rank of any one 
of the atoms which share the lowest tied rank by 1. 


This routine has been modified so that the tie is broken in a consistent way 
that depends on the atomic coordinates (if any are present). This is designed 
so that similar conformers will produce a similar smiles order, simplifying 
superimposition 


Requires:&nbsp;&nbsp; list of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of ranks <br>

### <a name="anchor152"></a>Dump_itp_data                  
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Write out debugging file with all itp bonds, pairs, angles etc 


Requires:&nbsp;&nbsp; molecule <br>
Writes&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; out sdf file for each bond, angle etc to a director <br>

### <a name="anchor153"></a>Edit_mdpfile                   
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Edit template mdp file 


Requires:&nbsp;&nbsp; template file, output file <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor154"></a>Edit_mdpfile                   
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Edit template mdp file 


Requires:&nbsp;&nbsp; template file, output file <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor155"></a>Electronegativities            
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Create an array of Pauling electronegativities. Atoms are given in order of 
atomic number (dummy atom is 0). 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing (creation of package global array) <br>

### <a name="anchor156"></a>Element_number2symbol          
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Convert an element number into the corresponding atomic symbol. 


Requires:&nbsp;&nbsp; element number. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; element number or 0 Du. <br>

### <a name="anchor157"></a>Element_symbol2number          
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Convert an element symbol into the corresponding atomic number. 


Requires:&nbsp;&nbsp; element symbol. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; element number or 0 if it fails. <br>

### <a name="anchor158"></a>En                             
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Check that two numbers are equal. Equality means that either: neither number 
is defined, or they are equal to each other (as ascertained using NOT !=). 


Inequality is assumed if one is defined and the other is not, even if the 
defined number is 0. 


Requires:&nbsp;&nbsp; numbers <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if they are, 0 otherwise <br>

### <a name="anchor159"></a>Ens                            
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Convert a list of molecules or ensembles to an ensemble 


Requires:&nbsp;&nbsp; molecule or ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; an ensemble <br>

### <a name="anchor160"></a>Ens_smiles_string              
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Calculate smiles string 


Requires:&nbsp;&nbsp; molecule, flag to use ChemAxon molconvert to generate smiles <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; mol-&gt;{SDF_DATA}{SMILES}, mol-&gt;{SDF_DATA}{SMILES_SOURCE} <br>

### <a name="anchor161"></a>Ens_smiles_string_chemaxon     
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Calculate canonical smiles string using the Chemaxon program 'molconvert' 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; mol-&gt;{SDF_DATA}{SMILES}, mol-&gt;{SDF_DATA}{SMILES_SOURCE} <br>

### <a name="anchor162"></a>Ensemble_calc_fragments        
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate molecular fragments for each molecule in an ensemble 


See mol_fragment for additional information. Sets $mol->{FORMULA}, 
$mol->{NUMHEAVY} and $mol->{SDF_DATA}{NUMHEAVY} 


Requires:&nbsp;&nbsp; Ensemble, typer (optional) <br>

### <a name="anchor163"></a>Ensemble_check                 
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Check all molecules in an ensemble. 


Calls 'mol_check'. 


Requires:&nbsp;&nbsp; ensemble. <br>

### <a name="anchor164"></a>Ensemble_cluster               
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Clusters molecules based on RMS distance from the lowest energy structure. 


Assumes that molecules are sorted by energy. Sets the value 
$mol->{CLUSTER_COUNT} to the number of molecules in that cluster. 


Requires:&nbsp;&nbsp; ensemble, threshold for clustering, flag to keep all molecules, flag to rename molecules with cluster membership <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble containing the first structure from each cluster, or all memebers of each cluster <br>

### <a name="anchor165"></a>Ensemble_connect_atoms         
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Connect all atoms in an ensemble by distance (only within each molecule 
though) 


Calls:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Connect_atoms_by_distance <br>
Requires:&nbsp;&nbsp; Ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor166"></a>Ensemble_consolidate_nocopy    
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Combines all molecules in an ensemble into a single molecule This is the same 
as the version in silico_molecules but without the deep copy - which is much 
faster. 


Requires:&nbsp;&nbsp; Ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; New ensemble with all atoms in the molecule $newens-&gt;[0] <br>

### <a name="anchor167"></a>Ensemble_consolidate_old       
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Combines all molecules in an ensemble into a single molecule 


Works with gromacs .itp files 


Requires:&nbsp;&nbsp; Ensemble, flag to prevent use of deep_copy <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; New ensemble with all atoms in the molecule $newens-&gt;[0] <br>

### <a name="anchor168"></a>Ensemble_consolidate_small_molecules 
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Combine molecules with less than a specified number of atoms 


Requires:&nbsp;&nbsp; ensemble, min molecule size <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble <br>

### <a name="anchor169"></a>Ensemble_consolidate{          
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Combine all molecules in an ensemble into a single molecule 


Rewritten version. DKC Septemer 2025 


Works with gromacs .itp files 


Requires:&nbsp;&nbsp; Ensemble, flag to prevent use of deep_copy <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; New ensemble with all atoms in the molecule $newens-&gt;[0] <br>

### <a name="anchor170"></a>Ensemble_copy                  
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Creates a copy of an ensemble. 


Calls:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Molecule_copy <br>
Requires:&nbsp;&nbsp; Ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Duplicate ensemble as an arrayref <br>

### <a name="anchor171"></a>Ensemble_insight_rename        
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Generate insight-safe names for an ensemble. Calls: insight_rename. 


Requires:&nbsp;&nbsp; ensemble. <br>

### <a name="anchor172"></a>Ensemble_print_connect         
/vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm 


Print out connection table 


Has not been used for some time. Check before using. 


Requires:&nbsp;&nbsp; ensemble, output filename (optional). <br>

### <a name="anchor173"></a>Ensemble_printout              
/vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm 


Print out all the key value pairs for a molecule. 


Only the first few molecules are printed out unless the optional second 
argument is set to 'all'. This is passed on to molecule_printout. 


Requires:&nbsp;&nbsp; ensemble, 'all' (optional), filehandle (optional)  <br>

### <a name="anchor174"></a>Ensemble_renumber              
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Renumber all atoms in an ensemble. 


Calls:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Molecule_renumber2 <br>
Requires:&nbsp;&nbsp; Ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor175"></a>Ensemble_split                 
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Split an ensemble into seperate molecules by connectivity 


Calls molecule_split 


Requires:&nbsp;&nbsp; ensemble, minimum number of atoms (default 1); <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; new ensemble containing unconnected molecules. <br>

### <a name="anchor176"></a>Ensemble_split_keeping_largest_mols 
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Retain only the largest molecule (by connectivity) from each Silico "molecule" 
in an ensemble 


Requires:&nbsp;&nbsp; ensemble, minimum number of atoms (default 1); <br>

### <a name="anchor177"></a>Et                             
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Check that two text strings are equivalent. 


Equivalence means that either: neither string is defined, or they are equal to 
each other (as ascertained using NOT ne). 


Non-equivalence is assumed if one is defined and the other is not, even if the 
defined string is the empty string. 


Requires:&nbsp;&nbsp; strings <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if they are, 0 otherwise <br>

### <a name="anchor178"></a>Euclidian                      
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate the Euclidian coefficient given two hashes of fragments 


Euclidian coeff = 1 - (Num unshared fragments mol1 + Num unshared fragements 
mols)/ (Total fragments mol1, mol2) 


Requires: hash of fragments 1, hash of fragments 2 


Returns: Euclidian coefficient 


### <a name="anchor179"></a>Extract_mol2_sdf_data          
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read the sdf type data from mol2 file comment lines and put into molecule 


Removes this data from the comments array 


Requires:&nbsp;&nbsp; molecule, comments array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; comments lines with sdf data removed or undef on failure <br>

### <a name="anchor180"></a>Fgroup                         
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Make a functional group 


### <a name="anchor181"></a>File_modified_since            
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Check when file last modified (minutes) 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; time since modified in minutes <br>

### <a name="anchor182"></a>File_read_error                
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Print an error on failing to read a file 


Requires:&nbsp;&nbsp; filename, flag to write error message and die <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor183"></a>File_write_error               
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Print an error on failing to write a file 


Requires:&nbsp;&nbsp; filename, flag to write error message and die <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor184"></a>Find_aggregates                
/vcp1/people/david/bin/Silico1.15/bin/find_aggregate 


Subroutine to find aggregates using the faster 'find_close_atoms_binned' 
routine 


Clusters are defined has having carbon atoms within distance 'cutoff' 


Requires;&nbsp;&nbsp; molecule, submolecules, cutoff distance <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; Array of clusters. Each custer is a hash containing: {SUBMOLS} array of submolecules <br>

### <a name="anchor185"></a>Find_atom_paths                
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Find the path connecting each atom to a given root atom 


Requires:&nbsp;&nbsp; molecule, starting atom, optional maximum path length, optional exclusion of H <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing (information added to $atom-&gt;{PATH}) <br>

### <a name="anchor186"></a>Find_atom_rings                
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Find all rings that belong to a particular atom 


Calls traverse. 


Requires:&nbsp;&nbsp; molecule, starting atom, maximum search depth. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; a pointer to array of arrays containing ring atoms or undef if no rings are found. <br>

### <a name="anchor187"></a>Find_atom_rings_not_recursive  
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Subroutine to find rings, based on a non-recursive algorithm 


Algorithm by Balducci and Pearlman, JCICS 1994, 34, 822-831 


Slightly modified to suit Silico conventions 


Requires:&nbsp;&nbsp; Molecule, starting atom, maximum ring size <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; List of rings containing this atom <br>

### <a name="anchor188"></a>Find_available_localhost_cpu   
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Find cpus available on local workstations 


Requires:&nbsp;&nbsp; hostlist (array), sleep time (optional) <br>
Flags:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; max-jobs, number-threads <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; (host, numthreads) <br>

### <a name="anchor189"></a>Find_available_localhost_gpu   
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Find gpus available on local sysgtems 


Requires:&nbsp;&nbsp; hostlist (array), maximum number of jobs to run on each gpu is provided as a flag <br>
Uses:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; List @procnames which is a list of process names to avoid when scheduling <br>
Flags:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; max-jobs <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; host, gpuid, nprocgpu (or nothing) <br>

### <a name="anchor190"></a>Find_bondorder                 
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Returns the bondorder between two atoms 


Requires:&nbsp;&nbsp; atomnum1, atomnum2, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; bondorder or -1 if not connected <br>

### <a name="anchor191"></a>Find_bondorder2                
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Returns the bondorder between two atoms 


Same as find_bondorder but uses atoms as arguments rather than atom numbers. 


Requires:&nbsp;&nbsp; atom1, atom2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; bondorder or -1 if not connected <br>

### <a name="anchor192"></a>Find_bondorders_ring           
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Return array containing bondorders of bonds in ring 


Requires:&nbsp;&nbsp; ring (array of atom numbers), molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of bondorders in ring <br>

### <a name="anchor193"></a>Find_close_atoms_binned        
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find atoms in mol2 which are close to mol1 


Requires:&nbsp;&nbsp; molecule, molecule, cutoff <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of close atoms <br>

### <a name="anchor194"></a>Find_closest_atoms_periodic    
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find closest atoms in two molecules, taking into account periodic boundary 
conditions. 


Simple-minded algorithm for testing stuff 


Requires:&nbsp;&nbsp; molecule1, molecule2, maximum distances, flag to ignore hydrogens <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; sets of atom pairs with distances <br>

### <a name="anchor195"></a>Find_closest_atoms_periodic_binned 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find closest atoms in two molecules, taking into account periodic boundary 
conditions. 


Faster algorithm 


Requires:&nbsp;&nbsp; molecule1, molecule2, maximum distances, flag to ignore hydrogens <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; sets of atom pairs with distances <br>

### <a name="anchor196"></a>Find_decoys_actives            
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Identify decoys and actives 


Requires:&nbsp;&nbsp; molecules <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; total decoys, total actives, total decoys in file, total actives in file <br>

### <a name="anchor197"></a>Find_elements_ring             
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Return array containing elements of atoms in ring and a hash counting the 
number of times each element occurs 




Requires:&nbsp;&nbsp; ring, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to array of elements, hash containing a count of each element <br>

### <a name="anchor198"></a>Find_hbonds                    
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Find hydrogen bonds between ligand and receptor and add to SDF_DATA 


Requires:&nbsp;&nbsp; molecule, protein <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of HBonds found <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; mol -&gt;{SDF_DATA}{Num_HB} mol-&gt;{SDF_DATA}{HBonds} <br>

### <a name="anchor199"></a>Find_host_queue                
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Find information job submission queue 


Returns&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; slurm_cluster or 'local <br>

### <a name="anchor200"></a>Find_hostname                  
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Find information about host from hostname 


Returns&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; host name <br>

### <a name="anchor201"></a>Find_intermolecular_hbonds_h   
/vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm 


Find hydrogen bonds between two molecules using hydrogen atom information 


mol_label_donors_acceptors must be run before commencing this routine 


Requires:&nbsp;&nbsp; Two molecules <br>
Optional:&nbsp;&nbsp; D-A distance, H-A distance <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;D-H-A angle, H-A-R angle  <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; List of hydrogen bonds (D-H-A) <br>

### <a name="anchor202"></a>Find_intermolecular_hbonds_h_simple 
/vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm 


Find hydrogen bonds between two molecules using information about the position 
of the hydrogen atom 


Uses simple cutoff between donor hydrogen and acceptor 


You must call mol_find_donors_acceptors on each molecule first 


### <a name="anchor203"></a>Find_intermolecular_hbonds_noh 
/vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm 


Find hydrogen bonds between two molecules _without_ using information about 
the position of the hydrogen atom 


Uses simple cutoff between donor and acceptor heteroatoms 


You must call mol_find_donors_acceptors on each molecule first 


### <a name="anchor204"></a>Find_intramolecular_hbonds_h   
/vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm 


Find hydrogen bonds between atoms within the one molecule using hydrogen 
positional information 


mol_find_donors_acceptors must be run prior to commencing this routine 


Requires:&nbsp;&nbsp; molecule <br>
Optional:&nbsp;&nbsp; D-A cutoff, H-A cutoff <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;D-H-A angle cutoff, H-A-R angle cutoff  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;flag to include waters  <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of hydrogen bonds <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $atom-&gt;{HBONDS} - a list of hydrogen bonds belonging to each atom <br>

### <a name="anchor205"></a>Find_nearby_charges_recursive  
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Recursive subroutine to find atoms with formal charges 


### <a name="anchor206"></a>Find_nearest_atoms             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Find the nearest atoms to a nominated atom and return list of atoms 


Requires:&nbsp;&nbsp; atom, molecule, cutoff (default 5 angstroms) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to an array of atoms <br>

### <a name="anchor207"></a>Find_ngpu                      
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Find number of gpus on host 


Requires:&nbsp;&nbsp; hostname <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Count of GPUs <br>

### <a name="anchor208"></a>Find_nproc                     
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Find number of cpus on host using ssh 


Requires:&nbsp;&nbsp; hostname <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Count of CPUs <br>

### <a name="anchor209"></a>Find_polar_atoms               
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Mark atoms as polar or nonpolar 


Polar atoms are defined as being non C or H atoms or C or H atoms that are 
bonded to an element other than C or H. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; <br>

### <a name="anchor210"></a>Find_schrodinger_exe           
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Find the Schrodinger executableis 


Reqires:&nbsp;&nbsp;&nbsp;&nbsp; Executable name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Path to Macromodel executable or undef <br>

### <a name="anchor211"></a>Find_substructure              
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Find the substructure around an atom. 


Assumes that a change in residue name or ID terminates the substructure. 


Assumes that bond orders are correct and that aromatic bonds and rings have 
been correctly identified. 


Requires:&nbsp;&nbsp; molecule, starting atom, maximum search depth <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; munted SMILES string <br>

### <a name="anchor212"></a>Findclash                      
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Find the distance to the closest atom from a point x, y, z 


Requires:&nbsp;&nbsp; molecule, x, y, z <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; distance <br>

### <a name="anchor213"></a>Fix_cell_values                
/vcp1/people/david/bin/Silico1.15/bin/find_aggregateSet cell values from 
command line only if not defined in input molecule file 


### <a name="anchor214"></a>Fix_dative_bonds               
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


### <a name="anchor215"></a>Fix_double_bonds_carbonyl      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Assign carbonyl bond orders 


Uses molecule geometry to generate bond orders 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor216"></a>Fix_double_bonds_carboxyl      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Assign ester, carboxylic acid, carboxylate, amide and guanidinium bond orders 


Uses molecule geometry to assign bond orders 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor217"></a>Fix_double_bonds_general       
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Fix general planar double bond orders 


Uses molecule geometry to generate assignment 


Contains a distance check 


Requires:&nbsp;&nbsp; molecule, ccflag (only generate double C=C double bonds) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor218"></a>Fix_hydrogens                  
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Clean up hydrogens added to molecules by sybyl 


Removes hydrogens attached to nitro groups, on oxygen bonded to sulfur or 
phosphorous. Also removes comments 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor219"></a>Fix_pdb_chains                 
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Renumber pdb chains based on changes in CHAIN, SEGID or a backwards step in 
SUBID 


If 'start' and the 'force flag' are not specified this routine will only 
renumber chains if required 


If 'start' is set and the 'force flag' is not set then the routine will 
renumber if 'start' does not match the starting letter and if it is necessary 


If 'force flag' is set then the chains will be renumbered from 'A' by default. 


Note: This subroutine deletes the 'TERCOUNT' field can potentially cause 
problems if the pdb file has been modified 


Requires:&nbsp;&nbsp; molecule, starting chain letter (optional), flag to force renumbering <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; last chain letter of current structure, flag to indicate that chains were renumbered <br>

### <a name="anchor220"></a>Flag_is_boolean                
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Return true if a flag is a simple boolean switch 


Searches by short name (in most cases) or long name (if 'l' is supplied as a 
second argument). 


Requires:&nbsp;&nbsp; flag name (string), search type (optional string) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor221"></a>Fnfp                           
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Format Number For Printing. 


Take&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; a numeric value, and print it. If the numeric value is not defined, return a string of the user's choice (default: 'not defined'). <br>

This can handle NaN strings, but unlike the subroutine ftfp, it does not put 
the strings in quotes, so whitespace is not obvious. 


Requires:&nbsp;&nbsp; string, optional 'not defined' string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor222"></a>Format_doc_html                
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; Silico markup formatted as text <br>

### <a name="anchor223"></a>Format_doc_html_header_footer  
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Format&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; a tagged document as html and add header and footer files <br>

### <a name="anchor224"></a>Format_doc_markdown            
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Format a tagged string into markdown 


Requires:&nbsp;&nbsp; string, linelength <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Silico markup formatted as markdown <br>

### <a name="anchor225"></a>Format_doc_txt                 
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Format a tagged string into ascii text 


### <a name="anchor226"></a>Format_oname                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Generate a numbered filename (various possible styles) 


Styles: 


molname:&nbsp;&nbsp;&nbsp;&nbsp; Molecule Name <br>
molname i: Renames molecule name using 'insight safe' name <br>
named:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &lt;base&gt;_&lt;name&gt;.&lt;ext&gt; <br>
numbered:&nbsp;&nbsp; &lt;base&gt;_&lt;num&gt;.&lt;ext&gt; <br>
fnumbered: &lt;base&gt;_000&lt;num&gt;.&lt;ext&gt; (%05d) <br>
fnumbered N: &lt;base&gt;_000&lt;num&gt;.&lt;ext&gt; (%0Nd) - ie number formated to N places <br>
Requires:&nbsp;&nbsp; style, molecule (requred for style 'name'), basename (required for numbered styles), number (required for numbered styles) or name (required for name style), extension (optional), compression (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; filename (string) <br>

### <a name="anchor227"></a>Format_string_html             
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Create html from a tagged string 


Tags 


&lt;LINK...&gt; <br>
&lt;ANCH...&gt; <br>
&lt;FILE&gt; <br>
&lt;HEADn&gt; <br>
&lt;STD&gt; <br>
&lt;STD DESC&gt; <br>
&lt;HANG&gt; <br>
&lt;HANG SHORT&gt; <br>
&lt;INDENT&gt; <br>
&lt;LIST&gt; <br>

### <a name="anchor228"></a>Formatdata                     
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Format SDF_DATA for output to PDB file 


Long data is split up into lines. All data is surrounded by single quotes. A 
backslash at the end of the line indicates that the line is to be continued 


Requires:&nbsp;&nbsp; string, length of key <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array <br>

### <a name="anchor229"></a>Formatpara_txt                 
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Format a paragraph so that it fits onto a line with optional indent or hanging 
indent 


### <a name="anchor230"></a>Formula_string                 
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Convert molecular formula to a string 


Requires:&nbsp;&nbsp; formula hash <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor231"></a>Fr_printout                    
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Print out a silico file record for debugging purposes 


Requires:&nbsp;&nbsp; file record <br>

### <a name="anchor232"></a>Frag_aray_to_hashes            
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Generate md5 hashes from fragments 


Requres:&nbsp;&nbsp;&nbsp;&nbsp; array of fragments <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of md5 hashes <br>

### <a name="anchor233"></a>Fragment_printout              
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Print out list of fragments 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor234"></a>Fragmentlist_printout          
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Print out list of fragments 


Requires:&nbsp;&nbsp; pointer to fragments <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor235"></a>Ftfp                           
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Format Text For Printing. 


Take&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; a text value, enclose it in double quotes, and print it. If the text value is not defined, return a string of the user's choice (default: 'not defined'). <br>

Note that the 'not defined' string will not be printed in quotes. This allows 
the user to distinguish between a variable containing the text "not defined" 
and an undefined variable. 


This can handle numbers, but unlike the subroutine fnfp, it puts the data in 
quotes, which can appear ugly for numbers. 


Requires:&nbsp;&nbsp; string, optional 'not defined' string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor236"></a>Fundamental_constants          
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Set various mathematical and physical fundamental constants as read-only 
values (so they can't be reassigned by a wayward use of "=" instead of, for 
example, "=="). 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing (sets package global arrays and hashes) <br>

### <a name="anchor237"></a>General_distance_sq            
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the square of the distance between two atoms. Version that uses periodic 
informatin if available 


Requires:&nbsp;&nbsp; atom1, atom2, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; distance_squared. <br>

### <a name="anchor238"></a>General_distance_xyz           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the x, y, z components of vector between two atoms Version that uses 
periodic informatin if available 


Requires:&nbsp;&nbsp; atom1, atom2, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; x, y, z <br>

### <a name="anchor239"></a>Get_aa_frequency               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Calculate the frequency of each AA type from an array of AAs 


AAs are sorted by frequency 


Requires:&nbsp;&nbsp; pointer to array of AAs <br>
Retruns:&nbsp;&nbsp;&nbsp;&nbsp; Pointer to array of arrays. $array-&gt;[$i][0] = aa, $array-&gt;[$i][1] = count <br>

### <a name="anchor240"></a>Get_allflags                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Return a list of all command line flags 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; flags in array <br>

### <a name="anchor241"></a>Get_arguments2                 
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Improved routine to get arguments from command line. 


UNIX: 


Accepts wildcards '*', '?' and [...]. Skips any arguments starting with '-' 
which are used as flags. These can be found using 'get_flags' 


UNIX file read supports directories containing very large numbers of files 
where the standard UNIX Glob fails, if wildcard characters quoted (eg '*') on 
the command line. 


If a -help argument is found then the silico documentation is printed and the 
progarm exits 


Requires:&nbsp;&nbsp; arguments in @ARGV. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; argument list as an array. <br>

### <a name="anchor242"></a>Get_atomarray_using_specifier  
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Parse an asl string and return an array indicating selected atoms and key 
values 


Allowed specifier keys: ANUM: ANAME: CHARGE: ELEMENT: NAME: RESNAME: SEGID: 
SUBID: TYPE: X: Y: Z: 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of atom numbers (indexed from zero) <br>

### <a name="anchor243"></a>Get_atomnums_spechash          
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Use a Silico atom specifier string to select sets of atoms from a molecule 


Requires:&nbsp;&nbsp; molecule, specifier array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to array of atom numbers (indexed from zero) <br>

### <a name="anchor244"></a>Get_atomnums_using_specifier   
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Parse an asl string and return a list of atom numnbers 


Requires:&nbsp;&nbsp; molecule, string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to array of atoms <br>

### <a name="anchor245"></a>Get_atoms_using_specifier      
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Parse an asl string and return a list of atoms 


Requires:&nbsp;&nbsp; molecule, string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to array of atoms <br>

### <a name="anchor246"></a>Get_default_flags_standard     
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Complete flag setup by setting silico-wide default flags 


### <a name="anchor247"></a>Get_file_compression           
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get file compression method (.Z, .bz2 or .gz) 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor248"></a>Get_file_dir                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get file directory 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor249"></a>Get_filebase                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get file basename 


Removes only the final extension unless it is Z. .gz or .bz then the last two 
are removed 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor250"></a>Get_filebase_short             
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get file basename without any leading directories 


Removes the final extension unless it is .gz then the last two are removed 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor251"></a>Get_fileext                    
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get file extension name 


Note. Strips off any .gz, .bz2 or .Z extension before returning file extension 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor252"></a>Get_fitness_strings            
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 




### <a name="anchor253"></a>Get_flag                       
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Return the value held by a particular flag. 


Searches by short name (in most cases) or long name (if 'l' is supplied as a 
second argument). 


Requires:&nbsp;&nbsp; flag name (string), search type (optional string) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; flag value (or pointer to array of values if duplicate definition is allowed); <br>

### <a name="anchor254"></a>Get_gmx_cmd                    
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Formulate Gromacs execution command with flags 


Requires:&nbsp;&nbsp; initial gromacs exec command, number of threads, gpuid, output filebase <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; command string <br>

### <a name="anchor255"></a>Get_hosttype                   
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Find the host type 


Requires: nothing <br>
Returns: sgi, linux, linuxem64, Darwin (OS X) or undef <br>

### <a name="anchor256"></a>Get_ion_data                   
/vcp1/people/david/bin/Silico1.15/bin/water_to_ion 


Supply ion data to water_to_ion for a range of metallic and other simple ionic 
species. 


Requires:&nbsp;&nbsp; Ion name (element, charge optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Element, element number, MMFF charge code, residue name <br>

### <a name="anchor257"></a>Get_join_list                  
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Gets the join list in a single molecule 


Requires:&nbsp;&nbsp; molecule, atom name to look for as a join point <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Array containing those atoms which are join points <br>

### <a name="anchor258"></a>Get_key                        
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Find if molecule already exists using molecule name or smiles string 


### <a name="anchor259"></a>Get_lflag                      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get flag by long name 


Requires:&nbsp;&nbsp; flag name <br>

Calls get_flag 


### <a name="anchor260"></a>Get_membrane_cylinder_atoms    
/vcp1/people/david/bin/Silico1.15/bin/make_index 


Select membrane cylinder atoms 


Requires:&nbsp;&nbsp; molecule, xy-distance, membrane atoms, soluta atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; membrane atoms <br>

### <a name="anchor261"></a>Get_metal_oxidation_state      
/vcp1/people/david/bin/Silico1.15/bin/water_to_ion 


Choose an oxidation state from a list of possibles 


Requires:&nbsp;&nbsp; list of numbers <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; oxidation state chosen <br>

### <a name="anchor262"></a>Get_ofilebase                  
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get output filebase. Takes command line flags into account 


Filebase is obtained from 


1.&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; -O flag on command line <br>
2.&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Name depending on SOURCE_FILE_NAME <br>
3.&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 'outfile' <br>
Input:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ensemble (optional), obase (optional), append_string (optional); <br>
Output:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; filebase <br>

### <a name="anchor263"></a>Get_ofilename                  
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get the output filebase and format of an ensemble checking flags and input 
format 


'Base name' and 'output format' will be overridden by command line flags if 
they are supplied. The 'default format' will also be overridden by command 
line options. 


See also: get_ofilename2 


Requires:&nbsp;&nbsp; (ensemble, base name, format, append string, default_format) all optional splitflag to provide separate filename and extension, flag force use of supplied names ignoring command line flags <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; filename if 'splitflag' not set or filebase & format if it is set. <br>

### <a name="anchor264"></a>Get_ofilename2                 
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Split version of get_ofilename 


See get_ofilename 


### <a name="anchor265"></a>Get_oformat                    
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get an appropriate output file format. 


Selection priority: 


format&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ($_[1]) <br>
-o&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &lt;ext&gt; flag <br>
-O&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &lt;file.ext&gt; flag <br>
format&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; of source file <br>
provided&nbsp;&nbsp;&nbsp;&nbsp; default format ($_[2]) <br>
mol2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <br>



Input:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ensemble or molecule (optional), default format (optional) <br>
Output:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; format string <br>

### <a name="anchor266"></a>Get_polar_nonpolar_centres     
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Get molecule polar and nonpolar centres of mass 


Requres:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; polar and nonpolar xyz coordinates <br>

Polar atoms are defined as N,O atoms. Everything else is nonpolar. Hydrogens 
are ignored 


### <a name="anchor267"></a>Get_sflag                      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Get flag by short name 


Requires:&nbsp;&nbsp; flag name <br>

Calls get_flag 


### <a name="anchor268"></a>Get_std_output_options         
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Read a standard set of output options from the command line 


Deprecated subroutine; 


Requires:&nbsp;&nbsp; a list of options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; option string <br>

### <a name="anchor269"></a>Get_tempname                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Make a temporary filename base by concatenating program_name and a random 
number Uses $Silico::temp_dir (from flag --tempdir) or /usr/tmp (or /tmp) if 
this is not set 


Requires:&nbsp;&nbsp; temp directory <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; filename base <br>

### <a name="anchor270"></a>Getcol                         
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Find a single column number that corresponds to a particular key 


Case insensitive match 


Requires:&nbsp;&nbsp; data structure, list of keys <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of column numbers <br>

### <a name="anchor271"></a>Getcols                        
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Find column numbers that correspond to a particular header 


Case insensitive match 


Requires:&nbsp;&nbsp; data structure, list of keys <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of column numbers <br>

### <a name="anchor272"></a>Getcols_all                    
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Return column list of all keys sorted according to the order in 
@{$data->{HEADERS}} 


Requires:&nbsp;&nbsp; data structure <br>
Requires:&nbsp;&nbsp; pointer to array of column names <br>

### <a name="anchor273"></a>Getcols_not                    
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Find column numbers except specified headers 


Inverse of getcol. Case insensitive match 


Requires:&nbsp;&nbsp; data structure, list of keys <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of column numbers <br>

### <a name="anchor274"></a>Getheaders_all                 
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Get all headers 


Requires:&nbsp;&nbsp; tabledata structure <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Pointer to arrray of headers <br>

### <a name="anchor275"></a>Getlines                       
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Return array of all lines in text data structure 


Requires:&nbsp;&nbsp; data structure <br>

### <a name="anchor276"></a>Getrows_all                    
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Get all rows as array of arrays. Values are in header order 


Requires:&nbsp;&nbsp; tabledata structure <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of arrays <br>

### <a name="anchor277"></a>Gmx_make_heavy_h               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Increase the mass of hydrogen atoms by a scale factor (default 4) and reduce 
the mass of the attached heavy atoms 


Requires:&nbsp;&nbsp; molecule, scale factor <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor278"></a>Gmx_merge_coords_into_top      
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Merge coordinates into ITP file 


Requires:&nbsp;&nbsp; Topology file, coordinate file <br>

### <a name="anchor279"></a>Gmx_replace_peptide_dihedrals  
/vcp1/people/david/bin/Silico1.15/bin/write_itp 


Replace the dihedral parameters of peptide backbone atoms 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of dihedrals replaced <br>

### <a name="anchor280"></a>Gmx_replace_peptide_dihedrals  
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Replace the dihedral parameters of peptide backbone atoms 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of dihedrals replaced <br>

### <a name="anchor281"></a>Gnuplot_plot_bins              
/vcp1/people/david/bin/Silico1.15/lib/silico_gnuplot.pm 


Use gnuplot to plot histograms from bin datastructures 


Calls write_binfile 


Requires: output filename base,bins structure 


Returns: one or undef if file open failed. 


### <a name="anchor282"></a>Grom_putline                   
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Return a gromacs line 


Requires:&nbsp;&nbsp; line, file_record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor283"></a>Grom_readline                  
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read a gromacs .itp topology file line 


Filters out blank lines and comments 


Requires:&nbsp;&nbsp; file_record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; line <br>

### <a name="anchor284"></a>Gromacs_check_itp              
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Check a gromacs included topology data and make sure that it is consistent 
with the bonded molecular structure 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if write failed, otherwise 1 <br>

### <a name="anchor285"></a>Gromacs_regenerate_pairs       
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Regenerate 1-4 atom pairs list 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor286"></a>Heading                        
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Format a heading 


Requires:&nbsp;&nbsp; String, optional underline character (defaults to -) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Formatted string <br>

### <a name="anchor287"></a>Host_is_slurm_cluster          
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Is host a slurm cluster by checking if squeue command exists 


Returns&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor288"></a>Identify_heterocycle           
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Identify a heterocycle 


This routine can name a number of common heterocycles. 


Requires that find_rings has been run and that bond orders are present and 
that aromatic rings are represented using aromatic bonds (ie 
make_aromatic_bonds has been run) 


Requires;&nbsp;&nbsp; ring, molecule <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; heterocycle label or HETEROCYCLE &lt;standardised_string&gt; if unknown; <br>

### <a name="anchor289"></a>Insight_rename                 
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Generate insight-safe names 


Insight has a number of reqirements for molecule names. For example they can 
not contain spaces, punctuation, start with an underscore or a digit and are 
of limited length. This routine generates unique, Insight-safe names by using 
the hash %insight_namecount to store previously used names. 


Requires:&nbsp;&nbsp; name, pointer to a hash to store names in. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; modified and unique name. <br>

### <a name="anchor290"></a>Is_aa_backbone                 
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Return true if atom is in backbone residue (C O CA N) 


Requires:&nbsp;&nbsp; atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor291"></a>Is_gap                         
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Determine if a residue is a gap 


Gaps can be marked with -,+ or maybe . 


Requires:&nbsp;&nbsp; Silico residue record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or 0 <br>

### <a name="anchor292"></a>Is_hbond_mcdonald_thornton     
/vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm 


Identify a hydrogen bond (D-H:::A) according to the definition used by 
McDonald and Thornton 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;J. Mol. Biol. 1994, 238, 777-793  <br>
Requires:&nbsp;&nbsp; donor molecule, donor atom, acceptor molecule, acceptor atom (donor cutoff distance, H cutoff distance, dha angle max, HAR angle max - defaults 3.9, 2.5, 90, 90) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of the hbonds. Each record contains (dmol, datom, hydrogen, amol, aatom) <br>

### <a name="anchor293"></a>Is_integer                     
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Return 1 if a number is an integer 


Requires:&nbsp;&nbsp; Number, optional accuracy (defaults to 6 decimal places) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or 0 <br>

### <a name="anchor294"></a>Is_ring_aromatic_by_bondorder  
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Find aromatic rings ie those with all aromatic bonds or 4n+2 pi electrons 


Subroutine assumes that it is operating on a planar ring 


Requires:&nbsp;&nbsp; molecule , ring <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if aromatic, 0 otherwise <br>

### <a name="anchor295"></a>Is_ring_aromatic_by_connectivity 
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Find 4n+2 member aromatic rings (ie those with all aromatic bonds) by 
connectivity where bondorders and hydrogens are present or missing 


Subroutine assumes that it is operating on a planar ring 


Requires:&nbsp;&nbsp; Molecule, ring, flag to indicate hydrogens are present <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if aromatic, 0 otherwise <br>

### <a name="anchor296"></a>Is_ring_planar                 
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Robust ring planarity and aromaticity 


This routine must be able to cope with: molecules with no hydrogens and/or no 
bondorders, molecules with very poor geometry (eg 2D molecules) and molecules 
with distorted geometries (such as from high temperature MD simulations). 


This routine is expected to fail if the molecule has distorted geometry, bad 
bondorders and missing atoms 


Requires:&nbsp;&nbsp; molecule, ring (array of atoms) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; flag for planarity, flag for aromaticity <br>

### <a name="anchor297"></a>Is_ring_planar_by_dihedral     
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Finds planar rings by checking dihedral angles 


Tests dihedrals in rings 


Attempts to determine planarity (as distinct from aromaticity) 


Returns 0 if any single in-ring dihedral is larger than the hardcoded 
tolerance: 18 for 5-membered rings 26 for all other rings 


Requires:&nbsp;&nbsp; molecule (pointer to hash), $atoms (pointer to array of atom numbers) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 2 if atoms are few enough in number that they must lie in a plane (&lt;= 3) <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1 if there are enough atoms that they could be non-planar, but are planar (within tolerance)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0 otherwise (i.e., not planar within tolerance)  <br>

### <a name="anchor298"></a>Itp_delete_angle_parameters_atom 
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Delete Gromacs GMX_ANGLES where the specified atom is a CENTRAL atom 


Requires&nbsp;&nbsp;&nbsp;&nbsp; molecule, list of atom NUMBERS (not indexes) <br>

### <a name="anchor299"></a>Itp_delete_parameters_atom     
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Delete Gromacs GMX_BONDS GMX_PAIRS GMX_ANGLES GMX_DIHEDRALS or GMX_EXCLUSIONS 
where the specified atom is involved 


Requires&nbsp;&nbsp;&nbsp;&nbsp; molecule, thing, list of atom NUMBERs (not indexes) <br>

### <a name="anchor300"></a>Itp_make_angle                 
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make a Gromacs itp angle 


Note: Uses atom numbers (not atom indexes). 


Requires&nbsp;&nbsp;&nbsp;&nbsp; molecule, atom Number 1, atom Number 2, atom Number 3, angle type, format, etc  <br>

### <a name="anchor301"></a>Itp_make_bond                  
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make a Gromacs itp bond 


Note: Uses atom NUMBERS (not atom indexes). 


Requires&nbsp;&nbsp;&nbsp;&nbsp; molecule, atom Number 1, atom Number 2, type, etc  <br>

### <a name="anchor302"></a>Itp_make_dihedral              
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make a Gromacs itp dihedral (mol->GMX_DIHEDRALS) 


Note: Uses atom numbers (not atom indexes). 


Requires&nbsp;&nbsp;&nbsp;&nbsp; molecule, atom Number 1, atom Number 2, atom Number 3, atom Number 4, dihedral type (1=proper, 2=improper), format, etc <br>

Note. We do not read individual values for default values or where codes are 
used to define parameters 


### <a name="anchor303"></a>Itp_make_dihedral_restraint    
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make a Gromacs itp dihedral restraint (mol->GMX_DIHEDRAL_RESTRAINTS) 


Note: Uses atom numbers (not atom indexes). 


Requires&nbsp;&nbsp;&nbsp;&nbsp; molecule, atom Number 1, atom Number 2, atom Number 3, atom Number 4, function type ( usually 1) <br>

### <a name="anchor304"></a>Itp_make_pair                  
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make a Gromacs itp pair 


Note: Uses atom numbers (not atom indexes). 


Requires&nbsp;&nbsp;&nbsp;&nbsp; molecule, atom Number 1, atom Number 2, type  <br>

### <a name="anchor305"></a>Itp_make_settle                
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make a Gromacs settle 


Not implemented 


### <a name="anchor306"></a>Itp_restrain_dihedrals         
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make a set of dihedral restraints 


Requires:&nbsp;&nbsp; molecule, dihedral list, force constant, angle delta <br>

### <a name="anchor307"></a>Label_aa_backbone              
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label alpha amino acid backbone atoms 


Add the labels AA_N, AA_CA, AA_C to the N, CA anc C atoms of amino acids. 
Terminal amino acids are also labelled AA_CTER or AA_NTER. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; numbers main chain-AAs, C-terminal AAs, N-terminal AAs <br>

### <a name="anchor308"></a>Label_amides                   
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label amides AMIDE_C, AMIDE_O, AMIDE_N 


Bondorders must be correct 


Note: This algorithm handles carbamates OK, but ureas somewhat imprecisely. 


Requires:&nbsp;&nbsp; molecule, list of atoms to check (uses entire molecule if this is not defined) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor309"></a>Label_amino_acids              
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Create residues and identify them based on a SMILES residue database 


Uses residues as the basis on which to match an atom string. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor310"></a>Label_dendrimer_residues       
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Label residues in a molecule according to the Silico dendrimer scheme. Core 
COR1, First generation GAA, Second GAB, ... A rename of all atoms to COR1 is 
enforced if the flag -rc is used, or if atoms are found that are not COR or 
G**. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Highest residue ID found <br>

### <a name="anchor311"></a>Label_fg_alcohol               
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label alcohols, diols and triols 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor312"></a>Label_fg_aromatic              
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Create AROMATIC functional groups with rings up to 6 atoms 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor313"></a>Label_fg_benzhydrylamine       
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label benzhydrylamines 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor314"></a>Label_fg_carbanion             
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label carbanions 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor315"></a>Label_fg_carbon                
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label carbon 'functional groups' 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor316"></a>Label_fg_carbonyl              
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label carbonyl, thioketone and formyl groups 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor317"></a>Label_fg_halogen               
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label halogen atoms 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor318"></a>Label_fg_nitrogen              
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label nitrogen containing functional groups 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor319"></a>Label_fg_oxygen                
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label oxygen containing functional groups excluding CARBONYL_O, CARBOXYLIC_O 
and CARBOXYLATE_O 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor320"></a>Label_fg_phosphorous           
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label phosphorous containing functional groups 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor321"></a>Label_fg_planar_carbon         
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label functional groups containing planar carbon atoms such as carboxylate / 
carboxylic acids and guanidine / guanidinium groups 


Note. Amides are labelled by label_fg_nitrogen 


Carboxylate carbon if bonded to two oxygen atoms that have only one bonded 
atom each. Carbonyl oxygen Ester if connected to double bonded O and single 
bonded O with two carbon attachments if double bond to oxygen 


### <a name="anchor322"></a>Label_fg_sulphur               
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label sulphur containing functional groups 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor323"></a>Label_fg_tfm                   
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label trifluoromethyl groups 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor324"></a>Label_solvent                  
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label solvent molecule atoms $atom->{FG}{SOL}; 


Requires:&nbsp;&nbsp; molecule, maximum number of atoms in solvent <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of water molecules <br>

### <a name="anchor325"></a>Label_waters                   
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label water molecule atoms $atom->{FG}{W}; 


Requires:&nbsp;&nbsp; molecule, flag to force reading of all atoms in mol_has_h <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{NUMWATERS}. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of water molecules <br>

### <a name="anchor326"></a>Lcss                           
/vcp1/people/david/bin/Silico1.15/bin/sdf_merge 


Longest common substring. 


Taken from here https://www.perlmonks.org/?node_id=249239 


### <a name="anchor327"></a>Lipophilicity_dipole           
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Calculate the polar vector of a molecule along the Y axis 


Requires: molecule <br>

### <a name="anchor328"></a>Mae_get_tok                    
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Read mae file, process tokens and return one at a time 


Requires:&nbsp;&nbsp; filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; token (string) <br>

### <a name="anchor329"></a>Mae_ignore_block               
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Read a maestro data block but ignore the contents 


Requires:&nbsp;&nbsp; filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor330"></a>Mae_parse_atoms                
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Read a maestro atom block 


Can handle full atom records (f_m_ct) and partial atom records (p_m_ct) 


Requires:&nbsp;&nbsp; filehandle, molecule, number of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor331"></a>Mae_parse_block_header         
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Parse the header information in a maestro data block 


Requires:&nbsp;&nbsp; filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of headers, array of header types (r: real, i: integer, etc), array of header source programs <br>

### <a name="anchor332"></a>Mae_parse_bonds                
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Read a maestro bond data block 


Requires:&nbsp;&nbsp; filehandle, molecule, number of bonds <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor333"></a>Mae_parse_ct_data              
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Parse a maestro connection table data block (a molecule) 


Requires:&nbsp;&nbsp; filehandle, molecule (can be empty); <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor334"></a>Mae_split_data                 
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Split data into tokens 


Requires:&nbsp;&nbsp; lines from input file (string) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array <br>

### <a name="anchor335"></a>Magnitude                      
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates the magnitude of a vector. 


Requires:&nbsp;&nbsp; vector as an array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; scalar <br>

### <a name="anchor336"></a>Make_alternate_aa_forms        
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Generate alternate amino acid protonation states from parent structure 


Requires:&nbsp;&nbsp; Molecule, hash <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor337"></a>Make_amides_trans              
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Sets the C-N-C(O)-C bond in all secondary amides to 180 degrees 


Requires:&nbsp;&nbsp; molecule, list of atoms to check (uses entire molecule if this is not defined) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor338"></a>Make_aromatic_bonds            
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Make aromatic bonds (bondorder 4) in all 6-membered aromatic rings 


Rings must contain only C and N 


Requires:&nbsp;&nbsp; molecule, kekule flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor339"></a>Make_atom_names_unique         
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Make all atom names in a molecule unique 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor340"></a>Make_atomset                   
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Make an ATOMSET record from atoms carrying a particular flag 


Flag must be defined and be set to true (ie not zero or undefined). Atomsets 
are written out in mol2 format as static sets 


Requires:&nbsp;&nbsp; molecule, set name, flag1, flag2, ... <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; count of atoms having that flag <br>

### <a name="anchor341"></a>Make_atomset_colour            
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Make an ATOMSET record from atoms carrying an atom->{COLOUR} flag. This 
produces atomsets compatible with Sybyl 


Atomsets have names of the type 'ATOM$RED'. Sybyl colours are WHITE RED 
MAGENTA YELLOW ORANGE BLUE VIOLET GREEN BLUEGREEN REDORANGE CYAN PURPLE (In no 
particular order). 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; count of atoms having that flag <br>

### <a name="anchor342"></a>Make_atomset_fg                
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Make an ATOMSET record from atoms carrying a particular functional group flag 


Flag must be defined and be set to true (ie not zero or undefined). The set 
name '*' will make an atomset of all functional groups Atomsets are written 
out in mol2 format as static sets 


Requires:&nbsp;&nbsp; molecule, set name, flag1, flag2, ... <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; count of atoms having that flag <br>

### <a name="anchor343"></a>Make_atomset_formal_charge     
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Make an ATOMSET record from atoms carrying a formal charge 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor344"></a>Make_bins                      
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Create a bin data structure 


Bin data structure contains the following fields: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BINSIZE - bin size  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;RANGEMIN - minimium bin value  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;RANGEMAX - maximum bin value  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;UNDERFLOW - count of values less than rangemin  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;OVERFLOW - count of values greater than rangemin  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;COUNTS[i] - the number of values found for each key value for bin i  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NUMBINS - number of bins  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOT_NUMERIC - count of non-numeric values found  <br>

Requires: Binsize, range minimium, range maximum 


Returns: pointer to bin datastructure 


### <a name="anchor345"></a>Make_dendrimer                 
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Base dendrimer construction routine 


Requires:&nbsp;&nbsp; ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble containing a single molecule <br>

### <a name="anchor346"></a>Make_flag                      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Create a flag based on input options 


Use:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; make_flag (short_name, long_name, "description", expects_value, default_value, prompt, verify_string, duplicate defnition allowed) <br>

'short_name' is a short format switch starting with a single '-' 


'long_name' is a long format switch starting with '--' 


'description' is a text description of the flag 


'expects_value' 0 - flag is a switch, 1 - expects scalar value, 2 - expects 
array (values separated by spaces). 


'default_value' default value for flag (can be undef) 


'option_to_prompt' (1 or 0) when set to 1 silico will prompt for data if it 
undefined 


'verify_string' is used to check that the input data is of the correct type: 
integer, real number, string, etc. See 'verify_input_data' 


'duplicate_definititions_allowed'. This will cause make_flag to return an 
array of values for this flag This is required when we wish to set default 
values for a Silico standard flag 


Requires:&nbsp;&nbsp; Short name, Long name, Description, Expects value?, Default value, option to Prompt, verify string, duplicate definition allowed flag <br>

Note expects_value = 2 and duplicate_definitions_allowed returns an array 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; value the flag is to take. <br>

### <a name="anchor347"></a>Make_gromacs_cell_from_pdb_cell_and_angles 
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Convert PDB cell information into gromacs box vectors 


Requires:&nbsp;&nbsp; vector - a, b, c, alpha, beta, gamma <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; gromacs vector (6 numbers) <br>

### <a name="anchor348"></a>Make_gromacs_itp               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Make basic gromacs itp data where it does not exist Note: Blocks [ 
position_restraints ], [ molecules ] and [ system ] are ignored 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor349"></a>Make_hist                      
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Make a histogram of data and write to a a file 


Requires:&nbsp;&nbsp; data, key to graph, filebase, binsize, binmin, binmax <br>
Writes:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; delimited text file <br>

### <a name="anchor350"></a>Make_molecule_copies           
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Generate multiple copies of molecules to be packed 


Requires:&nbsp;&nbsp; ensemble of molecules to be packed, number to pack, starting chain, starting segid, number of bonds to randomise <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecules <br>

### <a name="anchor351"></a>Make_smiles_string_from_list   
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Generate, from a list of atoms and tokens, a SMILES string. 


Requires:&nbsp;&nbsp; List <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor352"></a>Make_subset                    
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Make a SUBSET (substructure set) record from atoms carrying a particular flag 


Flag must be defined and be set to true (ie not zero or undefined). Atomsets 
are written out in mol2 format as static sets 


Requires:&nbsp;&nbsp; molecule, set name, flag1, flag2, ... <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; count of substructures having that flag <br>

### <a name="anchor353"></a>Make_types_summary             
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Make a silico_msg type summary string 


Requires:&nbsp;&nbsp; Types hash <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor354"></a>Make_vectors_connected         
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Make two unit vectors which are normal to and in the plane of atom -> con1 -> 
con2 


### <a name="anchor355"></a>Makeboxmolecule                
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Make a box molecule using (on-axis cuboid) unit cell size 


Requires:&nbsp;&nbsp; molecule cell, flag to centre box on 0,0,0 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; box molecule <br>

### <a name="anchor356"></a>Makedir                        
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


### <a name="anchor357"></a>Match_amino_acid               
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Match a set of atoms against an amino acid. 


Requires:&nbsp;&nbsp; molecule, list of atoms, hash of amino acids <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; <br>

### <a name="anchor358"></a>Match_amino_acid2              
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Match a set of atoms against an amino acid. Revised version 


Requires:&nbsp;&nbsp; molecule, list of atoms, hash of amino acids <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; <br>

### <a name="anchor359"></a>Maximum                        
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Determines the maximum value within a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Scalar <br>

### <a name="anchor360"></a>Maximum_minus_mean             
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Determine the difference between the mean of a set of data and the maximum of 
the same set. 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Scalar <br>

### <a name="anchor361"></a>Maximum_valences               
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Returns a hash containing, for each element, the maximum usual neutral valence 
for that element. 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; hash <br>

### <a name="anchor362"></a>Mdrun_log_is_complete          
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Check if gmx mdrun log file is complete 


Requires:&nbsp;&nbsp; logfile <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if complete or 0 <br>

### <a name="anchor363"></a>Mdrun_log_is_complete          
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Check if gmx mdrun log file is complete 


Requires:&nbsp;&nbsp; logfile <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if complete or 0 <br>

### <a name="anchor364"></a>Mean                           
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Calculates the mean of a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; mean (scalar quantity) <br>

### <a name="anchor365"></a>Mean_minus_minimum             
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Determine the difference between the mean of a set of data and the minimum of 
the same set. 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Scalar <br>

### <a name="anchor366"></a>Median                         
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Calculates the median of a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; median (scalar quantity) <br>

### <a name="anchor367"></a>Median_absolute_deviation      
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Calculates the median and median absolute deviation 


See: en.wikipedia.org/wiki/Median_absolute_deviation 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; median (scalar quantity), median absolute deveiation (scalar) <br>

### <a name="anchor368"></a>Memuse                         
/vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm 


Print out memory usage found using the SZ record of the 'ps' command 


Requires:&nbsp;&nbsp; string to identify calling location. <br>

### <a name="anchor369"></a>Merck_calculate_formal_charge  
/vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm 


Calculate the merck charge code for .mrk files 


The valid charge codes are (From the charmm documentation): 


Code&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Charge Code  <br>
0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Neutral <br>
1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; +1 <br>
2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; -1 <br>
3&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Radical <br>
4&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; +2 <br>
5&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; -2 <br>
6&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; +3 <br>
7&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; -3 <br>
8&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; +4 <br>
9&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; -4   <br>
Requires:&nbsp;&nbsp; atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; charge code <br>

### <a name="anchor370"></a>Merck_charge_code_to_formal    
/vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm 


Convert a Merck charge code to a formal charge 


### <a name="anchor371"></a>Merck_formal_to_charge_code    
/vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm 


Convert a formal charge to a Merck charge code for .mrk files 


### <a name="anchor372"></a>Merge_filename                 
/vcp1/people/david/bin/Silico1.15/bin/sdf_merge 


Merge filenames removing largest common substring and then adding this to the 
end 


### <a name="anchor373"></a>Merge_itp_coord                
/vcp1/people/david/bin/Silico1.15/bin/write_itp 


Merge coordinates into ITP molecule 


Requires:&nbsp;&nbsp; ITP molecule, coordinate molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor374"></a>Merge_itp_coord                
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Merge coordinates into ITP molecule 


Requires:&nbsp;&nbsp; ITP molecule, coordinate molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor375"></a>Minimise_dendrimer             
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Minimise a dendrimer 


Requires:&nbsp;&nbsp; Molecule, layer number <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; In-situ modification of molecule <br>

### <a name="anchor376"></a>Minimum                        
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Determines the minimum value within a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Scalar <br>

### <a name="anchor377"></a>Mmod_anneal                    
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write Macromodel command file for simulated annealing run 


The first output structure produces is the initial, minimised structures one 
structure is then written after each annealing cycle 


Requires:&nbsp;&nbsp; molecule, output filebase, minimum temp, maximum temp, total cycle time (ps), number of annealing cycles, forcefield, solvent, molecular dynamics timestep (fs), number of structures to be saved from the cooling stage of the annealing run <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if run failed <br>

### <a name="anchor378"></a>Mmod_calc_energy               
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Use Batchmin to to a 'single point' energy calculation 


The output .mmo file is read and molecular energy vaules are stored as 
SDF_DATA fields 


Requires:&nbsp;&nbsp; molecule, output filebase, number of steps, solvent <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if run failed <br>

### <a name="anchor379"></a>Mmod_csearch                   
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write Macromodel command file for a conformational search run 


Requires:&nbsp;&nbsp; molecule or ensemble (all molecules must be identical). Optional or via flags: solvent, search_steps, filebase(otherwise temp filenames are used) Other arguments are passed via flags <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble unless nowait flag is specified (then 1) <br>

### <a name="anchor380"></a>Mmod_csearch_write_com         
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write Macromodel command file for a conformational search run 


Requires:&nbsp;&nbsp; molecule or ensemble (all molecules must be identical). Optional or via flags: solvent, search_steps, filebase(otherwise temp filenames are used) Other arguments are passed via flags <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if run failed <br>

### <a name="anchor381"></a>Mmod_logfile_normal_termination 
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Find if Batchmin logfile terminated properly 


### <a name="anchor382"></a>Mmod_logfile_num_conf          
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Count number of Conf lines in conformational search logfile 


### <a name="anchor383"></a>Mmod_make_complex              
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Create a receptor/ligand complex for use with Macromodel 


Can be used to specify fixed and frozen atoms in the receptor 


Defines atom->{MMOD}{FXAT} flag for receptor atoms and atom->{MMOD}{SUBS} flag 
for ligand. Also sets flags mol->{MMOD}{FXAT} and mol->{MMOD}{SUBS} 


DEBG 17 needs to be set so that all intramolecular interactions are not 
discarded by Macromodel 


Requires:&nbsp;&nbsp; Protein (receptor), ligand, forceconst1, forceconst2, forceconst3 freeze distahce, constraint scheme, flag to constrain hydrogens freeze atoms more than this distance from the ligand <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Molecule or undef if things go wrong <br>

### <a name="anchor384"></a>Mmod_min                       
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Minimise a molecule using macromodel 


If filebase is not specified, temporary files are written to 
$Silico::temp_dir, unless $Silico::debug is set, in which case they are 
written to the current working directory. 


Requires:&nbsp;&nbsp; Molecule, filebase, max number of steps, final gradient forcefield, solvent, minimsation method <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if things go wrong <br>

### <a name="anchor385"></a>Mmod_min_h                     
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Minimise hydrogens using macromodel. All other atoms are constrained 


Requires:&nbsp;&nbsp; Molecule, filebase, max number of steps, final gradient forcefield, solvent, minimsation method <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if things go wrong <br>

### <a name="anchor386"></a>Mmod_mult_min                  
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Use Batchmin to do a multiconformer minimisation 


Requires:&nbsp;&nbsp; molecule, output filebase, number of steps, solvent <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if run failed <br>

### <a name="anchor387"></a>Mmod_qikprop                   
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Run qikprop 


Requires:&nbsp;&nbsp; filebase of macromodel com file, flag not to remove run files <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef if things go wrong <br>

### <a name="anchor388"></a>Mmod_run                       
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Run macromodel 


Will wait for output and read it in. Setting the nowait flag will cause it to 
return as soon as the job has started, without returning output 


Requires:&nbsp;&nbsp; filebase of macromodel com file, flag not to wait for job to complete, flag not to remove run files, <br>
number&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; of used licenses (schrodinger_wait_for_license_availability has already been run); <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or 1 if nowait flag is set. <br>

### <a name="anchor389"></a>Mol2_atom_record               
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in a @<TRIPOS>ATOM record. 


Called from read_mol2. Checks to see if more than half the atoms have zero Z 
coordinates, and sets mol->{HAS_BAD_GEOM} if this is the case. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if successful, undef otherwise <br>

### <a name="anchor390"></a>Mol2_bond_record               
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in @<TRIPOS>BOND record. 


Called from read_mol2. 


Requires:&nbsp;&nbsp; molecule <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef on failure <br>

### <a name="anchor391"></a>Mol2_centroid_record           
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read a tripos centroid feature record 


The centroid is stored in $mol->{FEATURES} 


Note: This code assumes that the CENTROID record is located after the SET 
record. This is normally true but not strictly required. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor392"></a>Mol2_crystal_record            
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in a @<TRIPOS>CRYSIN record 


Called from read_mol2(). 


Unit cell data is placed in hash keys within $mol. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor393"></a>Mol2_fastsplit                 
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Split a multimolecule mol2 file into individual molecules without parsing the 
file 


Requires:&nbsp;&nbsp; infile, output directory (optional), output style (see subroutine format_oname) max number of structures per output file <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of files created <br>

### <a name="anchor394"></a>Mol2_molecule_record           
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in a @<TRIPOS>MOLECULE record. 


Called from read_mol2. 


Requires:&nbsp;&nbsp; molecule <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef on failure <br>

### <a name="anchor395"></a>Mol2_process_comments          
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Extract data from mol2 comments strings such as Dock energies and add it to a 
molecule 


Called from read_mol2. 


Requires:&nbsp;&nbsp; molecule, comments <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor396"></a>Mol2_set_formal_charges        
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Assign atomic formal charges based on FORMAL_CHARGE sets 


FORMAL_CHARGE atomsets are then deleted 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor397"></a>Mol2_set_record                
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in a @<TRIPOS>SET record. 


Called from read_mol2. Only reads static sets of atoms. Ignores dynamic sets 
or sets of bonds or substructures. 


Atoms are placed in a record: mol->{ATOMSET}{<setname>}[<atoms>] and also 
$mol->{ATOMSET_BY_NUMBER}[$i] 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor398"></a>Mol2_substructure_record       
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in a @<TRIPOS>SUBSTRUCTURE record. 


Called from read_mol2. Creates the following arrays: mol->{SUB_SUBNAME}, an 
array of substructure names; mol->{SUB_ROOTATOM}, an array of root atoms; 
mol->{SUB_CHAIN} (optional) an array of chains 


Requires:&nbsp;&nbsp; molecule <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef on failure <br>

### <a name="anchor399"></a>Mol_add_atom                   
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create a new atom and add it to the specified molecule. 


Requires:&nbsp;&nbsp; molecule, atom name, atom element, x, y, z, subname, subid. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; the new atom <br>

### <a name="anchor400"></a>Mol_add_hydrogens              
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add hydrogens to a molecule 


By default the routine tries to approximate the protonation state at 
physiological pH. The charge state is generated by mol_type_sybyl 


Carboxylic acids are deprotonated. 


Non-aromatic amines are protonated. 


Aromatic and heterocylic amines are not protonated 


This can be turned of using the fill_valence flag. 


The Sybyl atom typing routines are used to generate atom types 


Requires:&nbsp;&nbsp; Molecule, flag to add all hydrogens (routine otherwise approximates protonation state at physiological pH), flag to add polar (1) or nonpolar (carbon) hydrogens (2), flag to delete hydrogens only from atoms where hydrogens need to be added. Otherwise all hydrogens are deleted first and then appropriate hydrogens are added back. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of hydrogens added, number of hydrogens deleted, total change <br>

### <a name="anchor401"></a>Mol_add_hydrogens_nocoords     
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add hydrogens to a molecule that lacks coordinates (e.g. SMILES) 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of hydrogens added, number of hydrogens deleted (always zero), total change <br>

### <a name="anchor402"></a>Mol_add_lone_pairs             
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add lone pairs to N, O or S atoms in a molecule 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of lone pairs <br>

### <a name="anchor403"></a>Mol_aromatic_smiles            
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Of a list of atoms, mark each one as aromatic if it is. 


Requires:&nbsp;&nbsp; molecule, optional list <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor404"></a>Mol_bonded_interactions        
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Creates tables containing bonds, angles, dihedrals and improper dihedrals in a 
molecule 


Calls:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Mol_find_all_bonds Mol_find_all_angles Mol_find_all_dihedrals Mol_find_planar_impropers Mol_find_double_bond_impropers <br>
Requires:&nbsp;&nbsp; Molecule, flag to ignore waters, flag to skip hydrogen dihedrals, flag to find only unique dihedrals <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor405"></a>Mol_bv_mat                     
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Return gromacs box vectors as a transformation matricies in Ang 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; mol-&gt;{BV_MAT}, mol-&gt;{BV_MATINV} <br>

### <a name="anchor406"></a>Mol_calc_charge                
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Calculates the total of the partial atomic charges for a molecule and checks 
that it is integral 


Also can calculate the charge on each residue and on each segment 


Will return a warning if there are missing charges or if the total charge is 
not integral 


Requires:&nbsp;&nbsp; Molecule, flag to calculate residue totals, flag to calculate segment totals <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ERR if there are atoms with no set charge Total charge otherwise <br>

### <a name="anchor407"></a>Mol_calc_hcount                
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Calculate number of hydrogens on each heavy atom (but don't add them). 


Requires:&nbsp;&nbsp; Molecules, atomlist, fill_valence flag (default on) <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; atom-&gt;{HCOUNT}, $atom-&gt;{HMISSING} <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor408"></a>Mol_check                      
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Check a molecule for self-consistency 


This routine checks for problems with the internal representation of the 
molecule. 


Checks for undefined fields - ELEMENT, ELEMENT_NUM, NUM, NAME, that all atoms 
connected to another atom actually exist and that atoms are not connected to 
themselves. 


Atoms with problems are marked with the flag $atom->{ERROR} 


Requires:&nbsp;&nbsp; molecule. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; minor error count or -1 if there is a major problem <br>

### <a name="anchor409"></a>Mol_check_all                  
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Call all the mol_check routines 


Requires:&nbsp;&nbsp; molecule. <br>

Returns; 0 if everything is OK. Otherwise returns the minor error count and 
returns -1 if there is a major problem 


### <a name="anchor410"></a>Mol_check_amides               
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Subroutine that checks for non-trans amides 


Checks two dihedrals: C-C-N-C and O-C-N-H, as each of these should be 180 
degrees 


Requires:&nbsp;&nbsp; molecule, name of output file <br>

Returns: Number of atoms in distorted amide groups 


### <a name="anchor411"></a>Mol_check_aromatic_bonds       
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Subroutine to look for atoms, not in aromatic rings, which have aromatic bonds 


This benefits from mol_check_distorted_rings being run first though that is 
not strictly necessary 


Requires:&nbsp;&nbsp; molecule <br>

Returns: Number of atoms with out-of-place aromatic bonds 


### <a name="anchor412"></a>Mol_check_atom_overlap         
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Pariwise check that no atoms in molecule overlap 


Checks that atoms are not closer than 90% of the min VDW distance. 


Requires:&nbsp;&nbsp; molecule, flag to supress warning messages and atom labelling <br>

Returns: Overlap count 


### <a name="anchor413"></a>Mol_check_bondlengths          
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Check molecule bondlengths 


Sets $atom->{BONDLENGTH} on short and long bonds (generally those more than 
10% longer than standard bond lengths) 


Requires:&nbsp;&nbsp; molecule. <br>

Returns; 1 if OK or -1 if there is a major problem 


### <a name="anchor414"></a>Mol_check_bondorders           
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Simple check that bondorders are OK 


An atom->{ERROR} flag is set on each offending atom 


Requires:&nbsp;&nbsp; molecule, <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 0 if OK, otherwise number of unusual bonds found <br>

### <a name="anchor415"></a>Mol_check_cell                 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Check if periodic cell exists and create a functional one if it does not. 


Requires:&nbsp;&nbsp; molecule, flag to print cell <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; $cell <br>

### <a name="anchor416"></a>Mol_check_distorted_rings      
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Look for distorted aromatic rings 


Requires:&nbsp;&nbsp; molecule <br>

Returns: Number of bent aromatic rings 


### <a name="anchor417"></a>Mol_check_elements             
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Check to make sure that all atom elements are sensible 


Requires:&nbsp;&nbsp; molecule <br>

An atom->{ERROR} flag is set on each offending atom 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of errors <br>

### <a name="anchor418"></a>Mol_check_unit_cell            
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Make sure a molecule's unit cell is present 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if OK, 0 otherwise <br>

### <a name="anchor419"></a>Mol_check_valences             
/vcp1/people/david/bin/Silico1.15/lib/silico_check.pm 


Simple check that atom valences are OK 


Calls atom_check_valence 


Requires:&nbsp;&nbsp; molecule, flag to mark that molecule is missing hydrogen atoms, flag to convert non-ring aromatic bonds to single/double bonds <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Negative of number of atoms with abberant valences <br>

An atom->{VALENCE} flag is set on each offending atom 


### <a name="anchor420"></a>Mol_copy_rec                   
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Recursively split atoms into a new ensemble 


Called from mol_split 


Requires: molecule, atomnumber, ensemble, submol 


Returns: nothing 


### <a name="anchor421"></a>Mol_copy_unit_cell             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Copies as much as is defined of a unit cell record from one Silico molecule to 
another. 


Requires:&nbsp;&nbsp; two molecules <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor422"></a>Mol_count_heavy_atoms          
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate the number of heavy (non-hydrogen) atoms in a molecule. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; whole number <br>

### <a name="anchor423"></a>Mol_count_residue_contacts     
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Count the number of non-neigbour atom-atom contacts for each residue of a 
protein 


Requires:&nbsp;&nbsp; Molecule <br>

### <a name="anchor424"></a>Mol_count_rot_bonds            
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Count rotatable bonds 


Rotatable bonds are defined as being single, connected to more than one atom 
(excluding hydrogen), not amide bonds and not in a ring with less than 10 
atoms. 


This is close to the definition used in Veber et al. J. Med. Chem. 2002, 45, 
2615 


Requires:&nbsp;&nbsp; molecule <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{SDF_DATA}{ROT_BONDS}; <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; num rotatable bonds <br>

### <a name="anchor425"></a>Mol_count_valence_electrons    
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Simplistic routine to calculate the total number of valence electrons and work 
out if molecule is a radical or not 


Requires:&nbsp;&nbsp; Molecule, charge <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; total number of valence electrons, flag denoting odd number of valence electrons <br>

### <a name="anchor426"></a>Mol_delete_duplicate_atoms     
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Delete an apparently duplicated atom. Looks for atoms that are close in space 
(within 0.001 A). Then checks to make sure they are the same element. No 
further checks are done. Hydrogen atoms are deleted where these are attached 
to deleted heavy atoms, unless a flag is sent to keep them. This flag has no 
effect where the hydrogen atoms are deleted by reason of being duplicates in 
their own right. 


Requires:&nbsp;&nbsp; Molecule, tolerance, optional flag to keep hydrogens when these are attached to deleted atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of atoms deleted <br>

### <a name="anchor427"></a>Mol_find_all_angles            
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Find all atom-atom-atom angles in a molecule 


Angle array is stored in $mol->{ANGLES} 


Label_waters must be run first if water is to be ignored 


Requires:&nbsp;&nbsp; Molecule, (optional) flag to ignore waters <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Arrayref of angles, each of which is an arrayref of 3 atom numbers <br>

### <a name="anchor428"></a>Mol_find_all_bonds             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Find all bonds in a molecule 


Bond array is stored in $mol->{BONDS} 


Label_waters must be run first if water is to be ignored 


Requires:&nbsp;&nbsp; Molecule, (optional) flag to ignore waters <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Arrayref of bonds, each of which is an arrayref of 2 atom numbers <br>

### <a name="anchor429"></a>Mol_find_all_dihedrals         
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Find all atom-atom-atom-atom dihedral (torsion) angles in a molecule 


Dihedral array is stored in $mol->{DIHEDRALS} 


Label_waters must be run first if water is to be ignored 


Requires:&nbsp;&nbsp; Molecule, (optional) flag to ignore waters, (optional) flag to ignore dihedrals containing hydrogen, flag to only find unique dihedrals (ie only one dihedral per bond) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Arrayref of sorted dihedrals, each of which is an arrayref of 4 atom numbers <br>

### <a name="anchor430"></a>Mol_find_chiral_impropers      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Find improper torsion angles for chiral centres (some force fields require 
this) 


Impropers are stored in $mol->{C_IMPROPERS} 


Returns improper torsions where an atom has more than two attached atoms Only 
one improper is produced in cases where an atom has four attached atoms using 
the first three connected atoms. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Arrayref of improper dihedrals, each of which is an arrayref of 4 atom numbers <br>

### <a name="anchor431"></a>Mol_find_donors_acceptors      
/vcp1/people/david/bin/Silico1.15/lib/silico_hb.pm 


Count molecule H-bond donors and acceptors and label them 


Requires that hydrogens are present on molecule 


Donors are defined as N, O or S with attatched H. 


Acceptors: are N, O, but not in a planar 5-memebered ring. N is counted 
provided valence < 4 except for amides; S is counted provided valence < 3; 


Atoms are labelled as HBACCEPTOR and/or HBDONOR 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of donors (scalar), number of acceptors (scalar) <br>
Sets: $mol-&gt;{HBDONORS} and $mol-&gt;{HBACCEPTORS} (atomlists) <br>

### <a name="anchor432"></a>Mol_find_neighbouring_atoms    
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Make a list of connected atom properties in the 'CONNECTED' hash for each atom 


Labels counts heteroatoms as 'HET'. Counts nonhydrogens as 'HEAVY'. Carbonyls 
as 'CO'. Thioketo/nes as 'CS' 


Requires:&nbsp;&nbsp; Molecule, optional list of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor433"></a>Mol_find_planar_impropers      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Finds improper dihedrals centred on planar atoms 


Impropers are stored in $mol->{P_IMPROPERS} 




J 


| 


I 


/ \ 


K L 




Orders improper dihedrals suitable for the OPLS force field 


Atom I is always position 2, while the unique atom is usually position 1 


Note this subroutine needs a good clean up to remove duplication 


Requires:&nbsp;&nbsp; Molecule, flag to ignore waters <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Arrayref of improper dihedrals, each of which is an arrayref of 4 atom numbers <br>

### <a name="anchor434"></a>Mol_fix_atom_names             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Removes initial and final spaces from atom names. Other spaces are replaced by 
underscores. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor435"></a>Mol_fix_multiple_bonds_general 
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Determine the bondorder score of any bond.  Note: This subroutine is probably 
deficient in setting bondorders of 5-membered planar rings with poor bond 
lengths and no hydrogens. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor436"></a>Mol_fix_sodiums                
/vcp1/people/david/bin/Silico1.15/bin/make_index 


Ensures that all atoms named NA are recognised as sodium atoms. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor437"></a>Mol_fragment                   
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Generate molecule fragments and save in both an array and a hash 


By default all unique fragments are returned. ALL fragments (including 
duplicates) are returned if the 'return_paths' flag is set. $mol->{PATH_ARRAY} 
is also set in this case. 


Requires:&nbsp;&nbsp; molecule, fragment typing method, flag to add hydrogens, flag to delete preexisting fragments, flag to return paths, flag to be quiet  <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{FRAGMENT_HASH}, $mol-&gt;{FRAGMENT_ARRAY}, $mol-&gt;{FP_HASH} and $mol-&gt;{FP_ARRAY} and optionally PATH_ARRAY <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Array of fragments <br>

### <a name="anchor438"></a>Mol_fragment_and_path          
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Generate ALL molecule fragments and corresponding atom paths 


Paths and fragment arrays are stored as mol->{FRAGMENT_ARRAY} and {PATH_ARRAY} 


Requires:&nbsp;&nbsp; molecule, optional: atomlist, fragment typer, min fragment size max fragment size, flag to stop sorting fragments by size, flag to force generation of fragments rather than use cached ones <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Array of fragments <br>

### <a name="anchor439"></a>Mol_get_functional_group       
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Get array of functional groups given a functional group name 


Requires:&nbsp;&nbsp; molecule, group name (string) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to array of functional groups (each fg is a hash) <br>

### <a name="anchor440"></a>Mol_get_sequence               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Extract the amino or nucleic acid sequence from a molecule 


Requires:&nbsp;&nbsp; Molecule, optional options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ensemble of sequences. One sequence for each CHAIN or SUBID <br>

Options: NOHETATM - ignore all pdb HETATM records 


### <a name="anchor441"></a>Mol_has_h                      
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Find out if a molecule has no, polar or all hydrogens 


This check is quick and not exhaustive. Note that it is fooled by molecules 
that contain a some hydrogens but not all valences are full 


Requires:&nbsp;&nbsp; molecule, flag to check if ANY hydrogens are present in file <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 0 if no hydrogens, 1 if polar hydrogens, 2 nonpolar hydrogens (ie connected to carbon) or -1 if hydrogens are present but bonds are not. Also sets $mol-&gt;{HAS_HYDROGENS} <br>

### <a name="anchor442"></a>Mol_label_alpha_fg             
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label atoms as being alpha to a given functional group atom 


Requires that mol_label_functional_group has been run 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor443"></a>Mol_label_beta_fg              
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label atoms as being beta to a given functional group atom 


Requires that mol_label_functional_group and mol_label_alpha_fg have been run 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor444"></a>Mol_label_functional_group     
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


General functional group assigner 


Label atoms of important functional groups and generate approximate charge 
groups for carboxylates, amino groups, sulphate groups, guanidines and 
guanidiniums 


Identifies the following functional group atoms with the following 
$atom->{FG}{<group>} 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;AA_CA AA_CTER AA_N AA_NTER  ACETAL_O  ACID_HALIDE_C ACID_HALIDE_O  ALCOHOL_C ALDEHYDE_C ALDEHYDE_O ALKENE_CH0 ALKENE_CH1 ALKENE_CH2 ALKOXIDE_C ALKOXIDE_O ALKYL_PHOSPHATE_P ALKYNE ALKYNE_ALPHA_CH0 ALKYNE_ALPHA_CH1 ALKYNE_ALPHA_CH2 ALKYNE_ALPHA_CH3 ALKYNE_CH1 ALPHA_GUANIDINE_C ALPHA_GUANIDINIUM_C AMIDE_C AMIDE_O AMIDO_C AMIDO_N AMINO_C AMMONIUM_C AROM_NITRILE_C AROM_NITRILE_N  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;AROMATIC   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CARBAMATE_C CARBAMATE_N CARBAMATE_O CARBANION_ALPHA_C CARBANION_C CARBONYL_ALPHA_C CARBONYL_C CARBOXYLATE_C CARBOXYLATE_O CARBOXYLIC_C CARBOXYLIC_O CHLOROALKENE_C CHLOROALKENE_Cl DIALKYL_PHOSPHATE_ESTER_O DIALKYL_PHOSPHATE_P DICHLOROALKENE_C DIOL_C DISULFIDE_S DPE_O  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ESTER_C ESTER_O ETHER_C ETHER_O FORMYL_C FORMYL_O  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;GUANIDINE_C GUANIDINE_N GUANIDINIUM_C GUANIDINIUM_N   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;IMIDE_C IMIDE_N IMIDE_O   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;KETONE_C KETONE_O   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;METHYL_C  MONOALKYL_PHOSPHATE_ESTER_O MPE_O   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NITRILE_ANION_ALPHA_C NITRILE_ANION_C NITRILE_ANION_N NITRILE_C NITRILE_N NITRO_C NITRO_N NITRO_O   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;PHOSPHATE_ESTER_O PHOSPHATE_O PHOSPHATE_P PHOSPHONATE_C PHOSPHONATE_ESTER_C PHOSPHONATE_O PHOSPHONATE_P PHOSPHONIUM_C PHOSPHONIUM_P P_AMIDE_N P_AMINO_C P_AMINO_N P_AMMONIUM_C P_AMMONIUM_N   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Q_AMMONIUM_N   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;SN_ALPHA_C SULFIDE_C SULFIDE_S SULFONAMIDE_C SULFONAMIDE_N SULFONAMIDE_N_ALPHA_C SULFONAMIDE_O SULFONAMIDE_S SULFONATE_C SULFONATE_O SULFONATE_S SULFONE_O SULFONE_S SULFOXIDE_C SULFOXIDE_O SULFOXIDE_S S_AMIDE_N S_AMINO_C S_AMINO_N S_AMMONIUM_C S_AMMONIUM_N S_CARBAMATE_N   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;TFM_C TFM_F THIOLATE_C THIOLATE_S THIOL_C THIOL_S TRIOL_C T_AMIDE_C T_AMIDE_N T_AMINO_C T_AMINO_N T_AMMONIUM_C T_AMMONIUM_N T_CARBAMATE_N   <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;UREA_C UREA_N UREA_O   <br>

List of all functional groups are stored in mol->{FG} 


Charge groups are stored as an array of arrays in mol->{CHARGE_GROUP} 




Note 1: The routine molecule_find_planar_atoms should be run first although it 
will be run within this routine if it has not already 


Note 2: This routine requires that bond orders are correct 


### <a name="anchor445"></a>Mol_label_heterocycles         
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Label atoms in heterocyclic rings 


Requires that find_rings has been run and that bond orders are present and 
that aromatic rings are represented using aromatic bonds (ie 
make_aromatic_bonds has been run). List of heterocycles is also put into 
$mol->{HETEROCYCLES} 


Requires:&nbsp;&nbsp; molecule, maximum ring size (default 8), optional prefix to add to heterocycle name 'eg HET' <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; List of heterocycles in this molecule <br>

### <a name="anchor446"></a>Mol_label_smiles_invariants    
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Label all atoms in a molecule subset with their SMILES invariants 


This is based on the CANGEN algorithm described in Weininger, Weininger and 
Weininger, J. Chem. Inf. Comput. Sci. 1989, 29, (2), 97-101. 


Requires:&nbsp;&nbsp; molecule, optional subset of atoms to use <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor447"></a>Mol_label_united_atom          
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate missing hydrogens for carbon atoms and add UNITED_ATOM and 
MISSING_HYDROGEN labels 


Used for united atom forcefields 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor448"></a>Mol_map_sequence               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Map residues of molecule file onto a sequence 


Adds the atoms of each molecule residue to the sequence residue array 
@{%res->{ATOMS}} 


Requires;&nbsp;&nbsp; molecule, sequence <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor449"></a>Mol_move_atoms_into_cell       
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Translate all atoms into the unit cell (ie into the range 0 -> cell_dim). 


Requires:&nbsp;&nbsp; Molecule, cell <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor450"></a>Mol_move_into_cell             
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Translate a molecule into a unit cell using the centroid of the molecule as a 
reference 


Unit cell has coordinates starting at 0,0,0 


Requires:&nbsp;&nbsp; Molecule, cell, flag to move based on position the 'middle' atom of the molecule rather than centroid (much faster). <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor451"></a>Mol_move_into_cell_using_cell_vectors 
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Translate a molecule into a unit cell using cell_vecltors 


Requires:&nbsp;&nbsp; Molecule, cell, flag to move based on position the 'middle' atom of the molecule rather than centroid (much faster). <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor452"></a>Mol_orient_polarity            
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Orient molecule polarity dipole along axis 


Requres:&nbsp;&nbsp;&nbsp;&nbsp; molecule, align vector (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

Default aligns polar centre along +Y axis. -xy flag aligns polar end to +Z 


Nonpolar centre is translated to the origin 


### <a name="anchor453"></a>Mol_partition_on_bond          
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Label atoms on one side of a bond and not the other 


Is expected to fail if the bond is within a ring 


Requires:&nbsp;&nbsp; molecule, bond atom 1 (static), bond atom 2 (to move) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Time taken <br>

### <a name="anchor454"></a>Mol_partition_on_bond2         
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Label atoms on one side of a bond (best non-recursive version) 


Will fail if start atom is in a ring 


Requires:&nbsp;&nbsp; molecule, bond atom 1 (static), bond atom 2 (to move), anchor atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; List of stationary atoms, list of mobile atoms <br>

### <a name="anchor455"></a>Mol_path_to_fragments          
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Convert an atom path into fragments 


Uses $Silico::fp_fragment_typer. 


Atom&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; types; element (atomic element), p_np (polar [p] or nonpolar [n]) or none (all set to X) <br>
Bond&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; types; simple (set to single, double/aromatic, triple), none (all set to single) <br>
Requires:&nbsp;&nbsp; molecule, path, typer (optional otherwise uses $Silico::fp_fragment_typer); <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; fragment <br>

### <a name="anchor456"></a>Mol_print_warnings             
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Print warnings from $mol->{WARN} 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor457"></a>Mol_rama                       
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Find backbone dihedrals and phi, psi and omega angles 


Requires:&nbsp;&nbsp; Molecule, parent_ramadata(optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ramadata record <br>

### <a name="anchor458"></a>Mol_rama_angles                
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Caculate Ramachandran angles for RAMADATA 


Omega is undefined if the residue is C- or N-terminal 


Also stores mol->{ENERGY} in ramadata->[0]{ENERGY}; 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{RAMADATA}[$rescount]{PHI} and $mol-&gt;{RAMADATA}[$rescount]{PSI} = $psi <br>

### <a name="anchor459"></a>Mol_read_error                 
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Print an error on failing to read molecule data from a file 


Requires:&nbsp;&nbsp; filename, flag to write error message and die <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor460"></a>Mol_rename_atoms               
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Name atoms using one of several methods: 


Consecutive; Atom numbering increases by atom <br>
Consecutive-H; Atom numbering increases by atom, but hydrogens are named after the parent heavy atom <br>
Element;&nbsp;&nbsp;&nbsp;&nbsp; Atom numbering is independent for each element <br>
Element-H; As for element, but hydrogens are named after the parent heavy atom <br>



Requires:&nbsp;&nbsp; molecule, method (consecutive, element, consecutive-h), flag to only number if existing names are not unique <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor461"></a>Mol_renumber_residues          
/vcp1/people/david/bin/Silico1.15/bin/make_index 


Renumbers residues based on the order in which they are listed in the file 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor462"></a>Mol_renumber_substructures     
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Renumbers substructures so all residues are numbered sequentially from 1 


Increments count on SUBID, SUBNAME, SEGID, or CHAIN 


Requires:&nbsp;&nbsp; Molecule, starting number (optional, default 1) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Highest new substructure number <br>

### <a name="anchor463"></a>Mol_rescale_bonds              
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rescale all bond lengths in a molecule, based on either a scaling factor, or 
an optimum average C-C bond length. 


Requires:&nbsp;&nbsp; Molecule, one of either scaling factor or target length <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor464"></a>Mol_rotate_euler{              
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotates a molecule using Euler angles 


Uses the 'x convention'. See http://demonstrations.wolfram.com/EulerAngles/ 


Requires:&nbsp;&nbsp; mol, phi, theta, psi (angles in radians) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; rotated molecule <br>

### <a name="anchor465"></a>Mol_score_hybridisation        
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Determine the hybridisation state of each atom in a molecule. 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor466"></a>Mol_sdf_stereo                 
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Crudely use SDF stereo descriptors to convert 2D to 3D files 


WARNING: This method works for simple molecules but will fail for complex, 
fused systems. The minimised result from this should be checked with 
sdf_check_chiral_parity 


Assumes molecule has been squashed into the XY plane Adds a hydrogen atom to 
carbons if there valences are not filled 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor467"></a>Mol_sdf_stereo                 
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Crudely use SDF stereo descriptors to convert 2D to 3D files 


WARNING: This method works for simple molecules but will fail for complex, 
fused systems. The minimised result from this should be checked with 
sdf_check_chiral_parity 


Assumes molecule has been squashed into the XY plane Adds a hydrogen atom to 
carbons if there valences are not filled 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor468"></a>Mol_set_source_data            
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Set up molecule source data such as SOURCE_FILE_TYPE, SOURCE_FILE_NAME, etc 


Requires:&nbsp;&nbsp; molecule, file_record, format (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor469"></a>Mol_smiles_string              
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Create a SMILES string from a molecule. 


Requires:&nbsp;&nbsp; molecule, atom list, flags to generate explicit bondorders, explicit hydrogens (1 = all atoms, 2 = only where present), Kekule bonds, to mark missing hydrogens in molecule <br>

Sets $mol->{SMILES} 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor470"></a>Mol_sort_smiles                
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Sort the atoms in a molecule into a canonical order based on the smiles 
representation 


Runs mol_smiles_string if required; 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; canonical smiles string <br>
Modifies:&nbsp;&nbsp; sorted molecule <br>

### <a name="anchor471"></a>Mol_tanimoto                   
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate the Tanimoto coefficient given two molecules 


Tanimoto coeff = Num common fragments / Num fragments in mol1 + Num fragments 
in mol2 - Num common fragments 


Tversky coeff = Num common fragments / alpha (Num fragments in mol1 - Num 
common fragments) + beta (Num fragments in mol2 - Num common fragments) + Num 
common fragments 


alpha = 0 and beta = 1 => Molecule 2 as a substructure of Molecule 1 


alpha = 1 and beta = 2 => Molecule 2 as a superstructure of Molecule 1 


Requires: hash1, hash2, (alpha, beta coefficiencts for Tversky, optional) 


Returns: Tanimoto coefficient 


### <a name="anchor472"></a>Mol_transfer_coordinates       
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Transfer coordinates from molecule 2 to to molecule 1 


Requires:&nbsp;&nbsp; molecule 1, molecule 2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; new molecule with properties of molecule 1 and coordinates of molecule2 <br>

### <a name="anchor473"></a>Mol_type_mmod                  
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Macromodel atom type routine 


Assumes that structures are three-dimensional (ie not planar 2D 
representations) and have complete bond types. Atom geometry is used to 
identify aromatic rings and planar nitrogen atoms. Does not assume that 
hydrogens are present. Makes primary amines N.4 and makes O-C=O into 
carboxylate groups if explicit hydrogens are not present. 


Requires:&nbsp;&nbsp; molecule. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Macromodel atom type. <br>

### <a name="anchor474"></a>Mol_type_sybyl                 
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Generate Sybyl atom types for a molecule 


Types are put in to $atom->{TYPE}. 


This routine does not assume that hydrogens are present. 


Calls molecule_find_rings to rings and planar rings 


Identifies carboxylate, amide and guanidinium carbon atoms and marks them with 
the flags $atom->{FG}{CARBOXYLATE_C}, $atom->{FG}{AMIDE_C} and 
$atom->{FG}{GUANIDINIUM_C} respectively. 


If hydrogens are not present then this routine uses simple rules to try to put 
molecules in physiologialy relevant ionisation state. Makes primary amines N.4 
and makes O-C=O into carboxylate groups. 


This routine makes Ti,Cr and Co octahedral. 


Requires:&nbsp;&nbsp; atom, molecule. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; -1 if there was an atom typing error. <br>

### <a name="anchor475"></a>Mol_warn                       
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Keep track of errors in $mol->{WARN} 


Requires:&nbsp;&nbsp; Molecule, warning string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor476"></a>Molecule_bin_atoms             
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Bin atoms 


A faster bin atoms routine. This routine can optionally set atom->{BIN} 


Requires:&nbsp;&nbsp; molecule, flag to add bin to atom <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of binned atoms <br>

### <a name="anchor477"></a>Molecule_bin_atoms_old         
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Bin atoms 


A faster bin atoms routine. This routine can optionally set atom->{BIN} 


Requires:&nbsp;&nbsp; molecule, atomlist (optional), binsize (optional), numbins array (optional), flag to add bin to atom (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of binned atoms <br>

### <a name="anchor478"></a>Molecule_centre                
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate the x, y, z extents of a molecule and the centre of the box. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Array ({maxx, maxy, maxz, minx, miny, minz, centx, centy, centz}) <br>

### <a name="anchor479"></a>Molecule_centre_of_mass        
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate the centre of mass of a molecule or atom subset 


Requires:&nbsp;&nbsp; Molecule, (optional) atom list <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Centre of mass as an array ({x, y, z}) <br>

### <a name="anchor480"></a>Molecule_check_and_fix_connectivity 
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Check that a molecule has bonds and that correct bond orders. Generate them if 
they are missing 


A reasonable number of bonds is deemed to be (numatoms-1)/2; 


Valid options BOND, NOBOND 


Requires:&nbsp;&nbsp; molecule, $options (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef on error <br>

### <a name="anchor481"></a>Molecule_check_join_points     
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Check a dendrimer monomer for exactly one join point to a putative ancestor. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor482"></a>Molecule_connect_pdb_atoms_by_name 
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Bond atoms together using a pdb lookup table 


This routine bonds together atoms proteins using a lookup table 


This routine depends on the atom names being correct (See also 
molecule_pdb_rename_h). Terminal residues etc are not present in 
amino_acid.dat file 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor483"></a>Molecule_copy_coordinates      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Copy coordinates from one molecule to another (similar) molecule 


Requires:&nbsp;&nbsp; two molecules <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor484"></a>Molecule_copy_header           
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Copy a molecule's header data to another molecule. 


Requires:&nbsp;&nbsp; two molecules <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor485"></a>Molecule_copy_noatoms          
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create a copy of a molecule, but without any atoms 


Note: this routine fails for cyclic references. These could be implemented by 
keeping a list of references in a hash. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; The same molecule, but containing no $mol-&gt;{ATOMS} records or anything in $mol-&gt;{NUMATOMS} <br>

### <a name="anchor486"></a>Molecule_count_numbonds        
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Return the number of bonds. 


Note: Each bond is specified twice in CONNECT records, once for each of the 
two atoms in the bond. So the total number of CONNECT records is divided by 
two, giving the number of bonds. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of bonds <br>

### <a name="anchor487"></a>Molecule_delete_alternate_atoms 
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Remove alternate atoms from a pdb file 


Requires:&nbsp;&nbsp; Molecule, aternate atom to RETAIN (taken from get flag or default value A) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Molecule, modified and renumbered <br>

### <a name="anchor488"></a>Molecule_delete_atom           
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Deletes a single atom by offset. 


Warning: You will need to run 'molecule_pack' after deleting a set of atoms or 
all hell will break loose. 


Requires:&nbsp;&nbsp; molecule, number of atom to be deleted. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 <br>

### <a name="anchor489"></a>Molecule_delete_atoms          
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Delete a list of atoms, identified via atom offsets. 


Requires:&nbsp;&nbsp; molecule, list <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor490"></a>Molecule_delete_dummy          
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Delete all dummy atoms (eg lone pairs) from a molecule 


Requires:&nbsp;&nbsp; Molecule, optional list identifying a subset <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of deleted atoms <br>

### <a name="anchor491"></a>Molecule_delete_h              
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Delete all hydrogens from a molecule 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of deleted hydrogens <br>

### <a name="anchor492"></a>Molecule_delete_h_resname      
/vcp1/people/david/bin/Silico1.15/bin/mol_del_h 


Delete all hydrogens from a particular residue type within a molecule 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of deleted hydrogens <br>

### <a name="anchor493"></a>Molecule_delete_nonpolar_h     
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Delete nonpolar hydrogens from a molecule 


Flag will to retain hydrogens bonded to SP2 carbons (aldehydes, aromatic 
rings, double bonds, etc) 


Charges on the hydrogens being deleted are transferred on to the parent atom. 


Requires:&nbsp;&nbsp; molecule, flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of deleted hydrogens <br>

### <a name="anchor494"></a>Molecule_delete_polar_h        
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Delete polar hydrogens (ie those not attached to carbon) from a molecule 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of deleted hydrogens <br>

### <a name="anchor495"></a>Molecule_dimensions            
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculate the x, y, z dimensions of a molecule 


Requires:&nbsp;&nbsp; Molecule, margin to add (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Dimensions <br>

### <a name="anchor496"></a>Molecule_find_planar_atoms     
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Call geometry or bondorder based routines to find planar atoms 


Uses geometry routine by default 


Note that the geometry and bondorder routines do not return the same sets of 
atoms. Also the results will vary if hydrogens are missing 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor497"></a>Molecule_find_planar_atoms_bondorders 
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Find planar atoms using bondorders only 


All atoms which have double bonds are planar. This includes S=O and P=O which 
may cause some problems 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; an array of planar atoms and sets $mol-&gt;{PLANAR_ATOMS} <br>

### <a name="anchor498"></a>Molecule_find_planar_atoms_connections 
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Find planar atoms using connected atoms only (assumes no coordinates) 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; an array of planar atoms and sets $mol-&gt;{PLANAR_ATOMS} <br>

### <a name="anchor499"></a>Molecule_find_planar_atoms_geom 
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Find planar atoms using molecule geometry or bondorder 


A planar atom is defined as one with having a double bond, aromatic bond or 
three substituents and an improper dihedral less than 15 degrees from planar. 


Does not find planar tetracoordinate atoms. 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; an array of planar atoms and sets $mol-&gt;{PLANAR_ATOMS} <br>

### <a name="anchor500"></a>Molecule_find_rings            
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Find rings and planar rings in a molecule. 


Rings are arrays of atom numbers. This routine returns all rings found up to 
max_ring_size. It does not find the minimum set of smallest rings. So 
naphthalene will return three rings, two of size six and one of size 10. 


Sets the following arrays for each molecule: 


@{$molecule-&gt;{RINGS}} All rings in the molecule <br>
@{$molecule-&gt;{PRINGS}} All planar rings <br>
@{$molecule-&gt;{ARINGS}} All aromatic rings (where a planar ring is not aromatic) <br>

The maximum ring size is stored in $mol->{RINGSIZE_MAX}. 


The following arrays are set for each atom. Where the atom is not a member of 
any ring, the array is empty. 


@{$atom-&gt;{RINGS}} All rings that that atom belongs to <br>
@{$atom-&gt;{PRINGS}} All planar rings that that atom belongs to <br>

In addition, the following flags are set 


$atom-&gt;{AROMATIC RING} flag for all 5 and 6 membered aromatic rings. <br>



$atom-&gt;{NOTRING} flag for all non-ring atoms <br>
Requires:&nbsp;&nbsp; ensemble, maximum ring size (optional, default 8), use non-recursive subroutine to find rings (optional), force (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; rings, planar rings, aromatic rings <br>

### <a name="anchor501"></a>Molecule_fix_and_get_residues  
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Return an array of residues. Existing residues are returned if they exist and 
seem sane otherwise new residues are generated 


Calls either 'molecule_get_residues' or 'molecule_make_residues' 


Sets $atom->{SUBCOUNT} (numbered from 1) 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of residues. Each residue is an array of atoms belonging to that residue <br>

### <a name="anchor502"></a>Molecule_fix_sybyl_substructures 
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Creates records for sybyl substructures. Also generates BACKBONE status flags 
if amino acids are identified. 


Requires:&nbsp;&nbsp; molecule. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing. <br>

### <a name="anchor503"></a>Molecule_formal_charge         
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate the formal charge on a molecule. Note. Formal charges are 
recalcuated for each atom which may REMOVE formal charges that are already 
present. This happens particularly if no hydrogens are present in the molecule 


Calls atom_formal_charge 


Requires:&nbsp;&nbsp; molecule <br>
Optional:&nbsp;&nbsp; atom_list, optional flag to denote missing hydrogens, flag to print atoms with nonzero formal charge <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; total formal charge <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{FORMAL_CHARGE}, $mol-&gt;{CHARGED_ATOMS}; <br>

### <a name="anchor504"></a>Molecule_formula               
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate the molecular formula of a molecule 


Missing hydrogens are calculated and they are added to the formula as 'H*' 


Sets $mol->{FORMULA}; 


Requires:&nbsp;&nbsp; Molecule, flag to suppress no-hydrogens warning (optional), atomlist (optional), flag to supress count of missing hydrogens. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Molecular formula as a hash <br>

### <a name="anchor505"></a>Molecule_formula_string        
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Return molecule formula as a string 


Requires:&nbsp;&nbsp; molecule - see other options for molecule_formuula <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor506"></a>Molecule_generate_bondorders   
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Calculate bondorders for bonds 


Requires:&nbsp;&nbsp; molecule, flag to reset all bondorders to single <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor507"></a>Molecule_generate_bonds        
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Create new bonds for structure with no connection table for example a pdb file 


Uses a fast residue-based approach to generate bonds in waters if the molecule 
contains more than 500 atoms 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; bondcount <br>

### <a name="anchor508"></a>Molecule_get_chains            
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Return an array of chains 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of atom lists <br>

### <a name="anchor509"></a>Molecule_get_residues          
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Return an array of amino acid residues that already exist within a file. Each 
residue is a simple array of atoms 


Sets $atom->{SUBCOUNT} (numbered from 1) 


Setting the gap flag will add empty residues for any gaps in the AA sequence. 
The mark_first_residue flag will mark the first atom of each residue with a 
'STARTRES' flag. 


Requires:&nbsp;&nbsp; molecule, optional atom list containing a subset of atoms, gap flag, flag to mark the first atom of each residue <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of arrays of pointers to all atoms in each residue <br>

### <a name="anchor510"></a>Molecule_get_residues_hash     
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Return an array of residues. Each residue is a hash. 


Atoms are in $res->{ATOMS} 


Number (count) are in $res->{NUM} 


NAME is in $res->{NAME} 


SUBID is in $res->{SUBID} 


Requires:&nbsp;&nbsp; molecule, optional atom list containing a subset of atoms, gap flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of hashes <br>

### <a name="anchor511"></a>Molecule_label_submols         
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


A non-recursive subroutine to label submolecules based on connectivity 


Sets $atom->{SUBMOL}. Creates arrays of atoms in $mol->{SUBMOLS}[submol#] 


Requires:&nbsp;&nbsp; Molecule, flag to also set CHAIN <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; $mol-&gt;{SUBMOLS} <br>

### <a name="anchor512"></a>Molecule_make_residues         
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Divide a molecule into a NEW set residues in a molecule based on connectivity 
and amino acids. Old residues are overwritten. 


Sets $atom->{SUBCOUNT} (numbered from 1) 


Splits residues based on: 


Amide bonds 


Disulfides 


Ethers 


Esters 


Triazoles resulting from click reactions according to the atoms from the 
original azide and alkyne 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; List of residues ordered N-&gt;C as an array of arrays of pointers to all atoms in each residue or undef if there is a problem. Also sets $mol-&gt;{RESIDUES} <br>

### <a name="anchor513"></a>Molecule_mw                    
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Calculate molecule molecular weight 


Requires:&nbsp;&nbsp; Molecule, optional list of atoms, flag to suppress check for united atoms (not implemented) flag to supress 'no hydrogens' warning message <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Molecular weight as a scalar <br>

Calcluates where hydrogens are missing (for united atom forcefields). 


### <a name="anchor514"></a>Molecule_pack                  
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Remove any undefined atoms from a molecule. 


Bonds are updated. Atom->{NUM} is renumbered 


Requires:&nbsp;&nbsp; molecule, nowarnings flag. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; packed molecule <br>

### <a name="anchor515"></a>Molecule_path                  
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Call recursive routine to find all paths in a molecule 


Requires:&nbsp;&nbsp; molecule, optional pointer to array of atom numbers Max fragment size, min fragment size hydrogen treatment are optionally set using globals $Silico::fp_max, $Silico::fp_min <br>

Note will return a previously calculated, cached path unless the 'force' flag 
is set 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to array of atom number arrays (paths) <br>

### <a name="anchor516"></a>Molecule_pdb_rename_backbone   
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Rename pdb backbone atoms 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor517"></a>Molecule_pdb_rename_h          
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Rename protein hydrogens to conform to pdb nomenclature. Molecule is sorted if 
hydrogens are out of order in file 


Requires:&nbsp;&nbsp; molecule, aminoacid datafile (default amino_acid_atoms.dat), flag to delete unnamed hydrogens <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; number of unassigned hydrogens <br>

### <a name="anchor518"></a>Molecule_pdb_rename_h_dataread 
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in data for molecule_pdb_rename_h 


Requires:&nbsp;&nbsp; datafile <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array, array, hash of residues <br>

### <a name="anchor519"></a>Molecule_printout              
/vcp1/people/david/bin/Silico1.15/lib/silico_debug.pm 


Print out several atoms from a molecule record. 


Prints out all molecule records and those for the first 5 (MacOS) or 10 atoms. 
Takes optional second argument 'last' prints last 10 atoms. 'all' prints all 
atoms. 


Requires:&nbsp;&nbsp; molecule, 'all' (optional), filehandle (optional). <br>

### <a name="anchor520"></a>Molecule_random_move_cartesian 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Move and rotate a molecule (around a random axis) by a random amount 


Will use a periodic cell if supplied. Molecule is translated back into the 
cell 


Require:&nbsp;&nbsp;&nbsp;&nbsp; molecule, translation vector (default cell size, or 10, 10, 10), rotation vector (default random vector with up to 360 deg rotation), cell (optional) <br>

### <a name="anchor521"></a>Molecule_renumber              
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Renumber atoms in a molecule (Note!. Only renumbers $atom->{NUM}) 


Renumbers from 1 or from an optional second argument. 


Requires:&nbsp;&nbsp; molecule, (optional starting number). <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor522"></a>Molecule_renumber2             
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Renumber atoms in a molecule using an offset. 


Useful to renumber one molecule before combining with another 


Renumbers $atom->NUM, CONNECT, GMX_BONDS, GMX_ANGLES, GMX_DIHEDRALS, 
GMX_PAIRS, and GMX_EXCLUSIONS 


Requires:&nbsp;&nbsp; molecule, Initial value for atom-&gt;{NUM} (default 1). <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor523"></a>Molecule_renumber2_new         
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Renumber atoms in a molecule using an offset. 


Useful to renumber one molecule before combining with another 


Renumbers $atom->NUM, CONNECT, GMX_BONDS, GMX_ANGLES, GMX_DIHEDRALS, 
GMX_PAIRS, and GMX_EXCLUSIONS 


Requires:&nbsp;&nbsp; molecule, Initial value for atom-&gt;{NUM} (default 1). <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor524"></a>Molecule_reorder               
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Sort atom order within a molecule using a specified order 


Sort molecules according to specified field or $atom->{NEW_NUM} by default; 


Requires:&nbsp;&nbsp; molecule. <br>

### <a name="anchor525"></a>Molecule_rot_angle_x           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotate a molecule around X axis by angle (degrees). 


Requires:&nbsp;&nbsp; molecule, angle (deg); <br>

### <a name="anchor526"></a>Molecule_rot_angle_y           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotate a molecule around Y axis by angle (degrees). 


Requires:&nbsp;&nbsp; molecule, angle (deg); <br>

### <a name="anchor527"></a>Molecule_rot_angle_z           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotate a molecule around Z axis by angle (degrees). 


Requires:&nbsp;&nbsp; molecule, angle (deg); <br>

### <a name="anchor528"></a>Molecule_rotate                
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotate a molecule using a 3x3 matrix . 


Requires:&nbsp;&nbsp; molecule, m1..m9. <br>

### <a name="anchor529"></a>Molecule_rotate_axis           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotate atoms about an arbitrary axis using quaternions 


Require:&nbsp;&nbsp;&nbsp;&nbsp; molecule, angle (in radians), axis (x1, y1, z1), origin (x2, y2, z2) default 0,0,0 <br>

### <a name="anchor530"></a>Molecule_solvate               
/vcp1/people/david/bin/Silico1.15/lib/silico_solvate.pm 


Solvate a molecule 


Requires: molecule, solvent molecule, cell, number of molecules, bilayer flag, output format (pdb etc) <br>
Returns: molecule <br>

### <a name="anchor531"></a>Molecule_split                 
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


A non-recursive subroutine to split a 'molecule' into a series of actual 
molecules 21 October 2004: Improved to reduce memory use and produce atoms 
sorted in the order in which they were present in the original molecule file. 


Requires:&nbsp;&nbsp; Molecule, minimum molecule size to keep (default 1), flag to create 'minmal' molecules (i.e. Dont use molecule_copy_noatoms) <br>

Note. Using the molecule_copy_noatoms is slow and memory-consuming 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ensemble <br>

### <a name="anchor532"></a>Molecule_split_keeping_largest_mol 
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Split a Silico molecule up on the basis of connectivity and return the largest 
of its component molecules 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor533"></a>Molecule_trans_to_centre       
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Translate the centre of a molecule to 0,0,0 or to centre of cell 


Requires:&nbsp;&nbsp; molecule, specifier for origin or centre of cell ('cell' or 'origin'), specifier for whole molecule, largest substructure or hydrophobic centre ('all', 'largest' or 'hydrophobic') <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector of translation <br>

### <a name="anchor534"></a>Molecule_translate             
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Translate a molecule by a vector (x,y,z). 


Requires:&nbsp;&nbsp; molecule, x, y, z. <br>

### <a name="anchor535"></a>Molfile_divide                 
/vcp1/people/david/bin/Silico1.15/bin/mol_divide 


### <a name="anchor536"></a>Mols                           
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Return a list of molecules from a set of molecules or ensembles 


Requires:&nbsp;&nbsp; molecule or ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of molecules <br>

### <a name="anchor537"></a>Mopac_finalise                 
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Fix up atomic elements from mopac file read and perform checks 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor538"></a>Name_pdb_molecule              
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Generate molecule name for pdb molecule 


1. HEADER 2. COMPND or 3. filename 


### <a name="anchor539"></a>New_h                          
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Add a new hydrogens 


Note: The global Silco::add_h_element can be used to change the element being 
added 


Note: The global Silco::add_h_bo can be used to change the order of the new 
bond 


Requires:&nbsp;&nbsp; atom, molecule, offsetX, offsetY, offsetZ <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; new atom <br>

### <a name="anchor540"></a>New_table_delimited            
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Create a new empty delimited table 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; table <br>

### <a name="anchor541"></a>Number_range                   
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Turn a range of numbers (indicated by X-Y, where X and Y are integers) into a 
list of those numbers spanned by the range. 


Requires:&nbsp;&nbsp; integer range as text string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; List of integers <br>

### <a name="anchor542"></a>Open_file                      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


General subroutine to open molecule files for reading. 


Note open_file and close_file are not designed to handle concurrently open 
files 


Example: 


if (!open_file(*INFILE, $infile)) { 


print "Error (read_mol2): Can not open file $infile for reading!\n"; 


return undef; 


} 


Can handle compressed files on UNIX systems. 


Requires:&nbsp;&nbsp; filehandle, filename, 'single' flag to prevent opening files that are already open <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true if file opened ok or if the file was already open and the 'single' flag was set <br>

### <a name="anchor543"></a>Open_molfile                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Open molecule file for reading one molecule at a time 


Uses the extension to determine file type. 


Requires:&nbsp;&nbsp; file record, file name, mode (read(default), write, append), format, option string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; file record or undef if open failed. <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; GZ - gzip compression, BZ2 - bzip2 compression, QUIET - less output <br>

### <a name="anchor544"></a>Optimise_geometry              
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Attempts to find the most splayed geometry. In particular checks for overlap. 


Requires:&nbsp;&nbsp; Molecule, two lists of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Best (lowest) distance function value, and the lowest number of overlaps <br>

### <a name="anchor545"></a>Origins                        
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Origin points for holes o Single point at 0, 0, 0 a Single point in centre of 
cell b Single point in centre of cell (X, Y) and 0 in Z c 4 hexagonally 
arranged holes in xy plane 


### <a name="anchor546"></a>Orthogonal_least_squares_plane_atoms 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Fits a plane to a set of atoms, using the orthogonal least squares method. 


Requires:&nbsp;&nbsp; molecule, set of atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; One point in the plane, normal vector to the plane <br>

### <a name="anchor547"></a>Overwrite_check                
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Check to see if an output file exists and prompt for response if it does 


Check is overridden by -force flag 


Requires:&nbsp;&nbsp; filename, flag to disallow option of changing filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; new filename (if changed) or original filename. The '-force' flag is set if 'Yes to all' is selected. <br>

### <a name="anchor548"></a>Pack_gromacs_angles            
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Pack new format gromacs bonds and angles, etc. following atom deletion 


Atoms are renumbered 


Any bond or angle with undefined values is deleted. 


Requires:&nbsp;&nbsp; Molecule, ctable (array of old and new atom numbers); <br>

### <a name="anchor549"></a>Pack_molecules                 
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Pack molecules in a random or semirandom (e.g. bilayer) way 


Requires:&nbsp;&nbsp; array of molecules to pack, molecule to embed, molecule cell <br>
Also&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; requires subroutine 'randomise_and_test_molecule_position' <br>

### <a name="anchor550"></a>Parse_amino_acids_smiles       
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Parse a file containing SMILES strings of amino acids, putting the entries 
into a hash. 


Requires:&nbsp;&nbsp; nothing, but takes an alternative file as an optional argument <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; hash <br>

### <a name="anchor551"></a>Parse_comma_separated_numbers  
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Take a list of comma-separated numbers, which may include number ranges, and 
turn it into a list of individual numbers in array format. 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array <br>

### <a name="anchor552"></a>Parse_pdb_space_group          
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Parse the file pdb_space_group.dat 


Requires:&nbsp;&nbsp; File <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; hash <br>

### <a name="anchor553"></a>Parse_residue_dictionary       
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Parse a mol2 file containing residue definitions 


Revised version that uses a simpler residue database and generate common 
protonation states N and C termini etc. 


Requires:&nbsp;&nbsp; Nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Hash of molecules - key is SMILES string <br>

### <a name="anchor554"></a>Parse_smiles                   
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Parse a smiles string and return a molecule 


Note. Can not handle chirality. 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor555"></a>Pdb_cell_to_gromacs_bv         
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Create gromacs box vectors 


Requires;&nbsp;&nbsp; molecule <br>

### <a name="anchor556"></a>Pdb_fastsplit                  
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Quickly split a pdb file into component molecules/chains 


Only ATOM, HETATM and CRYST lines are printed. Files are written directly to 
disk named <filebase>_00001 etc, or by chain name if requested. 


Requires:&nbsp;&nbsp; infile, output directory ('.' by default), output style (see subroutine format_oname), flags to split on chain, END, ENDMDL and TER records, flag to name output files by chain <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of files created <br>

### <a name="anchor557"></a>Pdb_format_atom_name           
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Format atom name for pdb file 


Pads atom names with single character element names with a space (Insight 
sometimes doesn't work otherwise) 


Requires:&nbsp;&nbsp; atom, $mol-&gt;{WARN}{FIELD_TRUNCATION} as a local variable <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor558"></a>Pdb_format_atom_number         
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Format PDB atom numbers so that they change to base16 when greater than 9999 
starting at A0000 


Then change to base32 when greater than FFFFF starting at G0000 


Base16 is to maximise compatiblity with OpenFF 


Requires:&nbsp;&nbsp; decimal number <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor559"></a>Pdb_format_residue_number      
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Format PDB residue numbers so that they change to base16 when greater than 
9999 starting at A000 


Requires:&nbsp;&nbsp; decimal number <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor560"></a>Pdb_format_subname             
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Format substructure name to a length of four characters. 


Length can be set to 3 or 4 characters Substructure name is truncated if it is 
too long 


Requires:&nbsp;&nbsp; atom, molecule, maxumum substructure name length <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor561"></a>Pdb_get_ligands                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Extract ligands from pdb file 


Only the first example of each ligand residue type is extracted 


Requires:&nbsp;&nbsp; molecule, min heavy atoms, max heavy atoms, min carbon atoms <br>

### <a name="anchor562"></a>Pdb_get_prot                   
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Simple routine to extract pdb protein atoms 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor563"></a>Pdb_increment_chain            
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Increment pdb chain 


Requires:&nbsp;&nbsp; Alphabetical character <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor564"></a>Pdb_molecule_sort_atoms        
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Routine to sort atoms in to a standard order Sorts by SEGID, CHAIN, residue 
number then by atom number. HETATMS are sorted to the end of the file Atoms 
with a ' ' chain specifier are sorted to the end of the file. 


Atoms are then renumbered. 


Requires:&nbsp;&nbsp; Molecule. (optional, flags to ignore chain and segid when sorting). <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor565"></a>Pdb_rewind                     
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Rewind a pdb file to the start 


Requires:&nbsp;&nbsp; file record <br>
Requires:&nbsp;&nbsp; nothing <br>

### <a name="anchor566"></a>Plot_amides                    
/vcp1/people/david/bin/Silico1.15/bin/mol_amides 


Writes a grace parameter file for the amides plot, and calls grace 


Requires:&nbsp;&nbsp; Amide data file (.adf) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor567"></a>Pmmod                          
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write a formatted line of a macromodel com file 


Requires:&nbsp;&nbsp; arguments as an array <br>

### <a name="anchor568"></a>Primes                         
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Create an array of the first thousand prime numbers: @$Silico::primes 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing (creation of package global array) <br>

### <a name="anchor569"></a>Print_finished_message         
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Prints out a message to indicate a script is finished 


Requires:&nbsp;&nbsp; Optional message to print (default finished) <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;optional character to surround it with (default *), filehandle (optional)  <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor570"></a>Print_flags                    
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Print out valuues of all flags 


Requires:&nbsp;&nbsp; optional filehadle (default STDOUT) <br>

### <a name="anchor571"></a>Print_keys                     
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Printout SDF data keys present in molecule 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor572"></a>Print_timing_message           
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Prints out a message containing the total user time and total wall time used 
by the program 


Requires:&nbsp;&nbsp; Starting user time, filehandle (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor573"></a>Print_title                    
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Print a program title banner. 


Requires:&nbsp;&nbsp; program name, version (optional), filehandle (optional) <br>

### <a name="anchor574"></a>Printbox                       
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Write out a pdb format box 


Writes eight atoms. One at each cornder of the box. Compatible with the box 
produced by DOCK 


Requires:&nbsp;&nbsp; maxx, maxy, maxz, minz, miny, minz, (optional) filename default 'box.pdb' <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed <br>

### <a name="anchor575"></a>Printdoc                       
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Print marked up documentation for a file The pager 'more' is used if the 
parent program has been called from a shell 


Requires:&nbsp;&nbsp; filename (default executing program); <br>

### <a name="anchor576"></a>Printit                        
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Print routine for pack_molecules 


### <a name="anchor577"></a>Process_name                   
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Clean up molecule name 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule name, base name <br>

### <a name="anchor578"></a>Prompt                         
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Print a prompting message and read a reply from standard input. 


Requires:&nbsp;&nbsp; Text to prompt with <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Chomped reply from standard input <br>

### <a name="anchor579"></a>Prune_family_tree              
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Taking a list of branches in a family tree, throw away all those which are 
incomplete. 


Requires:&nbsp;&nbsp; list of branches <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of branches <br>

### <a name="anchor580"></a>Quaternion                     
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates a quaternion - a 'rotation matrix' when you want to rotate a point 
around a vector. 


Requires:&nbsp;&nbsp; Angle, rotation vector <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Rotation matrix as a 9-long (3x3) array <br>

### <a name="anchor581"></a>Quick_finalise                 
/vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm 


Fix up atomic elements from quick file read and perform checks 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor582"></a>Quiet                          
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Check if we are being quiet (i.e., the --quiet option is used). 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 (quiet), 0 (not quiet) <br>

### <a name="anchor583"></a>Rad_to_deg                     
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Convert radians to degrees 


Requires:&nbsp;&nbsp; scalar <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; scalar <br>

### <a name="anchor584"></a>Rama_find_backbone_dihedrals   
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Extract backbone dihedral information from a molecule 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; RAMADATA record <br>

### <a name="anchor585"></a>Ramaplot                       
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Writes a grace parameter file for the Ramachandran plot, and calls grace 


Requires:&nbsp;&nbsp; Ramachandran data file, format, hardcopy flag, residue label flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor586"></a>Ramaplot_by_mol_open           
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Open .ramol files and write headers 


Requires:&nbsp;&nbsp; ramadata record, output filename base, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; file record <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; 'CART' will output sin(phi), cos(phi), sin(psi), cos(psi) instead of phi, phsi. NOHEADER - don't write comment at top of file - required for R <br>

### <a name="anchor587"></a>Ramaplot_open                  
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Open .rama files and write headers 


Requires:output filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; file record <br>

### <a name="anchor588"></a>Randomchar                     
/vcp1/people/david/bin/Silico1.15/bin/mol_characterise 


Generate a random two-letter string. 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor589"></a>Randomise_and_test_molecule_position 
/vcp1/people/david/bin/Silico1.15/bin/bilayer_builder 


Perform random translation and rotation in the bilayer plane and small 
movements perpendicular to bilayer 


Requires:&nbsp;&nbsp; molecule, cell <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor590"></a>Range                          
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Determines the range of a set of data. 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; scalar <br>

### <a name="anchor591"></a>Rank_by_neighbouring_primes    
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Attempt to resolve ties in SMILES ranks using the product of the prime numbers 
corresponding to the ranks of the several neighbours of each of the atoms in 
question. 


Only atoms which have otherwise equivalent ranks will have their relative 
ranks altered by this procedure (though the absolute rank of any given atom 
may change as new ranks are created). 


Requires:&nbsp;&nbsp; list of atoms, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of ranks <br>

### <a name="anchor592"></a>Rank_molecules                 
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Create ranks for a data key. Correctly deals with tied ranks 


Ranks start from 0. 


Requires:&nbsp;&nbsp; molecules, SDF_DATA key for ranking <br>

### <a name="anchor593"></a>Rank_smiles_set                
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Rank a set of atoms on the basis of their SMILES invariants. 


This is based on the CANGEN algorithm described in Weininger, Weininger and 
Weininger, J. Chem. Inf. Comput. Sci. 1989, 29, (2), 97-101. 


Requires:&nbsp;&nbsp; molecule, list of atoms that together form SMILES set <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing; atoms are marked in $atom-&gt;{SMILES_RANK} <br>

### <a name="anchor594"></a>Read_atom_specifier_file       
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Read an atom specifier file 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of atom specifiers <br>

## Simple atom specifier examples: 


ANAME:CA&nbsp;&nbsp;&nbsp;&nbsp; Returns all atoms called 'CA'. <br>
ANAME:CA,CB,CG Returns all atoms called 'CA' or 'CB' or 'CD'. <br>
ANAME:CA,RESNAME:TRP Returns all atoms called 'CA' in all residues called TRP <br>
ANAME:CA,SUBID:4 All atoms called 'CA' in residue number 4 <br>

In the examples above, atoms are returned in the order found in the file. Atom 
specifiers are case sensitive. 


### Ordered atom specifier examples: 


Successive atom specifications can be made. Each is separated by a '|' 


ANAME:CB|ANAME:CA|ANAME:CD Returns all atoms called 'CB', 'CA', 'CD' in that order. <br>
ANAME:CA,RESNUM:1|ANAME:CA,RESNUM:4 Returns 'CA' atoms from residue 1 and 4 in that order. <br>

### Specifier file: 


Contains two sets of atom specifiers. Each set starts with a 'Molecule' line 
which can contain a specifier name (space delimited) 


eg: 


Molecule&nbsp;&nbsp;&nbsp;&nbsp; Backbone <br>
ANAME:C&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <br>
ANAME:CA&nbsp;&nbsp;&nbsp;&nbsp; <br>
ANAME:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; N <br>
Molecule&nbsp;&nbsp;&nbsp;&nbsp; Sidechain <br>
ANAME:CB&nbsp;&nbsp;&nbsp;&nbsp; <br>
ANAME:CG&nbsp;&nbsp;&nbsp;&nbsp; <br>
ANAME:CD&nbsp;&nbsp;&nbsp;&nbsp; <br>

### <a name="anchor595"></a>Read_clustal                   
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Read in Clustal format NA or Protein sequence files 


Requires:&nbsp;&nbsp; Filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Sequences record <br>

### <a name="anchor596"></a>Read_coord_mol                 
/vcp1/people/david/bin/Silico1.15/bin/write_itp 


Read in coordinate molcule, check connectivity, fix ATB problems 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor597"></a>Read_coord_mol                 
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read in coordinate molcule, check connectivity, fix ATB problems 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor598"></a>Read_divcon_cart               
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Read a DivCon Cartesian format file 


Reads only a single structure 


Requires:&nbsp;&nbsp; infile, undef, undef, options (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef on read error <br>

### <a name="anchor599"></a>Read_doc                       
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Read marked up documentation from a silico file 


Requires:&nbsp;&nbsp; Text file marked up using silico convention, flag to exit after encountering first end markup marker ('#&gt;') <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Tagged silico text <br>

Marked up text is also added to a global hash %$FILEHASH 


### <a name="anchor600"></a>Read_doc_subroutine            
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Read marked up documentation from silico subroutines 


Marked up text is returned in a global hash %$SUBHASH 


Requires:&nbsp;&nbsp; Text file marked up using silico convention <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; All lines read <br>

### <a name="anchor601"></a>Read_fasta                     
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Read in fasta sequence format 


INCOMPLETE 


Requires:&nbsp;&nbsp; Filename, $options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Sequences record <br>

### <a name="anchor602"></a>Read_function_line             
/vcp1/people/david/bin/Silico1.15/bin/rdf 


A subroutine to process a single function line from an RDF input file in 
preparation for adding to the FUNCLISTS element of the $input hash 


Passed&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; variables: @_ <br>
External&nbsp;&nbsp;&nbsp;&nbsp; variables: input hashref ($input) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; A single list of functions, with parameters, or -1 if an error occurred <br>

### <a name="anchor603"></a>Read_gamess_cart               
/vcp1/people/david/bin/Silico1.15/lib/silico_gamess.pm 


Simplistic routine to extract Cartesian coordinates from a Gamess output file 


Note all structures are read by default. To read only the final structure, use 
the --last-structure flag 


Requires:&nbsp;&nbsp; filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or zero if read failed. <br>

### <a name="anchor604"></a>Read_gaussian_cart             
/vcp1/people/david/bin/Silico1.15/lib/silico_gaussian.pm 


Simplistic routine to extract Cartesian coordinates from a Gaussian (98) 
output file using the standard orientation 


Note all structures are read by default. To read only the final structure, use 
the --last-structure flag 


Requires:&nbsp;&nbsp; filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or zero if read failed. <br>

### <a name="anchor605"></a>Read_gromacs_gro               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read a gromacs .gro coordinate file 


Requires:&nbsp;&nbsp; filename, undef, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble <br>

### <a name="anchor606"></a>Read_gromacs_gro_molecule      
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read in a single molecule record from a gromacs gro file. 


Requires;&nbsp;&nbsp; file_record, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor607"></a>Read_gromacs_gro_single        
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read in a single molecule record from a gromacs gro file. 


Requires:&nbsp;&nbsp; file record, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor608"></a>Read_gromacs_indexfile         
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read a gromacs indexfile 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Molecule index <br>

### <a name="anchor609"></a>Read_gromacs_itp               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read a Gromacs .itp or .top topology file 


See read_gromacs_top 


### <a name="anchor610"></a>Read_gromacs_mdp               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read an mdp file 


Requires:&nbsp;&nbsp; Filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Hash with key value pairs <br>

### <a name="anchor611"></a>Read_gromacs_rtp               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read a gromacs .rtp residue type file 


Requires:&nbsp;&nbsp; filename, undef, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if failed <br>

### <a name="anchor612"></a>Read_gromacs_top               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read a Gromacs .itp or .top topology file Not all block types are read (yet) 


Note: data read from the system-wide blocks ('defaults', 'atomtypes', etc.) 
are stored in the GMX_TOPDATA. This information is also linked to SDF_DATA in 
the ensemble 


Requires:&nbsp;&nbsp; inputfile, undef, undef, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor613"></a>Read_gromacs_trr               
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read a gromacs .trr or .xtc file using a wrapper script with trjconv 


Requires:&nbsp;&nbsp; filename(trr), filename(pdb,gro,etc), undef, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if failed <br>

### <a name="anchor614"></a>Read_maestro                   
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Schroedinger maestro read routine. 


Requires:&nbsp;&nbsp; filename, start molecule (currently ignored), max molecules, option string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if failed <br>

Options: Returns an empty array when there are no more structures. 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will respond to a -me (maximum energy) and max molecule (-ms) flags  <br>

### <a name="anchor615"></a>Read_maestro_single            
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Schroedinger maestro read routine. 


Requires:&nbsp;&nbsp; filehandle, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

Options: 


### <a name="anchor616"></a>Read_merck                     
/vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm 


Read in Merck format file 


Requires:&nbsp;&nbsp; filename, unused, unused, , options string; <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble <br>

### <a name="anchor617"></a>Read_mmod                      
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Macromodel read routine. 


Can read Macromodel 'short' text format. 


Requires:&nbsp;&nbsp; filename, start molecule (currently ignored), max molecules, option string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if failed <br>

### <a name="anchor618"></a>Read_mmod_mmo                  
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Read Macromodel .mmo energy file 


Requires:&nbsp;&nbsp; filename, <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if failed <br>

### <a name="anchor619"></a>Read_mol2                      
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in a mol2 file and return as an ensemble 


Reads multiple structures from a single file. Information is currently read 
from the MOLECULE, ATOM, BOND, SUBSTRUCTURE and SET records. Only static atom 
sets are supported. Comment lines preceeding the molecule data are preserved. 
Other records are ignored. 


DOCK binding energies are extracted from the Comment lines and stored as 
$mol->{DOCK_ENERGY} 


Requires:&nbsp;&nbsp; Filename, (optional), start record, maximum number of records, options string . <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if no molecule is read. <br>

Options: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will respond to a -me (maximum energy) flag  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print 'Reading' line  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOSORT - do not sort atoms  <br>

### <a name="anchor620"></a>Read_mol2_single               
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Read in a single record from a mol2 file. 


See read_mol2 for general description 


Requires:&nbsp;&nbsp; file record, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor621"></a>Read_mol_any                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Read in any supported molecule file type. 


Uses the extension to determine file type. 


Reads following extensions: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.arc (Mopac archive)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.gro (Gromacs)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.dat/.out (will determine the difference between Macromdel, Gaussian and Mopac files)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.mae,.maegz Schroedinger maestro formats  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.mol2 (Sybyl)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.mrk (Merck format used with MMFF forcefield in CHARMm)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.rtf (CHARMm topology definition format)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.sdf .mol .rdf (MDL format)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.pdb .ent .coor (text format from NAMD)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.psf (CHARMm connectivity file) format  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.xyz (Tinker or other xyz)  <br>

Will also handle gzipped (.gz), bzip2'd (.bz2) and compressed (.Z) versions of 
these files 


Requires:&nbsp;&nbsp; file name. Optionally can pass start structure, max molecules, and option string to read routines although not all routines support 'start structure' and 'max molecules' <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed. <br>

### <a name="anchor622"></a>Read_mol_single                
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Read a single molecule 


Uses the extension to determine file type. 


Requires:&nbsp;&nbsp; file record, filename (optional), options, start structure (optional), max molecules (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor623"></a>Read_mopac_cart                
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Read in a Cartesian mopac .dat, .arc or .out file 


Requires:&nbsp;&nbsp; filename, (optional) unused, unused, options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or zero if read failed. <br>

### <a name="anchor624"></a>Read_mopac_dat_single          
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Read a mopac .dat file 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor625"></a>Read_packings                  
/vcp1/people/david/bin/Silico1.15/bin/micelle 


### <a name="anchor626"></a>Read_pdb                       
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in a pdb file. 


Reads pdb files containing multiple MODEL records. Each is read into a 
separate molecule file 


Similarly, reads pdb files with multiple molecules separated by END records 
(as produced by VMD) 


Reads Autodock output files (this is now old and has not been checked 
recently) 


The $mol->{HAS_BAD_BONDORDERS} is set 


Saves some information from the PDB header records. This is these are: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;HEADER  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;COMPND  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;HETNAM  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;USER (Autodock energies)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;REMARK (PDB_VERSION, PDB_RESOLUTION, PDB_EXPT_TYPE)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;JRNL (PDB_JRNL_AUTH, PDB_JRNL_TITL, PDB_JRNL_REF)  <br>

Reads CRYST records and saves in Silico format 


Assumes that atom names starting with Q are pseudo atoms. These atoms are set 
to dummy (Du) type. 


The PDB file is read in two passes. First to get all non atom information 
(header info and CONECT records) then to read atoms. Header and CONECT record 
information is read until an END record is encountered. It is then copied into 
all subsequent molecules. This lets files containing MODELS be read. 


Requires:&nbsp;&nbsp; Filename, start structure (numbered from 0), max_molecules, option string <br>

Options: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print 'Reading' line or warnings  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DELALT - delete alternate atoms  <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ensemble containing molecule or undef if file open fails <br>

### <a name="anchor627"></a>Read_pdb_atom                  
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in an ATOM line from a pdb file. 


Requires;&nbsp;&nbsp; line, molecule, quiet flag, residue length (default 3) <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; new atom <br>

### <a name="anchor628"></a>Read_pdb_atom_fast             
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in an ATOM line from a pdb file, getting only some fields. 


Requires:&nbsp;&nbsp; line, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; An atom <br>

### <a name="anchor629"></a>Read_pdb_compnd                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB compnd line 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor630"></a>Read_pdb_conect                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in a CONECT line from a pdb file and save it in mol->{PDB_CONECT} 


Requires;&nbsp;&nbsp; line, molecule <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; mol-&gt;{PDB_CONECT} <br>

### <a name="anchor631"></a>Read_pdb_connected             
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in datafile of amino acid connected atoms 


Requires; optional datafile (Default datafile is 'amino_acid_atoms.dat' which 
is found in the SILICO_DATA directory) 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>
Sets&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::pdb_connected and $Silico::pdb_bondorder <br>

### <a name="anchor632"></a>Read_pdb_cryst                 
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB CRYST line 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor633"></a>Read_pdb_expdta                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB EXPDTA line 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor634"></a>Read_pdb_fast                  
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in a pdb file. 


Is able to read files containing multiple MODEL records and Autodock output 
files. Sets the $mol->{HAS_BAD_BONDORDERS} flag 


Also saves some information from the PDB header records these are: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;HEADER  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;COMPND  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;HETNAM  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;USER  <br>

Assumes that atom names starting with Q are pseudo atoms. These atoms are set 
to dummy (Du) type. 


Requires:&nbsp;&nbsp; Filename, unused, unused, option string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ensemble containing molecule or undef if file open fails <br>

### <a name="anchor635"></a>Read_pdb_header_line           
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB HEADER line 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor636"></a>Read_pdb_hetnam                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB HETNAM line 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor637"></a>Read_pdb_jrnl                  
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB JRNL record 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor638"></a>Read_pdb_molecule              
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read a single PDB molecule up to and END or ENDMDL record 


Requires:&nbsp;&nbsp; file record, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule, undef if end of file <br>

### <a name="anchor639"></a>Read_pdb_remark                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB REMARK record 


Extracts:&nbsp;&nbsp; PDB file version, Resolution and experiment type <br>
Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor640"></a>Read_pdb_single                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read in a single record from a pdb file. 


Requires:&nbsp;&nbsp; file record, options <br>

See read_pdb for general description 


Options: 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor641"></a>Read_pdb_title                 
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read PDB TITLE record 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor642"></a>Read_pdb_user                  
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Read USER record from pdb file 


Requires;&nbsp;&nbsp; line, molecule <br>

### <a name="anchor643"></a>Read_quick_in_single           
/vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm 


Read a quick .in file 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor644"></a>Read_quick_out                 
/vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm 


Read in a Quick out file 


Requires:&nbsp;&nbsp; filename, (optional) unused, unused, options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if read failed. <br>

### <a name="anchor645"></a>Read_rdf                       
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Read an RDF file. 


Only reads structure at present (no data). Sets the 3D flag on all structures 


Requires:&nbsp;&nbsp; Filename - optional: start structure number (counting from 1), number of structures to read. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if file open failed. <br>

### <a name="anchor646"></a>Read_rdf                       
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Read an RDF file. 


Only reads structure at present (no data). Sets the 3D flag on all structures 


Requires:&nbsp;&nbsp; Filename - optional: start structure number (counting from 1), number of structures to read. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if file open failed. <br>

### <a name="anchor647"></a>Read_rdf_input_file            
/vcp1/people/david/bin/Silico1.15/bin/rdf 


A subroutine to read the contents of an RDF input file, and put them into a 
hash 


Passed&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; variables: A file name ($file) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; An input hash ($input); nothing (undef) if an error occurred <br>

### <a name="anchor648"></a>Read_rtp_single                
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Read in a single record from an rtp file. 


Requires:&nbsp;&nbsp; file record, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor649"></a>Read_sdf                       
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Read an SDF file. 


SDF data fields are stored in mol->{SDF_DATA}{<propertyname>} 


Requires:&nbsp;&nbsp; Filename - optional: start structure number (counting from 1), number of structures to read, options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ensemble or undef if file open failed. <br>

Options: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print messages  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOPARSE - only read SDF data fields. Do not parse molecule (Faster if we are only interested in the data fields)  <br>

### <a name="anchor650"></a>Read_sdf                       
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Read an SDF file. 


SDF data fields are stored in mol->{SDF_DATA}{<propertyname>} 


Requires:&nbsp;&nbsp; Filename - optional: start structure number (counting from 1), number of structures to read, options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Ensemble or undef if file open failed. <br>

Options: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print messages  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOPARSE - only read SDF data fields. Do not parse molecule (Faster if we are only interested in the data fields)  <br>

### <a name="anchor651"></a>Read_sdf_atoms_bonds           
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Read in atoms and bonds in MDL mol format V2000 or V3000 formats 


Called from read_sdf and read_rdf. Sets flag $mol->{HAS_BAD_GEOM}, 
$mol->{GEOM_2D} to 1 if molecule is 2D 


Requires:&nbsp;&nbsp; file record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Molecule, undef if end of file <br>

### <a name="anchor652"></a>Read_sdf_atoms_bonds           
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Read in atoms and bonds in MDL mol format V2000 or V3000 formats 


Called from read_sdf and read_rdf. Sets flag $mol->{HAS_BAD_GEOM}, 
$mol->{GEOM_2D} to 1 if molecule is 2D 


Requires:&nbsp;&nbsp; file record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Molecule, undef if end of file <br>

### <a name="anchor653"></a>Read_sdf_single                
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Read in a single record from an sdffile. 


Requires:&nbsp;&nbsp; file record, options <br>

See read_pdb for general description 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor654"></a>Read_sdf_single                
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Read in a single record from an sdffile. 


Requires:&nbsp;&nbsp; file record, options <br>

See read_pdb for general description 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor655"></a>Read_seq_any                   
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Read any sequence format 


Requires:&nbsp;&nbsp; filename, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; sequences <br>

### <a name="anchor656"></a>Read_sleuth_abstract           
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Read a SLEUTH abstract file 


Requires:&nbsp;&nbsp; input file <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; sequence record or undef if file open failed otherwise returns 1. <br>

### <a name="anchor657"></a>Read_smiles                    
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Simple routine to read smiles strings (one per line) from file and return as 
an ensemble 


Reads multiple structures from a single file. Information 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print 'Reading' line  <br>

### <a name="anchor658"></a>Read_smiles_single             
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Read in a single smiles record from file. 


Requires:&nbsp;&nbsp; file record, options <br>

See read_pdb for general description 


Options: 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor659"></a>Read_table_delimited           
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Read a character delimited text file 


Creates a data structure that looks like: 


$data-&gt;{VALUES}[&lt;linennum&gt;][&lt;columnnum&gt;] Values in array <br>
$data-&gt;{HEADERS}[&lt;columnnum&gt;] Array of all column headers <br>
$data-&gt;{KEYHASH}{} Hash of all column headers to alow translation to column numbers <br>
$data-&gt;{NUMLINES} Total number of lines <br>
$data-&gt;{NUMKEYS} Total number of columns <br>
Requires:&nbsp;&nbsp; filename, delimiter (strings 'space', 'tab', or 'comma' with the default 'tab'), has_header flag, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; data structure or undef if file open failed <br>

Options: 


QUIET&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; do not print messages <br>
HEADER&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; first line is a header line <br>
CONVERT EMPTY convert empty values to '' (not undef) <br>
CASE INSENSITIVE convert all headers to upper case <br>

### <a name="anchor660"></a>Read_tcoffee_score             
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Read in tcoffee score file 


Requires:&nbsp;&nbsp; Filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Sequences record <br>

### <a name="anchor661"></a>Read_textfile                  
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Read a file into a string 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; text <br>

### <a name="anchor662"></a>Read_tinker_xyz                
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Tinker read routine. 


Requires:&nbsp;&nbsp; filename, (optional) start, max molecules, options string, Tinker format flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if failed <br>

### <a name="anchor663"></a>Read_tinker_xyz_single         
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Read in a single record from a Tinker xyz file 


Tinker format has no comment line (comments are on first line after number of 
atoms) 


Requires:&nbsp;&nbsp; file record, options, flag to read Tinker format xyz <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor664"></a>Read_xyz                       
/vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm 


XYZ read routine. 


Requires:&nbsp;&nbsp; filename, (optional) start, max molecules, options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ensemble or undef if failed <br>

### <a name="anchor665"></a>Read_xyz_single                
/vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm 


Read in a single record from an xyz file 


Tinker format has no comment line (comments are on first line after number of 
atoms) 


Requires:&nbsp;&nbsp; file record, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef <br>

### <a name="anchor666"></a>Record_best_coords             
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Records the current coordinates of each atom as being the best ones. 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor667"></a>Remove_aromatic_bondorders     
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Convert aromatic bondorders in carboxylate and guanidinium groups to single 
and double bonds 


Requires:&nbsp;&nbsp; molecule <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef on failure <br>

### <a name="anchor668"></a>Remove_duplicates              
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Remove duplicate entries in a list. 


Requires:&nbsp;&nbsp; list <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pruned list <br>

### <a name="anchor669"></a>Remove_option                  
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Delete a text string from the list of command-line options 


Requires:&nbsp;&nbsp; Option to remove (string), list of existing options (string) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Modifies string provided to function <br>

### <a name="anchor670"></a>Reos_filter                    
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Implements a simple filter for the Rapid Elimination of Swill 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor671"></a>Rescue_unbonded_hydrogens      
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Fix unbonded hydrogens 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; total number of bonds added <br>

### <a name="anchor672"></a>Residue_printout               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Print out all seq_residue information 


Requires;&nbsp;&nbsp; residue <br>

### <a name="anchor673"></a>Residues                       
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Return array of residues from a sequence record 


Requires:&nbsp;&nbsp; Silico sequence record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of residues <br>

### <a name="anchor674"></a>Revert_to_best_coords          
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Sets atoms back to its best known position. 


Requires:&nbsp;&nbsp; Molecule, atom list (optional) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor675"></a>Rgmx_setup                     
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Set up flags for Gromacs jobs 


Requires:&nbsp;&nbsp; Nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing. Sets flags <br>

### <a name="anchor676"></a>Rms                            
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the rms distance between two molecules (heavy atoms by default). 


Note. Not periodic 


Requires:&nbsp;&nbsp; Two molecules, flag to only calculate for heavy atoms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; rms or -ve numbers if something is wrong. <br>

### <a name="anchor677"></a>Rms_atomlist                   
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the rms distance between atomlists. 


Note: Not periodic 


Requires:&nbsp;&nbsp; Two atomlists, flag to return sum of squares rather than rms <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; rms or -ve numbers if something is wrong. <br>

### <a name="anchor678"></a>Rms_fp                         
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate the fragment-based RMS for two molecules 


Requires:&nbsp;&nbsp; molecule1, molecule2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; best fragment rms or -1 if no fit is found <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; {SDF_DATA}{RMS_FP_AV} {SDF_DATA}{MATCH_NAME} ; <br>

### <a name="anchor679"></a>Rotate_torsion_angle           
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotate a torsion angle 


Will do nothing if bond is in a ring 


The static atoms are overridden if an ANCHOR atom is set. 


Requires:&nbsp;&nbsp; molecule, 2 atoms (first one static), angle, anchor (optional), flag to rotate atoms already marked with PFLAG <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor680"></a>Rotate_torsion_angle2          
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Rotate a torsion angle 


Will do nothing if bond is in a ring 


Requires:&nbsp;&nbsp; molecule, vector atom1, vector_atom2, list of atoms to be rotated, angle in radians <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor681"></a>Run_gmx_local                  
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Run a gromacs job on local workstation cluster 


Requires:&nbsp;&nbsp; gromacs exec command, output file base name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor682"></a>Run_gmx_local                  
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Run a gromacs job on local workstation cluster 


Requires:&nbsp;&nbsp; gromacs exec command, output file base name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor683"></a>Run_gmx_slurm_multi            
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Run a gromacs job on Slurm cluster. All jobs in one batch submission) 


Requires:&nbsp;&nbsp; gromacs exec command, output file base name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor684"></a>Run_gmx_slurm_multi            
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Run a gromacs job on a slurm cluster. All jobs in one batch submission) 


Requires:&nbsp;&nbsp; gromacs exec command, output file base name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor685"></a>Run_gmx_slurm_single           
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Run a gromacs job on a slurm cluster, one slurm job per gmx job) 


Requires:&nbsp;&nbsp; gromacs exectue command, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor686"></a>Run_gmx_slurm_single           
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Run a gromacs job on a Slurm cluster, one slurm job per gmx job) 


Requires:&nbsp;&nbsp; gromacs exectue command, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor687"></a>Run_gnuplot                    
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Plots&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; multiple data columns on a single graph <br>
Requires:&nbsp;&nbsp; Gnuplot tab delimited data, output filename, optional title <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor688"></a>Run_gnuplot1                   
/vcp1/people/david/bin/Silico1.15/lib/silico_gnuplot.pm 


Plots multiple data columns on separate graphs 


Requires:&nbsp;&nbsp; Gnuplot tab delimited data, optional title <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor689"></a>Run_gnuplot2                   
/vcp1/people/david/bin/Silico1.15/lib/silico_gnuplot.pm 


Plots multiple data columns on a single graph 


Requires:&nbsp;&nbsp; Gnuplot tab delimited data, optional title <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor690"></a>Run_gromacs                    
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Run a gromacs job 


Requires:&nbsp;&nbsp; gromacs exec command, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor691"></a>Run_gromacs                    
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Run a gromacs job 


Requires:&nbsp;&nbsp; gromacs exec command, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor692"></a>Run_grompp_gmx_local           
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Run a grompp then gromacs job on local workstation cluster 


Requires:&nbsp;&nbsp; grompp command, gromacs comand, postprocess command, output file base name, hostlist, flag run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor693"></a>Run_grompp_gmx_local           
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Run a grompp then gromacs job on local workstation cluster 


Requires:&nbsp;&nbsp; grompp command, gromacs comand, postprocess command, output file base name, hostlist, flag run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor694"></a>Run_grompp_gmx_slurm_single    
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Run a gromp && gromacs job on slurm cluster, one slurm job per gmx job) 


Requires:&nbsp;&nbsp; gromacs exectue command, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor695"></a>Run_grompp_gmx_slurm_single    
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Run a gromp && gromacs job on a slurm cluster, one slurm job per gmx job) 


Requires:&nbsp;&nbsp; gromacs exectue command, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor696"></a>Run_grompp_gromacs             
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Run a grompp & gromacs job 


Requires:&nbsp;&nbsp; grommpp exe cmd, gromacs exec command, postprocess cmd, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor697"></a>Run_grompp_gromacs             
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Run a grompp & gromacs job 


Requires:&nbsp;&nbsp; grommpp exe cmd, gromacs exec command, postprocess cmd, output file base name, list of hosts (optional), flag to run job in background <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; job identifier <br>

### <a name="anchor698"></a>Run_mopac                      
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Run mopac minimisation or mopac ESP charge calculation on molecule 


MOPAC keywords should be in $mol->{KEYWORDS} 


Requires:&nbsp;&nbsp; Molecule, filebase, location of mopac_exe, flag to not clean up mopac files after running <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>

### <a name="anchor699"></a>Run_quick_batch                
/vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm 


Run quick minimisation or quick ESP charge calculation on molecule. Jobs are 
submitted as a batch 


Keywords should be in $mol->{SDF_DATA}{KEYWORDS} 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef if failed <br>

### <a name="anchor700"></a>Run_ssh                        
/vcp1/people/david/bin/Silico1.15/lib/silico_schedule.pm 


Run a command by ssh 


Command must be a single commad (i.e. no semicolons) 


Requires:&nbsp;&nbsp; host, commmand, flag to put job in background, flag to print output from command, flag to supress error message <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; output from command (if background flag not set). Undef if command failed <br>

### <a name="anchor701"></a>Same_substructure              
/vcp1/people/david/bin/Silico1.15/lib/silico_residue.pm 


Check if two atoms are in the same substructure. Looks at SUBNAME, SUBID, 
SEGID and CHAIN. 


Requires:&nbsp;&nbsp; Atom 1, atom 2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if they are, 0 otherwise. <br>

### <a name="anchor702"></a>Sanitise_hydrogens             
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Make sure that every hydrogen has the same residue name, number, segid and 
chain as the parent atom 


Requires:&nbsp;&nbsp; molecule <br>

### <a name="anchor703"></a>Sanitise_segid_chain           
/vcp1/people/david/bin/Silico1.15/bin/find_amino_acids 


Make sure that the chain is sane and delete SEGID 


### <a name="anchor704"></a>Scale_vector                   
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Multiply a vector by a scalar 


Requires:&nbsp;&nbsp; scalar, vector <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector <br>

### <a name="anchor705"></a>Schrodinger_wait_for_license_availability 
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Wait until there a licenses available 


Requires:&nbsp;&nbsp; program_name, quiet flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor706"></a>Schrodinger_wait_for_license_use 
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Wait until requested license has been used 


Requires:&nbsp;&nbsp; program_name, quiet flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor707"></a>Score_ring_het                 
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Score a heterocyclic ring so that it can be sorted into a canonical order 


Score is a string made of atom element_numbers printed as C = 006, O = 008 
etc. followed by a string of bondorders. Using this system, benzene is 
represented as '006006006006006006444444'. 


With added bridgeheads! 


Requires:&nbsp;&nbsp; array of ring atom numbers, molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; score string <br>

### <a name="anchor708"></a>Sdf_calc_atom_parity_error     
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Check that the chirality (parity) flags in an sdf file match the actual 
stereochemistry in the file. 


Requires:&nbsp;&nbsp; molecule, optional flag to mark failed atoms as Du <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; negative of the number of errors in parity <br>

### <a name="anchor709"></a>Sdf_calc_atom_parity_error     
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Check that the chirality (parity) flags in an sdf file match the actual 
stereochemistry in the file. 


Requires:&nbsp;&nbsp; molecule, optional flag to mark failed atoms as Du <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; negative of the number of errors in parity <br>

### <a name="anchor710"></a>Sdf_check_chiral_parity        
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Check that the chirality (parity) flags in an sdf file match the actual 
stereochemistry in the file. 


Requires:&nbsp;&nbsp; molecule, optional flag to mark failed atoms as Du <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; negative of the number of errors in parity <br>

### <a name="anchor711"></a>Sdf_check_chiral_parity        
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Check that the chirality (parity) flags in an sdf file match the actual 
stereochemistry in the file. 


Requires:&nbsp;&nbsp; molecule, optional flag to mark failed atoms as Du <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; negative of the number of errors in parity <br>

### <a name="anchor712"></a>Sdf_data_to_atom               
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Copy data from SDF_DATA to atom records 


Assumes SDF data is in an array 


Somewhat dangerous 


### <a name="anchor713"></a>Sdf_data_to_atom               
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Copy data from SDF_DATA to atom records 


Assumes SDF data is in an array 


Somewhat dangerous 


### <a name="anchor714"></a>Sdf_data_to_table_delimited    
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Copy ensemble sdf_data to table format 


Requires:&nbsp;&nbsp; Ensemble <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Table data structure <br>

### <a name="anchor715"></a>Sdf_error_mol                  
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Produce an empty molecule as a result of a read error 


### <a name="anchor716"></a>Sdf_error_mol                  
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Produce an empty molecule as a result of a read error 


### <a name="anchor717"></a>Sdf_fastsplit                  
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Split an sdf file into individual molecules without parsing the file 


Requires:&nbsp;&nbsp; input filename, output directory (optional), output style (see subroutine format_oname), max number of structures per output file, name of sdf field to rename structures, name of field to split structures, number of files to output <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; list of files created <br>

### <a name="anchor718"></a>Sdf_fastsplit_readmol          
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Read an individual SDF molecule from INFILE 


Requires:&nbsp;&nbsp; reanme field, split_field <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; text lines of molecule, value of 'rename field' from SDF data, value of 'split field' <br>

### <a name="anchor719"></a>Sdf_fastsplit_writemol         
/vcp1/people/david/bin/Silico1.15/lib/silico_split.pm 


Write an individual SDF molecule 


Requires:&nbsp;&nbsp; filebase, dir, output name format, text lines, filecount, name, filenames array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor720"></a>Select_pareto_front            
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Select molecules using a Pareto front approach 


Molecules are assigned a Pareto Rank depending on the number of dominating 
individuals (Method of Fonseca and Fleming). 


Requires:&nbsp;&nbsp; ensemble, array of SDF fields to be minimised, array of SDF fields to be maximised, approx number of compounds to retain (0 to retain all), name the PARETO_RANK field, quiet flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ranked ensemble <br>

### <a name="anchor721"></a>Select_residues                
/vcp1/people/david/bin/Silico1.15/bin/find_aggregate 


Select important residues 


Submolecules are identified by connectivity. Waters are excluded 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; array of residues, pseudo molecule containing all interesting atoms <br>

### <a name="anchor722"></a>Select_rotate_bond             
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Select a random rotatable bond from a list of atoms 


Requires:&nbsp;&nbsp; molecule, atom list (uses heavy atoms if this is undefined) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; atom1, atom2; undef if failed to find a rotatable bond <br>

### <a name="anchor723"></a>Seq_fix_residue_numbers        
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Make sure residue numbers and the sequence are consistent. 


Renumbers residues from 1 if any residue numbers are undefined. 


Adds gaps to sequences if any residue numbers are missing 


Requires:&nbsp;&nbsp; sequence <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor724"></a>Seq_label_by_field             
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Label molecule atoms in target sequence according to properties of property 
sequence using a definable mapping 


Requires:&nbsp;&nbsp; target sequene, property sequence, field, mappiing (hash containing value, label pairs Eg clustal mapping is (KEY, VALUE) *, CLUSTAL_CONS; :,CLUSTAL_STRONG; ., CLUSTAL_WEAK <br>

Target and property sequences can be identical 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor725"></a>Seq_label_by_homology          
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Label residues by homology 


Requires that sequences_consensus & map sequence have been run first 


Percentage conservation is marked with the flags: H_CONS, H_HIGH H_MED, H_LOW 
and H_VLOW. The occupancy of PDB files set to the fraction conservation 
(fraction of sequences with the most popular amino acid). The value 
$atom->{CONSERV} is also set to the fraction conservation 


Requires:&nbsp;&nbsp; Sequence molecule, Consensus sequence <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor726"></a>Seq_label_indels               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Label sequence insertion and deletion points 


Requires that sequences_consensus & map sequence have been run first 


Insertions and deletions is marked as the sets: INS and DEL 


Requires:&nbsp;&nbsp; Sequence molecule, Consensus sequence <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor727"></a>Seq_mark_start_end             
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Mark gaps at the start and end of as sequence with '+' 


Operates on non-consensus sequences 


### <a name="anchor728"></a>Seq_merge                      
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Merge fields from sequence1 into sequence2 


Requires;&nbsp;&nbsp; sequence1, sequence2, fields to merge <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor729"></a>Seq_printout                   
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Print out a single sequence record and sequence information 


Requires:&nbsp;&nbsp; Single sequence record, option string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor730"></a>Seq_printout_header            
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Print out data from sequence hash except the actual SEQ records 


Requires:&nbsp;&nbsp; sequence <br>

### <a name="anchor731"></a>Seq_printout_one               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Print out an AA sequence in one letter code 


Requires:&nbsp;&nbsp; Single sequence record, number of AAs per line (opt), number of AAs per group (opt) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor732"></a>Seq_printout_three             
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Print out an AA sequence in three letter code 


Requires:&nbsp;&nbsp; Single sequence record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor733"></a>Seq_read_error                 
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Print an error on failing to read sequence data from a file 


Requires:&nbsp;&nbsp; filename, flag to write error message and die <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor734"></a>Seq_residue_fields             
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Get all residue fields of a sequence 


Requires:&nbsp;&nbsp; sequence <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; fields as an array <br>

### <a name="anchor735"></a>Seq_string                     
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Return a sequence as a string of one-letter sequence 


Requires:&nbsp;&nbsp; sequence, flag to delete gaps <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor736"></a>Sequences_STD                  
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


## Select only amino acid sequences from a set of sequences 


Requires:&nbsp;&nbsp; sequences <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; sequences <br>

### <a name="anchor737"></a>Sequences_STD_nonSTD           
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


## Select both aa and non-amino acid sequences from a set of sequences 


Requires:&nbsp;&nbsp; sequences <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; standard sequences, non aa sequences <br>

### <a name="anchor738"></a>Sequences_clustal_homology     
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Calculate the homology of a set of sequences according to the clustal scheme 


Sets the 'CONS' field to values of ~, ., : or * for each AA 


Nonstandard sequences (ie seq->{TYPE} ne 'STD') and sequences with names 
starting with 


an underscore are ignored 


Requires:&nbsp;&nbsp; Sequences <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; a sequence containing clustal markers of homology <br>

### <a name="anchor739"></a>Sequences_consensus            
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Calculate a consensus sequences and other stats 


Nonstandard sequences (ie seq->{TYPE} ne 'STD) and sequences with names 
starting with an underscore are ignored 


Requires:&nbsp;&nbsp; Sequences <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; consensus sequence record <br>

### <a name="anchor740"></a>Sequences_get_column           
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Return one column from a set of sequences 


Requires:&nbsp;&nbsp; sequence, column number <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; pointer to array of aas <br>

### <a name="anchor741"></a>Sequences_length               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Find longest sequence in a set of sequences 


Requires:&nbsp;&nbsp; sequences <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; length of longest array indexed from 1 <br>

### <a name="anchor742"></a>Sequences_nonSTD               
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


## Select only non-amino acid sequences from a set of sequences 


Requires:&nbsp;&nbsp; sequences <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; sequences <br>

### <a name="anchor743"></a>Sequences_printout             
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Print out a set of sequences using one letter code 


Requires:&nbsp;&nbsp; sequences, options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor744"></a>Sequences_read_mols            
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Read in any molecule files that correspond to sequence names 


Only considers pdb and mol2 files 


Requires:&nbsp;&nbsp; sequences, flag to map residues in molecule to sequence (on by default) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor745"></a>Sequences_write_mols           
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Write any molecule files attached to sequences 


### <a name="anchor746"></a>Set_acidbase                   
/vcp1/people/david/bin/Silico1.15/lib/silico_hydrogens.pm 


Set formal charges on acidic/basic atoms 


Uses the subroutine find_nearby_formal_charges to prevent setting charges near 
to atoms that are already charged 


Currently set to search five bonds away 


Requires:&nbsp;&nbsp; atom, molecule, formal charge <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if charge set, 0 if not set <br>

### <a name="anchor747"></a>Set_debg17                     
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Work out if macromodel debug command DEBG 17 should be set 


Reqires:&nbsp;&nbsp;&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor748"></a>Set_default_oappend            
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Define a default oappend string 


Requires:&nbsp;&nbsp; string <br>

### <a name="anchor749"></a>Set_flag                       
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Set a flag to a value 


Searches by short name (default) or long name (if 'l' is supplied as a second 
argument). 


Requires:&nbsp;&nbsp; flag name (string), search type (optional string), value (default 1) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor750"></a>Set_flag_from_arguments        
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Set flag values from arguments on the command line 


Flag and value are removed from the Silico::arguments array 


Requires:&nbsp;&nbsp; flag record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor751"></a>Set_lflag                      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Set flag by long name 


Requires:&nbsp;&nbsp; flag name, value <br>

Calls set_flag 


### <a name="anchor752"></a>Set_sflag                      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Set flag by short name 


Requires:&nbsp;&nbsp; flag name, value <br>

Calls set_flag 


### <a name="anchor753"></a>Set_torsion_angle              
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Set a torsion angle to a particular angle 


Will rotate whole molecule if bond is in a ring 


The static atoms are overridden if an ANCHOR atom is set. 


Requires:&nbsp;&nbsp; Molecule, 4 atoms (first two static), angle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor754"></a>Setup_flags                    
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Initiate flag by checking for menu or help options 


### <a name="anchor755"></a>Setup_fp_flags                 
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Make flags required for fingerprints 


Requires: max fragment size (default 8), min fragment size (default 3) 


Returns: nothing 


### <a name="anchor756"></a>Setup_mopac_flags              
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Setup mopac flags 


Requires:&nbsp;&nbsp; Nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Mopac keywords <br>

### <a name="anchor757"></a>Setup_quick_flags              
/vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm 


Setup Quick flags 


Requires:&nbsp;&nbsp; Nothing <br>
Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; quick flags <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; quick keywords <br>

### <a name="anchor758"></a>Shared_rings                   
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Counts the number of common rings a pair of atoms share. 


Requires:&nbsp;&nbsp; Molecule, atom 1, atom 2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number of shared rings <br>

### <a name="anchor759"></a>Silico_msg                     
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Print out a note, warning, error or fatal-error message. 


Types are c, comment etc, g debug etc etc 


Requires:&nbsp;&nbsp; Type (single-character string), strings <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor760"></a>Silico_putline                 
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Put a line back on 'the stack' 


Requires:&nbsp;&nbsp; line, file record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor761"></a>Silico_readline                
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Read a line from $fr->{FH}. Remove carriage returns and line feeds 


Lines can be stored in var $fr->{READLINE} 


$fr->{END} is set if the end of the file is reached. 


Requires:&nbsp;&nbsp; file record <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; processed line <br>

### <a name="anchor762"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/bilayer_builder 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor763"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/checkbox 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor764"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/chop_cell 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor765"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/cluster 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor766"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/dist_filter 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor767"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor768"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/ens_randomise_order 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor769"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/ens_sort 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor770"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/extract_lig_prot 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor771"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/file_rename 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor772"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_aggregate 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor773"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_clash 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor774"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_close_atoms 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor775"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_close_water 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor776"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_groups 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor777"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_ligand_contacts 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor778"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_rings 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor779"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/find_similar 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor780"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/fix_cis_amides 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor781"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/formatdoc 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor782"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/make_box 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor783"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/make_index 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor784"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/measure_dist 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor785"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/micelle 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor786"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mmod_csearch 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor787"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mmod_min 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor788"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mmod_qikprop 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor789"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol2seq 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor790"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol2split 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor791"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_add_h 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor792"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_add_segid 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor793"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_amides 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor794"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_backbone 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor795"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_boltz_pop 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor796"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_bond 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor797"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_centre 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor798"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_centroid 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor799"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_characterise 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor800"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_charge 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor801"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_check 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor802"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_combine 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor803"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_del_atoms 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor804"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_del_dummy 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor805"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_del_h 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor806"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_del_solvent 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor807"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_divide 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor808"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_ensemble_average 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor809"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_extents 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor810"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_filter 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor811"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_flatten 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor812"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_hydrogen_bonds 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor813"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_label_fg 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor814"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_mw 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor815"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_orient 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor816"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_rama 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor817"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_rename 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor818"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_renumber 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor819"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_rescale_bonds 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor820"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_rot 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor821"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_rot_bond 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor822"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_rotrans 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor823"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_scale 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor824"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_segment 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor825"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_smiles 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor826"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_solvate 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor827"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_sort 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor828"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_split 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor829"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_split_segid 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor830"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_transfer_coords 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor831"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_unsplit 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor832"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/mol_wrap_cell 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor833"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/n_dist 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor834"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/pdb_chain_split 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor835"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/pdb_rename_hydrogens 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor836"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/pdbsplit 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor837"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/plot_dihedrals 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor838"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/radius_of_gyration 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor839"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/random_box 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor840"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/randomise 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor841"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/rdf 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor842"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/renumber_residues 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor843"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/residue_distance 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor844"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/residue_rename 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor845"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/rms 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor846"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/rms_pairs 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor847"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/run_mopac 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor848"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/run_quick 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor849"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/sdf_merge 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor850"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/sdfsplit 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor851"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/smooth 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor852"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor853"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/tabulate 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor854"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/vesicle_builder 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor855"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/water_to_ion 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor856"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_g03zmat 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor857"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_gro 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor858"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_itp 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor859"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_mae 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor860"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_merck 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor861"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_mmod 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor862"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_mol2 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor863"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_mopac 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor864"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_pdb 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor865"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_sdf 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor866"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_seq 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor867"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_tab 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor868"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_tinker 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor869"></a>Silico_setup                   
/vcp1/people/david/bin/Silico1.15/bin/write_xyz 


Locate the Silico libraries and read in the parent library, silico.pm 


Sets:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $Silico::home_dir, $Silico::lib_dir <br>
Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor870"></a>Silico_vdw_radii               
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Create a list of Silico van der Waals radii, one for each atom. 


This data should be used only to determine whether or not to create a bond 
when generating connectivity information. 


Some elements have their radii artifically set to zero, so as to prevent bonds 
from being created to these elements. 


The hydrogen atom's radius is reduced to 0.2 Angstroms. 


Bonds to (between?) C, H, N, O, P and S are determined elsewhere with 
reference to the expected lengths for bonds to these atoms. 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor871"></a>Simple_atomtype                
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Simple polar/nonpolar atom type routine 


### <a name="anchor872"></a>Sleep_or_stop                  
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Check for sleep and stop files 


Waits while a sleep file is present in the working directory 


Returns 1 if stop file found 


### <a name="anchor873"></a>Slurp_file                     
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Read a molecule file and split into separate molecules as TEXT files. 


Currently only splits pdb, mol2 and sdf files. All others are returned unsplit 


Requires:&nbsp;&nbsp; filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; an array containing each molecule as text <br>

### <a name="anchor874"></a>Smiles_invariant               
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Generates a SMILES invariant for each atom, so as to assist in forming Unique 
SMILES strings. 


Requires:&nbsp;&nbsp; atom, molecule, list of atoms in SMILES set <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; invariant string <br>

### <a name="anchor875"></a>Smiles_list_recursive          
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Determine the SMILES string representation for a given atom. 


Requires:&nbsp;&nbsp; atom, molecule, ... <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor876"></a>Smooth_table_delimited         
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Create running average of data in a delimited file over a window 


Window is the number of values on either side to average over. i.e. a window 
value of 1 will average over 3 values. 


### <a name="anchor877"></a>Sort_atoms_label               
/vcp1/people/david/bin/Silico1.15/bin/water_to_ion 


Routine to sort atoms using atom->{SORT_LABEL} 


Requires:&nbsp;&nbsp; molecule. <br>

### <a name="anchor878"></a>Sort_itp_by_atom_number        
/vcp1/people/david/bin/Silico1.15/bin/write_itp 


Sort GMX_BONDS GMX_PAIRS GMX_ANGLES GMX_DIHEDRALS GMX_EXCLUSIONS in ascending 
order by atom bumber 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor879"></a>Sort_itp_by_atom_number        
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Sort GMX_BONDS GMX_PAIRS GMX_ANGLES GMX_DIHEDRALS GMX_EXCLUSIONS in ascending 
order by atom bumber 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor880"></a>Sort_itp_by_smiles             
/vcp1/people/david/bin/Silico1.15/bin/write_itp 


Sort itp file by canonical smiles ascending order by atom bumber 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor881"></a>Sort_itp_by_smiles             
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Sort itp file by canonical smiles ascending order by atom bumber 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor882"></a>Space_group_pdb_to_silico      
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Turn a PDB space group string into a Silico space group code. 


This subroutine assumes that there is only one key (Silico space group code) 
for each value (PDB space group string). 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor883"></a>Space_group_silico_to_pdb      
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Turn a Silico space group code into a PDB space group string. 


Requires:&nbsp;&nbsp; string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; string <br>

### <a name="anchor884"></a>Splitline                      
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Take a line of text, clean it up and split it into an array 


Requires:&nbsp;&nbsp; Text <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Array <br>

### <a name="anchor885"></a>Standard_bond_lengths          
/vcp1/people/david/bin/Silico1.15/lib/silico_definitions.pm 


Create a hash containing standard chemical bond lengths. The hash key contains 
the bond: two atoms, between which is a character denoting the bond order. 
Dash ('-'): single bond Equals sign ('='): double bond Percent sign ("%"): 
triple bond 


At the moment, bond lengths are given twice: one for A-B, the other for B-A. 
This is because we can't be sure in advance which order the atoms will be 
given in. 


Bond lengths were sourced from 
http://chemviz.ncsa.uiuc.edu/content/doc-resources-bond.pdf, January 2007. 


Note: The bond length for Sb-Cl is for antimony trichloride, not antimony 
pentachloride. 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor886"></a>Standard_deviation_bias_corrected 
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Calculates the bias-corrected (n-1) standard deviation of a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; standard deviation (scalar quantity) <br>

### <a name="anchor887"></a>Standard_deviation_bias_uncorrected 
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Calculates the bias-uncorrected standard deviation of a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; standard deviation (scalar quantity) <br>

### <a name="anchor888"></a>Standardise_boolean            
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Turn a Silico boolean value into 1 or 0 depending on what it is. 


Requires:&nbsp;&nbsp; Text <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Number <br>

### <a name="anchor889"></a>Standardise_ring               
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Order a ring (list of atom numbers) that is in a standard 'orientation'. 


Rearrange atoms in a ring list so that atom[0] has the lowest number and 
atom[1] has a lower number than atom[-1]. 


Requires:&nbsp;&nbsp; ptr to array of atom numbers. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ptr to normalised array of atom numbers. <br>

### <a name="anchor890"></a>Standardise_ring_by_heteroatoms 
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Order a list of ring atoms so that is in a standard 'orientation' organised by 
hetero ring atoms and bondorders. 


Rearrange atoms in a ring list so that the highest atomic number has the 
lowest position in the ring with a subsequent requirement that the highest 
bondorder is at the lowest position in the ring. 


Requires:&nbsp;&nbsp; ptr to array of atom numbers. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; ptr to normalised array of atom numbers, string made of atom elements and bondorders - for example pyridine will be returned as 'N4C4C4C4C4C4'. <br>

### <a name="anchor891"></a>Stop_file_exists               
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx.pm 


Check if a 'stop' or 'stop.<jobid>' file exists 


File is removed if found 


Requires:&nbsp;&nbsp; optional die flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor892"></a>Stop_file_exists               
/vcp1/people/david/bin/Silico1.15/lib/silico_rgmx_1_1.pm 


Check if a 'stop' or 'stop.<jobid>' file exists 


File is removed if found 


Requires:&nbsp;&nbsp; optional die flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor893"></a>Submit_quick_job               
/vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm 


Submit Quick job in background 


Keywords should be in $mol->{SDF_DATA}{KEYWORDS} 


Requires:&nbsp;&nbsp; Molecule, filebase, location of quick_exe, flag to not clean up quick files after running <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; molecule or undef if failed <br>

### <a name="anchor894"></a>Submol_active_atoms            
/vcp1/people/david/bin/Silico1.15/bin/find_aggregate 


Find active atoms in submol 


### <a name="anchor895"></a>Submol_set_cluster             
/vcp1/people/david/bin/Silico1.15/bin/find_aggregate 


Assign all atoms in a submol to a cluster 


### <a name="anchor896"></a>Subname                        
/vcp1/people/david/bin/Silico1.15/lib/silico_control.pm 


Return the name of the subroutine in which this was called 


Requires:&nbsp;&nbsp; Nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Name <br>

### <a name="anchor897"></a>Sybyl_sort                     
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Ensure that molecule is written out with substructures in numerical order. 
This can cause problems with other software packages if it is not the case. 
Sybyl doesn't mind this though 


Requires:&nbsp;&nbsp; ensemble. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing. <br>

### <a name="anchor898"></a>Sybyl_spectrum                 
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Return a Sybyl colour based on a value and a maximum range 


Assumes that the value is between 0 and $range. Returns one of a spectrum of 
10 colours (RED REDORANGE ORANGE YELLOW GREEN GREENBLUE CYAN BLUE PURPLE 
VIOLET) 


Reqires:&nbsp;&nbsp;&nbsp;&nbsp; value, minimum value of colour range (default 0), maximum value of colour range (default 100) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; colour (string) <br>

### <a name="anchor899"></a>Table_delimited_addcolumn      
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Add a new column to a 'delimited' data structure 


Requires:&nbsp;&nbsp; Data structure, new key name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; new column number (zero based) <br>

### <a name="anchor900"></a>Table_delimited_getrow         
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Return a row from a delimited data structure 


Requires:&nbsp;&nbsp; Data structure, array of column names <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array or undef if finished <br>

Increments $data->{CURRENT_LINE} to keep track of row. Reset 
$data->{CURRENT_LINE} to undef to start from the beginning again 


### <a name="anchor901"></a>Table_delimited_getval         
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Return a value from the current line 


Requires:&nbsp;&nbsp; Data structure, column name <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; value <br>

Uses $data->{CURRENT_LINE} to keep track of row 


### <a name="anchor902"></a>Table_delimited_inc_row        
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Return a row from a delimited data structure 


Requires:&nbsp;&nbsp; Data structure <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; array or undef if finished <br>

Increments $data->{CURRENT_LINE} to keep track of row. Reset 
$data->{CURRENT_LINE} to undef to start from the beginning again 


### <a name="anchor903"></a>Tag_markup                     
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Convert silico markup to tags 


Tags are: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#! -&gt; &lt; HEAD# &gt;  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&gt; -&gt; &lt; STD &gt;  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#? -&gt; &lt; STD_DESC &gt;  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#; -&gt; &lt; HANG &gt;  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#: -&gt; &lt; HANG_SHORT &gt;  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#, -&gt; &lt; INDENT &gt;  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#- -&gt; &lt; LIST &gt;  <br>

Boilerplate flags #F and #SF cause the standard text to be inserted 


#INCLUDE <filename> causes the file from $SILICO_HOME/data to be inserted 


(Note that extra spaces have been added to the tags above to stop them being 
recognised by makedoc). 


The first part of a hanging indent can use underscores to represent spaces 


### <a name="anchor904"></a>Tan_substructure_frac          
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate fraction of substructure keys present in a parent molecule 


frac = Num common fragments / Num fragments in mol1 


Requires: hash of fragments 1, hash of fragments 2 


Returns: Tanimoto coefficient 


### <a name="anchor905"></a>Tanimoto                       
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate the Tanimoto (or Tversky) coefficient given two hashes of fragments. 
See http://www.chemaxon.com/jchem/doc/user/query_similarity.html#metrics 


Tanimoto coeff = Num common fragments / Num fragments in mol1 + Num fragments 
in mol2 - Num common fragments 


Tversky coeff = Num common fragments / alpha (Num fragments in mol1 - Num 
common fragments) + beta (Num fragments in mol2 - Num common fragments) + Num 
common fragments 


alpha = 0 and beta = 1 => Molecule 2 as a substructure of Molecule 1 


alpha = 1 and beta = 2 => Molecule 2 as a superstructure of Molecule 1 


Requires: hash1, hash2, (alpha, beta coefficiencts for Tversky, optional) 


Returns: Tanimoto coefficient 


### <a name="anchor906"></a>Tanimoto_freq                  
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate the Tanimoto coefficient given two hashes of fragments including 
fragment frequencies 


Requires: hash of fragments 1, hash of fragments 2 


Returns: Tanimoto coefficient 


### <a name="anchor907"></a>Tanimoto_matrix                
/vcp1/people/david/bin/Silico1.15/lib/silico_fp.pm 


Calculate a matrix of Tanimoto coefficients for two ensembles 


Requires: ensemble1, ensemble2 (optional), (alpha, beta coefficiencts for 
Tversky, optional) 


Returns: array 


### <a name="anchor908"></a>Tautomer_probability_field     
/vcp1/people/david/bin/Silico1.15/bin/dock_sort 


Find tautomer probability field 


### <a name="anchor909"></a>Test_all_shapes                
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Test whether a molecule complies with a set of geometric restrictions provided 
by flags 


Requires:&nbsp;&nbsp; molecule, cell <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor910"></a>Test_cylinder_arb              
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Test if all atoms of a molecule are INSIDE or OUTSIDE a set of cylinders lying 
along arbitrary axes 


If invert is FALSE then function returns TRUE if molecule is OUTSIDE ALL 
cylinders 


If invert is TRUE then function returns TRUE if molecule is INSIDE ANY 
cylinder 


Note the sense of this test is the opposite of the others. Probably should be 
modified. 


Requires:&nbsp;&nbsp; Molecule, cell, invert flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; True or false <br>

### <a name="anchor911"></a>Test_energy_molcount           
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Test to see if we have exceeded molecule maximum energy or maximum molecule 
count set by flags 


Requires:&nbsp;&nbsp; filehandle, molecule, max_energy (optional or from flag), max molcount (optional or from flag) <br>
Returns;&nbsp;&nbsp;&nbsp;&nbsp; true if we have exeeded counts/energy <br>

### <a name="anchor912"></a>Test_plane                     
/vcp1/people/david/bin/Silico1.15/lib/silico_rbox.pm 


Test if molecule lies completely within (or completely outside) a (set of) 
plane(s) 


Plane centres are set by by Silico:origin_points 


Plane direction default xz or xy by flag 


If invert NOT set 


Returns 1 if molecule lies completely inside any one plane. 


If invert IS set 


Returns 1 if molecule lies completely outside ALL planes 


Requires:&nbsp;&nbsp; molecule, cell, invert flag <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; true or false <br>

### <a name="anchor913"></a>Tinker_space_group             
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Convert a Silico space group (numeric code) to Tinker format based on 
Hermann-Mauguin symbols. 


Requires:&nbsp;&nbsp; Space group in Silico format <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Space group in Tinker format; undef if error <br>

### <a name="anchor914"></a>Traverse                       
/vcp1/people/david/bin/Silico1.15/lib/silico_rings.pm 


Recursive subroutine to find rings 


Called from find_atom_rings 


Requires:&nbsp;&nbsp; path <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor915"></a>Two_bond_angle                 
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Find the angle between two bonds (four atoms) 


Requires:&nbsp;&nbsp; atom1, atom2, atom3, atom4 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; angle in radians. <br>

### <a name="anchor916"></a>Two_bond_normal                
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Return a normal vector to two bonds (four atoms) 


Requires:&nbsp;&nbsp; atom1, atom2, atom3, atom4 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector (x, y, z) <br>

### <a name="anchor917"></a>Unit_normal_vector             
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Return a unit vector normal to a given vector 


Vector will also be normal to X, Y or Z axis 


Requires:&nbsp;&nbsp; vector <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector <br>

### <a name="anchor918"></a>Unit_vector                    
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Normalises a vector, providing a unit vector. 


Requires:&nbsp;&nbsp; array of three elements (x, y and z components) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; unit vector as an array <br>

### <a name="anchor919"></a>Valence                        
/vcp1/people/david/bin/Silico1.15/lib/silico_molecules.pm 


Return the valence of an atom 


Note that THIS ROUTINE CAN RETURN NON-INTEGRAL VALUES for bonds involving 
aromatic bond types (eg carboxylates or guanidines represented using aromatic 
bonds). The valence is corrected for aromatic bridgehead atoms which return a 
valence of 4 rather than 4.5 (which would result from three aromatic bonds). 


The minimum possible valence is returned for poorly defined bond types (>=5) 


Requires:&nbsp;&nbsp; atom, molecule (optional but necessary to detect lone pairs) <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; valence <br>

### <a name="anchor920"></a>Variance_bias_corrected        
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Calculates the bias-corrected (n-1) variance of a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; variance (scalar quantity) <br>

### <a name="anchor921"></a>Variance_bias_uncorrected      
/vcp1/people/david/bin/Silico1.15/lib/silico_statistics.pm 


Calculates the bias-uncorrected (n) variance of a set of data 


Requires:&nbsp;&nbsp; Array <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; variance (scalar quantity) <br>

### <a name="anchor922"></a>Vector                         
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Calculates a vector from two Cartesian points. 


Requires:&nbsp;&nbsp; coord1, coord2 <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector as an array <br>

### <a name="anchor923"></a>Vector_add                     
/vcp1/people/david/bin/Silico1.15/lib/silico_geom.pm 


Add two vectors (length 3) 


Requires:&nbsp;&nbsp; vectors, vector <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; vector <br>

### <a name="anchor924"></a>Verify_input_data              
/vcp1/people/david/bin/Silico1.15/lib/silico_data.pm 


Verifies that input data is of a specified type 


Requires:&nbsp;&nbsp; Data Expected data type (text string suitable for check_data_type) Prompt message (eg. "Timestep") Optional default value after first time <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Modifies value of Data directly <br>

### <a name="anchor925"></a>Veryquiet                      
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Check if we are being very quiet (i.e., the --very-quiet option is used). 


Requires:&nbsp;&nbsp; nothing <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 (very quiet), 0 (not very quiet) <br>

### <a name="anchor926"></a>Warn_atom_type_error           
/vcp1/people/david/bin/Silico1.15/lib/silico_prop.pm 


Add atom type warning to mol->{WARN}; 


Could be made simpler 


Requires:&nbsp;&nbsp; Molecule, atom, numsingle, numdouble, $numtriple, numatomatic, numbonds <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor927"></a>Wraphtml                       
/vcp1/people/david/bin/Silico1.15/lib/silico_doc.pm 


Add header and footer to formatted html 


### <a name="anchor928"></a>Write_asl_atoms                
/vcp1/people/david/bin/Silico1.15/lib/silico_asl.pm 


Write out atoms selected in index groups to a file 


Requires:&nbsp;&nbsp; molecule, list of index groups, filebase <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor929"></a>Write_binfile                  
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Write out a tab-delimited file of binned data 


Requires: output filename 


Returns: one or undef if file open failed. 


### <a name="anchor930"></a>Write_clustal                  
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Write a clustal format file 


Requires:&nbsp;&nbsp; Sequences, filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DELIMITER='x' Delimiter between fields  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NUMLINE=XX Print XX residues per line  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;WIDTH=XX Print XX characters per residue  <br>

### <a name="anchor931"></a>Write_divcon_cart              
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Print out a DivCon Cartesian format file 


Currently prints out only the first molecule of the ensemble 


Options:&nbsp;&nbsp;&nbsp;&nbsp; CLUSTER <br>
Flags:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; nmr (peptide, all), opt (num optimization steps) <br>
Requires:&nbsp;&nbsp; Ensemble (or molecule), filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open fails otherwise returns 1. <br>

### <a name="anchor932"></a>Write_fasta                    
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Write fasta format 


Requires:&nbsp;&nbsp; Sequences, filename, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor933"></a>Write_gaussian_cart            
/vcp1/people/david/bin/Silico1.15/lib/silico_gaussian.pm 


Print out a gaussian Cartesian format file 


Requires:&nbsp;&nbsp; ensemble (or molecule), output filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if write failed <br>

### <a name="anchor934"></a>Write_gaussian_zmatrix         
/vcp1/people/david/bin/Silico1.15/lib/silico_gaussian.pm 


Gaussian Z-matrix writer. 


Requires:&nbsp;&nbsp; ensemble or molecule, filename, forcefield file (optional) to type atoms. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed otherwise returns 1. <br>

### <a name="anchor935"></a>Write_gromacs_gro              
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Write a gro file 


Requires:&nbsp;&nbsp; Ensemble, filename, $options. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Zero if file open failed otherwise returns 1. <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;SPLIT - print out an ensemble as separate Gromacs files  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;PRECISE - increase precision of gro file to 0.001 Angstroms  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;VERYPRECISE - increase precision of gro file to 0.0001 Angstroms  <br>
Requires:&nbsp;&nbsp; Ensemble, filename, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if write failed, otherwise 1 <br>

### <a name="anchor936"></a>Write_gromacs_gro_molecule     
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Print out a single gro format molecule 


Called from write_gromacs_gro 


Requires:&nbsp;&nbsp; Molecule, optional flag to increase precision of file <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor937"></a>Write_gromacs_index_file       
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Write gromacs index file 


Requires:&nbsp;&nbsp; Output filename, hash of atomlists, array of hash keys, flag to index atom numbers from zero <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor938"></a>Write_gromacs_itp              
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Write a gromacs included topology file 


Only writes first molecule 


Requires:&nbsp;&nbsp; Ensemble (or molecule), options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if write failed, otherwise 1 <br>

### <a name="anchor939"></a>Write_gromacs_itp_mol          
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Write a gromacs included topology file 


Assumes that bonds are present and that bond orders are correct 


Data is stored in atoms (several fields), GMX_BONDS, GMX_PAIRS, GMX_ANGLES, 
GMX_DIHEDRALS and GMX_EXCLUSIONS 


Requires:&nbsp;&nbsp; file_record, molecule, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if write failed, otherwise 1 <br>
Note&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; that these subroutines use a new BONDS and ANGLES format currently called GMX_BONDS etc <br>

### <a name="anchor940"></a>Write_gromacs_top              
/vcp1/people/david/bin/Silico1.15/lib/silico_gromacs.pm 


Write a gromacs full topology file 


Requires:&nbsp;&nbsp; Ensemble (or molecule), options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if write failed, otherwise 1 <br>

### <a name="anchor941"></a>Write_mae                      
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Basic Maestro write routine. 


Requires:&nbsp;&nbsp; ensemble (or molecule), filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed otherwise returns 1. <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; GZ (writes maegz format) <br>

### <a name="anchor942"></a>Write_mae_molecule             
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write a single record from a pdb file. 


Requires:&nbsp;&nbsp; file record, molecule, options <br>

See read_pdb for general description 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef if failed <br>

### <a name="anchor943"></a>Write_maegz                    
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Basic Maestro maegz write routine (wrapper for write_mae) 


Requires:&nbsp;&nbsp; ensemble (or molecule), filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed otherwise returns 1. <br>

### <a name="anchor944"></a>Write_merck                    
/vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm 


Print out an ensemble as separate Merck molecules 


Requires:&nbsp;&nbsp; Ensemble (or molecule), filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if failed <br>

### <a name="anchor945"></a>Write_merck_mol                
/vcp1/people/david/bin/Silico1.15/lib/silico_merck.pm 


Print out a single Merck format molecule 


Currently crudely attempts to assign Merck formal charge parameters. This 
makes the assumption that all hydrogens are present and all bond valences are 
correct 


Attempts to fix valence of dataive bonds to single and use alternating double 
and single bonds for aromatic rings 


Requires:&nbsp;&nbsp; Molecule, optional output options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 <br>

### <a name="anchor946"></a>Write_mmod                     
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Macromodel write routine. 


Requires:&nbsp;&nbsp; ensemble (or molecule), filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed otherwise returns 1. <br>

### <a name="anchor947"></a>Write_mmod_com_mdyn_stage      
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write the molecular dynamics stage of a macromodel com file 


Requires:&nbsp;&nbsp; output filename, simulation time (ps), temperature (K), timestep, initial temp (opt), final temp (opt), shake flag (0 - none, 1 - hydrogens, 2 - all bonds), stochastic dynamics flag, output frequency (ps) <br>

### <a name="anchor948"></a>Write_mmod_com_mini_stage      
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write the minimisation stage of a macromodel com file 


Requires:&nbsp;&nbsp; number of steps, final gradient, macromodel minimisation method code. Flag to loop minimisation <br>

### <a name="anchor949"></a>Write_mmod_com_setup           
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write initial section of Macromodel command file for a minimisation or other 
simulation 


Requires:&nbsp;&nbsp; molecule, output filename (required), structure input filename (opt), structure output filename (opt), forcefield (opls3, opls2005, opls2001, amber94, amber, mm3, mm2), solvent (water, chloroform, octanol, none), minimisation method (macromodel codes, default 20), <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if write failed <br>

### <a name="anchor950"></a>Write_mmod_com_write_stage     
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write a macromodel 'WRIT' command to a macromodel com file 


Requires:outfile <br>

### <a name="anchor951"></a>Write_mmod_sbc                 
/vcp1/people/david/bin/Silico1.15/lib/silico_mmod.pm 


Write Macromodel constraint commands (SUBS, FXAT) to .sbc or .com files 


Uses flags $atom->{MMOD}{XXX} where XXX is SUBS, FXAT, etc 


Force constants and other MMOD parameters are stored on a per atom basis in 
$atom->{MMOD}{ZZZ} where ZZZ can be FXAT_WELL_HALF_WIDTH, FXAT_FORCECONST, 
FXAT_X, FXAT_Y, FXAT_Z 


Requires:&nbsp;&nbsp; molecule, filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if write failed <br>

### <a name="anchor952"></a>Write_mol2                     
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write an ensemble as a mol2 file. 


Carries out the following steps: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sort atoms so that substructures are written out in numerical order  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generate bonds if there seem to be too few (less than (NUMATOMS -1)/2)  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convert kekule representation of aromatic rings to aromatic type bonds  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convert guanidinium and carboxylates to aromatic type bonds  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generate @&lt;TRIPOS&gt;SUBSTRUCTURE record  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generate Sybyl atom types  <br>



Requires:&nbsp;&nbsp; ensemble (or molecule), filename, optional options string. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed or other error, otherwise 1. <br>



Options: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BOND - Force bond generation  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DIVIDE - Write each molecule to a separate file  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;FAST - Equivalent to NOSORT, NOBOND, NOTYPE. This option can be used if the file read was already a good quality mol2 file  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;FLAT - atom type molecule using routines that do not require good geometry  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;MOLMOL - produce nonstandard mol2 file that is compatible with molmol  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOSORT - do not sort atom order before writing  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOTYPE - do not perform Sybyl atom typing  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOBOND - do not create any bonds  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOSDF_DATA - do not write SDF_DATA into comments  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NO_AROM_DELOC_BONDS - do not convert carboxylate and guanidinium bonds to aromatic  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;PROTEIN - write in Sybyl protein format  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print 'Writing' line  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;SMALL - write in Sybyl small molecule format  <br>



Note that by default write_mol2 will change the order of the atoms in the 
molecule data structure. This may upset subsequent routines. 


### <a name="anchor953"></a>Write_mol2_atoms               
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write out mol2 atom records 


Requires:&nbsp;&nbsp; molecule, filehandle, options; <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor954"></a>Write_mol2_bonds               
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write a mol2 bond records 


Note. This routine converts carboxylate and guanidinium groups to aromatic 
bond order (in the output file) 


Requires:&nbsp;&nbsp; molecule, filehandle, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor955"></a>Write_mol2_comments            
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write mol2 comments 


Requires:&nbsp;&nbsp; molecule, filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor956"></a>Write_mol2_crystal             
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write out the CRYSTAL record to a mol2 file 


Requires:&nbsp;&nbsp; Molecule, filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor957"></a>Write_mol2_molecule            
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write a single molecule record to a mol2 file 


Requires:&nbsp;&nbsp; file record, molecule, options <br>

See read_mol2 for general description 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef if failed <br>

### <a name="anchor958"></a>Write_mol2_sdfdata             
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write SDF datainto mol2 comment lines 


Requires:&nbsp;&nbsp; molecule, filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor959"></a>Write_mol2_set                 
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write mol2 static sets 


Requires:&nbsp;&nbsp; molecule, filehandle <br>

### <a name="anchor960"></a>Write_mol2_substructures       
/vcp1/people/david/bin/Silico1.15/lib/silico_mol2.pm 


Write a mol2 substructure record 


Requires:&nbsp;&nbsp; molecule, filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor961"></a>Write_mol_any                  
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Write any supported molecule file type. 


Requires:&nbsp;&nbsp; ensemble (or molecule), filebase (determined by subroutine 'obase' if not specified), format (determined by subroutine 'get_oformat' if not specified), (optional) options string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed or other problem <br>

### <a name="anchor962"></a>Write_mol_any_check            
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Check that a specified output format is valid for write_mol_any 


Requires:&nbsp;&nbsp; format string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or 0 <br>

### <a name="anchor963"></a>Write_mol_single               
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Write a single molecule 


Requires:&nbsp;&nbsp; file_record, molecule, filename base (optional), file format (optional), filename append_string (optional), options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef <br>

### <a name="anchor964"></a>Write_mopac_cart               
/vcp1/people/david/bin/Silico1.15/lib/silico_mopac.pm 


Print out a mopac Cartesian format file 


Currently prints out only the first molecule of the ensemble 


Requires:&nbsp;&nbsp; Ensemble or molecule, filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open fails otherwise returns 1. <br>

### <a name="anchor965"></a>Write_pdb                      
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Write a pdb file 


Requires:&nbsp;&nbsp; Ensemble or molecule, filename, $options. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Zero if file open failed otherwise returns 1. <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; FAST, NOSORT, NOCONECT, BOND, MODEL <br>

Note: this routine renumbers the ensemble of pdb files 


Also note: the handling of pdb chains is somewhat confused 


A TER record will be written following an atom with atom->{TER} set 


### <a name="anchor966"></a>Write_pdb_cryst                
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Write PDB crystal record 


Requires:&nbsp;&nbsp; molecule, filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor967"></a>Write_pdb_molecule             
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Write a single record from a pdb file. 


Options:&nbsp;&nbsp;&nbsp;&nbsp; FAST, NOSORT, BOND, NOCONECT, DUCONECT, MODEL <br>
Requires:&nbsp;&nbsp; file record, molecule, options <br>

See read_pdb for general description 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef if failed <br>

### <a name="anchor968"></a>Write_pdb_sdfdata              
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Print SDF_data to pdb file 


New version November 2024 


Note. There is currently no read routine for this format 


Requires:&nbsp;&nbsp; field_name, data_to_print, filehandle, data level <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor969"></a>Write_pdbx                     
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Write a pdb file 


Requires:&nbsp;&nbsp; Ensemble or molecule, filename, $options. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Zero if file open failed otherwise returns 1. <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; BOND NOBOND QUIET <br>

### <a name="anchor970"></a>Write_pdbx_cryst               
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Write PDBX crystal record 


Requires:&nbsp;&nbsp; molecule, filehandle <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor971"></a>Write_pdbx_molecule            
/vcp1/people/david/bin/Silico1.15/lib/silico_pdb.pm 


Write a single record from a pdb file. 


Options:&nbsp;&nbsp;&nbsp;&nbsp; BOND NOBOND <br>
Requires:&nbsp;&nbsp; file record, molecule, options <br>

See read_pdb for general description 


Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or undef if failed <br>

### <a name="anchor972"></a>Write_pir                      
/vcp1/people/david/bin/Silico1.15/lib/silico_sequence.pm 


Write a pir format file (also known as NBRF format) that is compatible with 
Modeller 


Requires:&nbsp;&nbsp; Sequences, filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor973"></a>Write_pymol_load_file          
/vcp1/people/david/bin/Silico1.15/bin/mol_divide 


### <a name="anchor974"></a>Write_quick_in                 
/vcp1/people/david/bin/Silico1.15/lib/silico_quick.pm 


Write a Quick input format file 


Currently prints out only the first molecule of the ensemble 


Requires:&nbsp;&nbsp; Ensemble or molecule, filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open fails otherwise returns 1. <br>

### <a name="anchor975"></a>Write_rama_structure           
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Write out a _rama.txt file of a single structure suitable for the subroutine 
'ramaplot' 


Requires:&nbsp;&nbsp; molecule, filehandle, optional residue number (indexed from 0). <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor976"></a>Write_rama_structure_by_mol    
/vcp1/people/david/bin/Silico1.15/lib/silico_rama.pm 


Write out a .ramol file of a single structure with all phi and psi angles 
written on a single line 


Requires:&nbsp;&nbsp; molecule, file record, optional residue number (indexed from 0), options <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; 'CART' will output sin(phi), cos(phi), sin(psi), cos(psi) instead of phi, phsi <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor977"></a>Write_sdf                      
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Write an SDF file. 


Additional properties are written from mol->{SDF_DATA}{<propertyname>} 


All aromatic bonds are converted to kekule representation 


Requires:&nbsp;&nbsp; ensemble (or molecule), filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if file open fails otherwise returns one. <br>
Note:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Aromatic bondorders are only allowd in database searches. therefore aromatic bonds must be converted to single/double. This is currently done for aromatic rings, carboxylates and guanadines. Hence, database searches containing aromatic bonds will be modified adversely.  <br>

Options: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BOND - force bond generation  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOOND - do not generate bonds  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print 'Writing' line  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2D - Set 2D flag in header (default 3D). May be requred for CACTVS csbr 3D may be required to read in molecules to Insight  <br>

### <a name="anchor978"></a>Write_sdf                      
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Write an SDF file. 


Additional properties are written from mol->{SDF_DATA}{<propertyname>} 


All aromatic bonds are converted to kekule representation 


Requires:&nbsp;&nbsp; ensemble (or molecule), filename. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Undef if file open fails otherwise returns one. <br>
Note:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Aromatic bondorders are only allowd in database searches. therefore aromatic bonds must be converted to single/double. This is currently done for aromatic rings, carboxylates and guanadines. Hence, database searches containing aromatic bonds will be modified adversely.  <br>

Options: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BOND - force bond generation  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NOOND - do not generate bonds  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;QUIET - do not print 'Writing' line  <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2D - Set 2D flag in header (default 3D). May be requred for CACTVS csbr 3D may be required to read in molecules to Insight  <br>

### <a name="anchor979"></a>Write_sdf_atoms_bonds          
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm 


Write an MDL Mol record 


Note: 2D option forces all Z coordinates to zero 


Options:&nbsp;&nbsp;&nbsp;&nbsp; 3D (flag --sdf-3D), 2D (flag --sdf-2D), NO_SQUASH_HCOUNT (flag --sdf-squash-hcount) <br>
Requires:&nbsp;&nbsp; file record, molecule, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if error <br>

### <a name="anchor980"></a>Write_sdf_atoms_bonds          
/vcp1/people/david/bin/Silico1.15/lib/silico_sdf.pm.new2 


Write an MDL Mol record 


Note: 2D option forces all Z coordinates to zero 


Options:&nbsp;&nbsp;&nbsp;&nbsp; 3D (flag --sdf-3D), 2D (flag --sdf-2D), NO_SQUASH_HCOUNT (flag --sdf-squash-hcount) <br>
Requires:&nbsp;&nbsp; file record, molecule, options <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if error <br>

### <a name="anchor981"></a>Write_seq_any                  
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Write any supported sequence file type. 


Requires:&nbsp;&nbsp; sequence ensemble, filebase, format <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed or other problem <br>

### <a name="anchor982"></a>Write_seq_any_check            
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Check that a specified output format is valid for write_seq_any 


Requires:&nbsp;&nbsp; format string <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 or 0 <br>

### <a name="anchor983"></a>Write_smiles                   
/vcp1/people/david/bin/Silico1.15/lib/silico_smiles.pm 


Write an ensemble as a smiles file 


Basic implementation. Default is tab delimited 


Requires:&nbsp;&nbsp; ensemble or molecule, output filename, options, $b, $b, $k - smiles string options <br>
Options:&nbsp;&nbsp;&nbsp;&nbsp; QUIET - be quiet, <br>

NOHEADER - dont write a header, 


NODATA - don't write out molecule data 


DTAB - tab delimited (default) 


DCOMMA - comma delimited 


DSPACR - space delimieted 


### <a name="anchor984"></a>Write_starmaker_comment        
/vcp1/people/david/bin/Silico1.15/bin/starmaker 


Write comments that will appear in the output file 


Requires:&nbsp;&nbsp; molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; nothing <br>

### <a name="anchor985"></a>Write_table_delimited          
/vcp1/people/david/bin/Silico1.15/lib/silico_tabledata.pm 


Write a character delimited text file 


Requires:&nbsp;&nbsp; silico tabulated data structure, output filename, delimiter (default tab), options, list of keys to write out (default all). Missing values are written as a pair of single quotes ('') <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 1 if OK or undef if file open failed <br>

Options: None yet 


QUIET&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Do not print advisory messages <br>

### <a name="anchor986"></a>Write_textfile                 
/vcp1/people/david/bin/Silico1.15/lib/silico_io.pm 


Write a string to a file 


Requires:&nbsp;&nbsp; filename, text; <br>

### <a name="anchor987"></a>Write_tinker_prm               
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Write a TINKER force field parameter file 


Requires:&nbsp;&nbsp; Ensemble, filename <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; 0 if any molecules failed, 1 if success <br>

### <a name="anchor988"></a>Write_tinker_prm_mol_dihedrals_oplsaa 
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Writes out parameters for each dihedral type in the molecule 


Requires:&nbsp;&nbsp; Molecule <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Nothing <br>

### <a name="anchor989"></a>Write_tinker_prm_oplsaa        
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Write those parts of a TINKER parameter file that pertain to the OPLS-AA force 
field 


Requires:&nbsp;&nbsp; <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; <br>

### <a name="anchor990"></a>Write_tinker_xyz               
/vcp1/people/david/bin/Silico1.15/lib/silico_tinker.pm 


Tinker write routine. 


Requires:&nbsp;&nbsp; ensemble (or molecule), filename, forcefield file (optional) to type atoms. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed otherwise returns 1. <br>

### <a name="anchor991"></a>Write_xyz                      
/vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm 


Write an xyz file 


Requires:&nbsp;&nbsp; Ensemble or molecule, filename, $options. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; Zero if file open failed otherwise returns 1. <br>

### <a name="anchor992"></a>Write_xyz_molecule             
/vcp1/people/david/bin/Silico1.15/lib/silico_xyz.pm 


Write xyz molecule 


Requires:&nbsp;&nbsp; ensemble (or molecule), filename, forcefield file (optional) to type atoms. <br>
Returns:&nbsp;&nbsp;&nbsp;&nbsp; undef if file open failed otherwise returns 1. <br>
